<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>C API Reference &mdash; MNE 0.12.0 documentation</title>
    
    <link rel="stylesheet" href="../_static/basic.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/gallery.css" type="text/css" />
    <link rel="stylesheet" href="../_static/bootswatch-3.3.4/flatly/bootstrap.min.css" type="text/css" />
    <link rel="stylesheet" href="../_static/bootstrap-sphinx.css" type="text/css" />
    <link rel="stylesheet" href="../_static/style.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '0.12.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="../_static/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="../_static/js/jquery-fix.js"></script>
    <script type="text/javascript" src="../_static/bootstrap-3.3.4/js/bootstrap.min.js"></script>
    <script type="text/javascript" src="../_static/bootstrap-sphinx.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.ico"/>
    <link rel="top" title="MNE 0.12.0 documentation" href="../index.html" />
    <link rel="up" title="User Manual" href="index.html" />
    <link rel="next" title="Interactive analysis with mne_analyze" href="gui/analyze.html" />
    <link rel="prev" title="Pitfalls" href="pitfalls.html" />

<link href='http://fonts.googleapis.com/css?family=Open+Sans:400italic,700italic,400,700' rel='stylesheet' type='text/css'>


    <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-37225609-1']);
    _gaq.push(['_trackPageview']);

    (function() {
        var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
        ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
        var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
    </script>



    <script type="text/javascript">
    !function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0];if(!d.getElementById(id)){js=d.createElement(s);
    js.id=id;js.src="http://platform.twitter.com/widgets.js";
    fjs.parentNode.insertBefore(js,fjs);}}(document,"script","twitter-wjs");
    </script>



    <script type="text/javascript">
    (function() {
    var po = document.createElement('script'); po.type = 'text/javascript'; po.async = true;
    po.src = 'https://apis.google.com/js/plusone.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(po, s);
    })();
    </script>


  </head>
  <body>





  <div id="navbar" class="navbar navbar-default navbar-fixed-top">
    <div class="container">
      <div class="navbar-header">
        <!-- .btn-navbar is used as the toggle for collapsed navbar content -->
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".nav-collapse">
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="../index.html"><img src="../_static/mne_logo_small.png">
           </a>
        <span class="navbar-text navbar-version pull-left"><b>0.12.0</b></span>
      </div>

        <div class="collapse navbar-collapse nav-collapse">
          <ul class="nav navbar-nav">
            
                <li><a href="../getting_started.html">Get started</a></li>
                <li><a href="../tutorials.html">Tutorials</a></li>
                <li><a href="../auto_examples/index.html">Gallery</a></li>
                <li><a href="../python_reference.html">API</a></li>
                <li><a href="index.html">Manual</a></li>
                <li><a href="../faq.html">FAQ</a></li>
            
            
              <li class="dropdown globaltoc-container">
  <a role="button"
     id="dLabelGlobalToc"
     data-toggle="dropdown"
     data-target="#"
     href="../index.html">Site <b class="caret"></b></a>
  <ul class="dropdown-menu globaltoc"
      role="menu"
      aria-labelledby="dLabelGlobalToc"><ul>
<li class="toctree-l1"><a class="reference internal" href="../getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tutorials.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../auto_examples/index.html">Examples Gallery</a></li>
<li class="toctree-l1"><a class="reference internal" href="../faq.html">Frequently Asked Questions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../contributing.html">Contribute to MNE</a></li>
</ul>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../python_reference.html">Python API Reference</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="index.html">User Manual</a></li>
<li class="toctree-l1"><a class="reference internal" href="../whats_new.html">What&#8217;s new</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../cite.html">How to cite MNE</a></li>
<li class="toctree-l1"><a class="reference internal" href="../references.html">Related publications</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cited.html">Publications from MNE users</a></li>
</ul>
</ul>
</li>
              
                <li class="dropdown">
  <a role="button"
     id="dLabelLocalToc"
     data-toggle="dropdown"
     data-target="#"
     href="#">Page <b class="caret"></b></a>
  <ul class="dropdown-menu localtoc"
      role="menu"
      aria-labelledby="dLabelLocalToc"><ul>
<li><a class="reference internal" href="#">C API Reference</a><ul>
<li><a class="reference internal" href="#list-of-components">List of components</a><ul>
<li><a class="reference internal" href="#software-components">Software components</a></li>
<li><a class="reference internal" href="#utilities">Utilities</a></li>
</ul>
</li>
<li><a class="reference internal" href="#software-component-command-line-arguments">Software component command-line arguments</a><ul>
<li><a class="reference internal" href="#mne-analyze">mne_analyze</a></li>
<li><a class="reference internal" href="#mne-average-estimates">mne_average_estimates</a><ul>
<li><a class="reference internal" href="#the-description-file">The description file</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-browse-raw">mne_browse_raw</a></li>
<li><a class="reference internal" href="#mne-compute-mne">mne_compute_mne</a></li>
<li><a class="reference internal" href="#mne-compute-raw-inverse">mne_compute_raw_inverse</a></li>
<li><a class="reference internal" href="#mne-convert-mne-data">mne_convert_mne_data</a><ul>
<li><a class="reference internal" href="#guide-to-combining-options">Guide to combining options</a></li>
<li><a class="reference internal" href="#matlab-data-structures">Matlab data structures</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-do-forward-solution">mne_do_forward_solution</a></li>
<li><a class="reference internal" href="#mne-do-inverse-operator">mne_do_inverse_operator</a></li>
<li><a class="reference internal" href="#mne-forward-solution">mne_forward_solution</a></li>
<li><a class="reference internal" href="#mne-inverse-operator">mne_inverse_operator</a></li>
<li><a class="reference internal" href="#mne-make-movie">mne_make_movie</a><ul>
<li><a class="reference internal" href="#input-files">Input files</a></li>
<li><a class="reference internal" href="#times-and-baseline">Times and baseline</a></li>
<li><a class="reference internal" href="#options-controlling-the-estimates">Options controlling the estimates</a></li>
<li><a class="reference internal" href="#visualization-options">Visualization options</a></li>
<li><a class="reference internal" href="#thresholding">Thresholding</a></li>
<li><a class="reference internal" href="#output-files">Output files</a></li>
<li><a class="reference internal" href="#label-processing">Label processing</a></li>
<li><a class="reference internal" href="#using-stc-file-input">Using stc file input</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-make-source-space">mne_make_source_space</a></li>
<li><a class="reference internal" href="#mne-process-raw">mne_process_raw</a></li>
<li><a class="reference internal" href="#mne-redo-file">mne_redo_file</a></li>
<li><a class="reference internal" href="#mne-redo-file-nocwd">mne_redo_file_nocwd</a></li>
<li><a class="reference internal" href="#mne-setup-forward-model">mne_setup_forward_model</a></li>
<li><a class="reference internal" href="#mne-setup-mri">mne_setup_mri</a></li>
<li><a class="reference internal" href="#mne-setup-source-space">mne_setup_source_space</a></li>
<li><a class="reference internal" href="#mne-show-environment">mne_show_environment</a></li>
</ul>
</li>
<li><a class="reference internal" href="#utility-command-line-arguments">Utility command-line arguments</a><ul>
<li><a class="reference internal" href="#mne-add-patch-info">mne_add_patch_info</a><ul>
<li><a class="reference internal" href="#purpose">Purpose</a></li>
<li><a class="reference internal" href="#command-line-options">Command line options</a></li>
<li><a class="reference internal" href="#computational-details">Computational details</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-add-to-meas-info">mne_add_to_meas_info</a></li>
<li><a class="reference internal" href="#mne-add-triggers">mne_add_triggers</a><ul>
<li><a class="reference internal" href="#id23">Purpose</a></li>
<li><a class="reference internal" href="#id24">Command line options</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-annot2labels">mne_annot2labels</a></li>
<li><a class="reference internal" href="#mne-anonymize">mne_anonymize</a></li>
<li><a class="reference internal" href="#mne-average-forward-solutions">mne_average_forward_solutions</a></li>
<li><a class="reference internal" href="#mne-brain-vision2fiff">mne_brain_vision2fiff</a></li>
<li><a class="reference internal" href="#mne-change-baselines">mne_change_baselines</a></li>
<li><a class="reference internal" href="#mne-change-nave">mne_change_nave</a></li>
<li><a class="reference internal" href="#mne-check-eeg-locations">mne_check_eeg_locations</a></li>
<li><a class="reference internal" href="#mne-check-surface">mne_check_surface</a></li>
<li><a class="reference internal" href="#mne-collect-transforms">mne_collect_transforms</a></li>
<li><a class="reference internal" href="#mne-compensate-data">mne_compensate_data</a></li>
<li><a class="reference internal" href="#mne-copy-processing-history">mne_copy_processing_history</a></li>
<li><a class="reference internal" href="#mne-convert-dig-data">mne_convert_dig_data</a><ul>
<li><a class="reference internal" href="#the-hpts-format">The hpts format</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-convert-lspcov">mne_convert_lspcov</a></li>
<li><a class="reference internal" href="#mne-convert-ncov">mne_convert_ncov</a></li>
<li><a class="reference internal" href="#mne-convert-surface">mne_convert_surface</a><ul>
<li><a class="reference internal" href="#babeabaa">command-line options</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-cov2proj">mne_cov2proj</a><ul>
<li><a class="reference internal" href="#id42">Purpose</a></li>
<li><a class="reference internal" href="#id43">Command line options</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-create-comp-data">mne_create_comp_data</a></li>
<li><a class="reference internal" href="#mne-ctf2fiff">mne_ctf2fiff</a></li>
<li><a class="reference internal" href="#mne-ctf-dig2fiff">mne_ctf_dig2fiff</a></li>
<li><a class="reference internal" href="#mne-dicom-essentials">mne_dicom_essentials</a></li>
<li><a class="reference internal" href="#mne-edf2fiff">mne_edf2fiff</a><ul>
<li><a class="reference internal" href="#using-mne-edf2fiff">Using mne_edf2fiff</a></li>
<li><a class="reference internal" href="#post-conversion-tasks">Post-conversion tasks</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-epochs2mat">mne_epochs2mat</a><ul>
<li><a class="reference internal" href="#id50">Command-line options</a></li>
<li><a class="reference internal" href="#the-binary-epoch-data-file">The binary epoch data file</a></li>
<li><a class="reference internal" href="#id51">Matlab data structures</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-evoked-data-summary">mne_evoked_data_summary</a></li>
<li><a class="reference internal" href="#mne-eximia2fiff">mne_eximia2fiff</a></li>
<li><a class="reference internal" href="#mne-fit-sphere-to-surf">mne_fit_sphere_to_surf</a><ul>
<li><a class="reference internal" href="#id55">Purpose</a></li>
<li><a class="reference internal" href="#id56">Command line options</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-fix-mag-coil-types">mne_fix_mag_coil_types</a></li>
<li><a class="reference internal" href="#mne-fix-stim14">mne_fix_stim14</a></li>
<li><a class="reference internal" href="#mne-flash-bem">mne_flash_bem</a></li>
<li><a class="reference internal" href="#mne-insert-4d-comp">mne_insert_4D_comp</a></li>
<li><a class="reference internal" href="#mne-kit2fiff">mne_kit2fiff</a></li>
<li><a class="reference internal" href="#mne-list-bem">mne_list_bem</a></li>
<li><a class="reference internal" href="#mne-list-coil-def">mne_list_coil_def</a></li>
<li><a class="reference internal" href="#mne-list-proj">mne_list_proj</a></li>
<li><a class="reference internal" href="#mne-list-source-space">mne_list_source_space</a></li>
<li><a class="reference internal" href="#mne-list-versions">mne_list_versions</a></li>
<li><a class="reference internal" href="#mne-make-cor-set">mne_make_cor_set</a></li>
<li><a class="reference internal" href="#mne-make-derivations">mne_make_derivations</a><ul>
<li><a class="reference internal" href="#id69">Purpose</a></li>
<li><a class="reference internal" href="#id70">Command-line options</a></li>
<li><a class="reference internal" href="#derivation-file-formats">Derivation file formats</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-make-eeg-layout">mne_make_eeg_layout</a><ul>
<li><a class="reference internal" href="#id72">Purpose</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-make-morph-maps">mne_make_morph_maps</a></li>
<li><a class="reference internal" href="#mne-make-uniform-stc">mne_make_uniform_stc</a></li>
<li><a class="reference internal" href="#mne-mark-bad-channels">mne_mark_bad_channels</a></li>
<li><a class="reference internal" href="#mne-morph-labels">mne_morph_labels</a></li>
<li><a class="reference internal" href="#mne-organize-dicom">mne_organize_dicom</a></li>
<li><a class="reference internal" href="#mne-prepare-bem-model">mne_prepare_bem_model</a></li>
<li><a class="reference internal" href="#mne-process-stc">mne_process_stc</a></li>
<li><a class="reference internal" href="#mne-raw2mat">mne_raw2mat</a><ul>
<li><a class="reference internal" href="#id81">Command-line options</a></li>
<li><a class="reference internal" href="#id82">Matlab data structures</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-rename-channels">mne_rename_channels</a></li>
<li><a class="reference internal" href="#mne-sensitivity-map">mne_sensitivity_map</a><ul>
<li><a class="reference internal" href="#id85">Purpose</a></li>
<li><a class="reference internal" href="#id86">Command line options</a></li>
<li><a class="reference internal" href="#available-sensitivity-maps">Available sensitivity maps</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-sensor-locations">mne_sensor_locations</a></li>
<li><a class="reference internal" href="#mne-show-fiff">mne_show_fiff</a></li>
<li><a class="reference internal" href="#mne-simu">mne_simu</a><ul>
<li><a class="reference internal" href="#id90">Purpose</a></li>
<li><a class="reference internal" href="#id91">Command-line options</a></li>
<li><a class="reference internal" href="#noise-simulation">Noise simulation</a></li>
<li><a class="reference internal" href="#simulated-data">Simulated data</a></li>
<li><a class="reference internal" href="#source-waveform-expressions">Source waveform expressions</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-smooth">mne_smooth</a></li>
<li><a class="reference internal" href="#mne-surf2bem">mne_surf2bem</a><ul>
<li><a class="reference internal" href="#surface-options">Surface options</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-toggle-skips">mne_toggle_skips</a></li>
<li><a class="reference internal" href="#mne-transform-points">mne_transform_points</a><ul>
<li><a class="reference internal" href="#id96">Purpose</a></li>
<li><a class="reference internal" href="#id97">Command line options</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-tufts2fiff">mne_tufts2fiff</a></li>
<li><a class="reference internal" href="#mne-view-manual">mne_view_manual</a></li>
<li><a class="reference internal" href="#mne-volume-data2mri">mne_volume_data2mri</a></li>
<li><a class="reference internal" href="#mne-volume-source-space">mne_volume_source_space</a></li>
<li><a class="reference internal" href="#mne-watershed-bem">mne_watershed_bem</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</ul>
</li>
              
            
            
            
            
            
          </ul>

          
            
<form class="navbar-form navbar-right" action="../search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
          
        </div>
    </div>
  </div>

<div class="container">
  <div class="row">
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
            <p class="logo"><a href="../index.html">
              <img class="logo" src="../_static/mne_logo_small.png" alt="Logo"/>
            </a></p><ul>
<li><a class="reference internal" href="#">C API Reference</a><ul>
<li><a class="reference internal" href="#list-of-components">List of components</a><ul>
<li><a class="reference internal" href="#software-components">Software components</a></li>
<li><a class="reference internal" href="#utilities">Utilities</a></li>
</ul>
</li>
<li><a class="reference internal" href="#software-component-command-line-arguments">Software component command-line arguments</a><ul>
<li><a class="reference internal" href="#mne-analyze">mne_analyze</a></li>
<li><a class="reference internal" href="#mne-average-estimates">mne_average_estimates</a><ul>
<li><a class="reference internal" href="#the-description-file">The description file</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-browse-raw">mne_browse_raw</a></li>
<li><a class="reference internal" href="#mne-compute-mne">mne_compute_mne</a></li>
<li><a class="reference internal" href="#mne-compute-raw-inverse">mne_compute_raw_inverse</a></li>
<li><a class="reference internal" href="#mne-convert-mne-data">mne_convert_mne_data</a><ul>
<li><a class="reference internal" href="#guide-to-combining-options">Guide to combining options</a></li>
<li><a class="reference internal" href="#matlab-data-structures">Matlab data structures</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-do-forward-solution">mne_do_forward_solution</a></li>
<li><a class="reference internal" href="#mne-do-inverse-operator">mne_do_inverse_operator</a></li>
<li><a class="reference internal" href="#mne-forward-solution">mne_forward_solution</a></li>
<li><a class="reference internal" href="#mne-inverse-operator">mne_inverse_operator</a></li>
<li><a class="reference internal" href="#mne-make-movie">mne_make_movie</a><ul>
<li><a class="reference internal" href="#input-files">Input files</a></li>
<li><a class="reference internal" href="#times-and-baseline">Times and baseline</a></li>
<li><a class="reference internal" href="#options-controlling-the-estimates">Options controlling the estimates</a></li>
<li><a class="reference internal" href="#visualization-options">Visualization options</a></li>
<li><a class="reference internal" href="#thresholding">Thresholding</a></li>
<li><a class="reference internal" href="#output-files">Output files</a></li>
<li><a class="reference internal" href="#label-processing">Label processing</a></li>
<li><a class="reference internal" href="#using-stc-file-input">Using stc file input</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-make-source-space">mne_make_source_space</a></li>
<li><a class="reference internal" href="#mne-process-raw">mne_process_raw</a></li>
<li><a class="reference internal" href="#mne-redo-file">mne_redo_file</a></li>
<li><a class="reference internal" href="#mne-redo-file-nocwd">mne_redo_file_nocwd</a></li>
<li><a class="reference internal" href="#mne-setup-forward-model">mne_setup_forward_model</a></li>
<li><a class="reference internal" href="#mne-setup-mri">mne_setup_mri</a></li>
<li><a class="reference internal" href="#mne-setup-source-space">mne_setup_source_space</a></li>
<li><a class="reference internal" href="#mne-show-environment">mne_show_environment</a></li>
</ul>
</li>
<li><a class="reference internal" href="#utility-command-line-arguments">Utility command-line arguments</a><ul>
<li><a class="reference internal" href="#mne-add-patch-info">mne_add_patch_info</a><ul>
<li><a class="reference internal" href="#purpose">Purpose</a></li>
<li><a class="reference internal" href="#command-line-options">Command line options</a></li>
<li><a class="reference internal" href="#computational-details">Computational details</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-add-to-meas-info">mne_add_to_meas_info</a></li>
<li><a class="reference internal" href="#mne-add-triggers">mne_add_triggers</a><ul>
<li><a class="reference internal" href="#id23">Purpose</a></li>
<li><a class="reference internal" href="#id24">Command line options</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-annot2labels">mne_annot2labels</a></li>
<li><a class="reference internal" href="#mne-anonymize">mne_anonymize</a></li>
<li><a class="reference internal" href="#mne-average-forward-solutions">mne_average_forward_solutions</a></li>
<li><a class="reference internal" href="#mne-brain-vision2fiff">mne_brain_vision2fiff</a></li>
<li><a class="reference internal" href="#mne-change-baselines">mne_change_baselines</a></li>
<li><a class="reference internal" href="#mne-change-nave">mne_change_nave</a></li>
<li><a class="reference internal" href="#mne-check-eeg-locations">mne_check_eeg_locations</a></li>
<li><a class="reference internal" href="#mne-check-surface">mne_check_surface</a></li>
<li><a class="reference internal" href="#mne-collect-transforms">mne_collect_transforms</a></li>
<li><a class="reference internal" href="#mne-compensate-data">mne_compensate_data</a></li>
<li><a class="reference internal" href="#mne-copy-processing-history">mne_copy_processing_history</a></li>
<li><a class="reference internal" href="#mne-convert-dig-data">mne_convert_dig_data</a><ul>
<li><a class="reference internal" href="#the-hpts-format">The hpts format</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-convert-lspcov">mne_convert_lspcov</a></li>
<li><a class="reference internal" href="#mne-convert-ncov">mne_convert_ncov</a></li>
<li><a class="reference internal" href="#mne-convert-surface">mne_convert_surface</a><ul>
<li><a class="reference internal" href="#babeabaa">command-line options</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-cov2proj">mne_cov2proj</a><ul>
<li><a class="reference internal" href="#id42">Purpose</a></li>
<li><a class="reference internal" href="#id43">Command line options</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-create-comp-data">mne_create_comp_data</a></li>
<li><a class="reference internal" href="#mne-ctf2fiff">mne_ctf2fiff</a></li>
<li><a class="reference internal" href="#mne-ctf-dig2fiff">mne_ctf_dig2fiff</a></li>
<li><a class="reference internal" href="#mne-dicom-essentials">mne_dicom_essentials</a></li>
<li><a class="reference internal" href="#mne-edf2fiff">mne_edf2fiff</a><ul>
<li><a class="reference internal" href="#using-mne-edf2fiff">Using mne_edf2fiff</a></li>
<li><a class="reference internal" href="#post-conversion-tasks">Post-conversion tasks</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-epochs2mat">mne_epochs2mat</a><ul>
<li><a class="reference internal" href="#id50">Command-line options</a></li>
<li><a class="reference internal" href="#the-binary-epoch-data-file">The binary epoch data file</a></li>
<li><a class="reference internal" href="#id51">Matlab data structures</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-evoked-data-summary">mne_evoked_data_summary</a></li>
<li><a class="reference internal" href="#mne-eximia2fiff">mne_eximia2fiff</a></li>
<li><a class="reference internal" href="#mne-fit-sphere-to-surf">mne_fit_sphere_to_surf</a><ul>
<li><a class="reference internal" href="#id55">Purpose</a></li>
<li><a class="reference internal" href="#id56">Command line options</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-fix-mag-coil-types">mne_fix_mag_coil_types</a></li>
<li><a class="reference internal" href="#mne-fix-stim14">mne_fix_stim14</a></li>
<li><a class="reference internal" href="#mne-flash-bem">mne_flash_bem</a></li>
<li><a class="reference internal" href="#mne-insert-4d-comp">mne_insert_4D_comp</a></li>
<li><a class="reference internal" href="#mne-kit2fiff">mne_kit2fiff</a></li>
<li><a class="reference internal" href="#mne-list-bem">mne_list_bem</a></li>
<li><a class="reference internal" href="#mne-list-coil-def">mne_list_coil_def</a></li>
<li><a class="reference internal" href="#mne-list-proj">mne_list_proj</a></li>
<li><a class="reference internal" href="#mne-list-source-space">mne_list_source_space</a></li>
<li><a class="reference internal" href="#mne-list-versions">mne_list_versions</a></li>
<li><a class="reference internal" href="#mne-make-cor-set">mne_make_cor_set</a></li>
<li><a class="reference internal" href="#mne-make-derivations">mne_make_derivations</a><ul>
<li><a class="reference internal" href="#id69">Purpose</a></li>
<li><a class="reference internal" href="#id70">Command-line options</a></li>
<li><a class="reference internal" href="#derivation-file-formats">Derivation file formats</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-make-eeg-layout">mne_make_eeg_layout</a><ul>
<li><a class="reference internal" href="#id72">Purpose</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-make-morph-maps">mne_make_morph_maps</a></li>
<li><a class="reference internal" href="#mne-make-uniform-stc">mne_make_uniform_stc</a></li>
<li><a class="reference internal" href="#mne-mark-bad-channels">mne_mark_bad_channels</a></li>
<li><a class="reference internal" href="#mne-morph-labels">mne_morph_labels</a></li>
<li><a class="reference internal" href="#mne-organize-dicom">mne_organize_dicom</a></li>
<li><a class="reference internal" href="#mne-prepare-bem-model">mne_prepare_bem_model</a></li>
<li><a class="reference internal" href="#mne-process-stc">mne_process_stc</a></li>
<li><a class="reference internal" href="#mne-raw2mat">mne_raw2mat</a><ul>
<li><a class="reference internal" href="#id81">Command-line options</a></li>
<li><a class="reference internal" href="#id82">Matlab data structures</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-rename-channels">mne_rename_channels</a></li>
<li><a class="reference internal" href="#mne-sensitivity-map">mne_sensitivity_map</a><ul>
<li><a class="reference internal" href="#id85">Purpose</a></li>
<li><a class="reference internal" href="#id86">Command line options</a></li>
<li><a class="reference internal" href="#available-sensitivity-maps">Available sensitivity maps</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-sensor-locations">mne_sensor_locations</a></li>
<li><a class="reference internal" href="#mne-show-fiff">mne_show_fiff</a></li>
<li><a class="reference internal" href="#mne-simu">mne_simu</a><ul>
<li><a class="reference internal" href="#id90">Purpose</a></li>
<li><a class="reference internal" href="#id91">Command-line options</a></li>
<li><a class="reference internal" href="#noise-simulation">Noise simulation</a></li>
<li><a class="reference internal" href="#simulated-data">Simulated data</a></li>
<li><a class="reference internal" href="#source-waveform-expressions">Source waveform expressions</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-smooth">mne_smooth</a></li>
<li><a class="reference internal" href="#mne-surf2bem">mne_surf2bem</a><ul>
<li><a class="reference internal" href="#surface-options">Surface options</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-toggle-skips">mne_toggle_skips</a></li>
<li><a class="reference internal" href="#mne-transform-points">mne_transform_points</a><ul>
<li><a class="reference internal" href="#id96">Purpose</a></li>
<li><a class="reference internal" href="#id97">Command line options</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mne-tufts2fiff">mne_tufts2fiff</a></li>
<li><a class="reference internal" href="#mne-view-manual">mne_view_manual</a></li>
<li><a class="reference internal" href="#mne-volume-data2mri">mne_volume_data2mri</a></li>
<li><a class="reference internal" href="#mne-volume-source-space">mne_volume_source_space</a></li>
<li><a class="reference internal" href="#mne-watershed-bem">mne_watershed_bem</a></li>
</ul>
</li>
</ul>
</li>
</ul>

  <li>
    <a href="pitfalls.html" title="Previous Chapter: Pitfalls"><span class="glyphicon glyphicon-chevron-left visible-sm"></span><span class="hidden-sm hidden-tablet">&laquo; Pitfalls</span>
    </a>
  </li>
  <li>
    <a href="gui/analyze.html" title="Next Chapter: Interactive analysis with mne_analyze"><span class="glyphicon glyphicon-chevron-right visible-sm"></span><span class="hidden-sm hidden-tablet">Interactive a... &raquo;</span>
    </a>
  </li>
<form action="../search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
        </div>
      </div>
    <div class="col-md-12">
      
  <div class="section" id="c-api-reference">
<span id="c-reference"></span><h1>C API Reference<a class="headerlink" href="#c-api-reference" title="Permalink to this headline">¶</a></h1>
<p>Note that most programs have the options <tt class="docutils literal"><span class="pre">--version</span></tt> and <tt class="docutils literal"><span class="pre">--help</span></tt> which
give the version information and usage information, respectively.</p>
<div class="contents local topic" id="contents">
<p class="topic-title first">Contents</p>
<ul class="simple">
<li><a class="reference internal" href="#list-of-components" id="id103">List of components</a><ul>
<li><a class="reference internal" href="#software-components" id="id104">Software components</a></li>
<li><a class="reference internal" href="#utilities" id="id105">Utilities</a></li>
</ul>
</li>
<li><a class="reference internal" href="#software-component-command-line-arguments" id="id106">Software component command-line arguments</a><ul>
<li><a class="reference internal" href="#mne-analyze" id="id107">mne_analyze</a></li>
<li><a class="reference internal" href="#mne-average-estimates" id="id108">mne_average_estimates</a></li>
<li><a class="reference internal" href="#mne-browse-raw" id="id109">mne_browse_raw</a></li>
<li><a class="reference internal" href="#mne-compute-mne" id="id110">mne_compute_mne</a></li>
<li><a class="reference internal" href="#mne-compute-raw-inverse" id="id111">mne_compute_raw_inverse</a></li>
<li><a class="reference internal" href="#mne-convert-mne-data" id="id112">mne_convert_mne_data</a></li>
<li><a class="reference internal" href="#mne-do-forward-solution" id="id113">mne_do_forward_solution</a></li>
<li><a class="reference internal" href="#mne-do-inverse-operator" id="id114">mne_do_inverse_operator</a></li>
<li><a class="reference internal" href="#mne-forward-solution" id="id115">mne_forward_solution</a></li>
<li><a class="reference internal" href="#mne-inverse-operator" id="id116">mne_inverse_operator</a></li>
<li><a class="reference internal" href="#mne-make-movie" id="id117">mne_make_movie</a></li>
<li><a class="reference internal" href="#mne-make-source-space" id="id118">mne_make_source_space</a></li>
<li><a class="reference internal" href="#mne-process-raw" id="id119">mne_process_raw</a></li>
<li><a class="reference internal" href="#mne-redo-file" id="id120">mne_redo_file</a></li>
<li><a class="reference internal" href="#mne-redo-file-nocwd" id="id121">mne_redo_file_nocwd</a></li>
<li><a class="reference internal" href="#mne-setup-forward-model" id="id122">mne_setup_forward_model</a></li>
<li><a class="reference internal" href="#mne-setup-mri" id="id123">mne_setup_mri</a></li>
<li><a class="reference internal" href="#mne-setup-source-space" id="id124">mne_setup_source_space</a></li>
<li><a class="reference internal" href="#mne-show-environment" id="id125">mne_show_environment</a></li>
</ul>
</li>
<li><a class="reference internal" href="#utility-command-line-arguments" id="id126">Utility command-line arguments</a><ul>
<li><a class="reference internal" href="#mne-add-patch-info" id="id127">mne_add_patch_info</a></li>
<li><a class="reference internal" href="#mne-add-to-meas-info" id="id128">mne_add_to_meas_info</a></li>
<li><a class="reference internal" href="#mne-add-triggers" id="id129">mne_add_triggers</a></li>
<li><a class="reference internal" href="#mne-annot2labels" id="id130">mne_annot2labels</a></li>
<li><a class="reference internal" href="#mne-anonymize" id="id131">mne_anonymize</a></li>
<li><a class="reference internal" href="#mne-average-forward-solutions" id="id132">mne_average_forward_solutions</a></li>
<li><a class="reference internal" href="#mne-brain-vision2fiff" id="id133">mne_brain_vision2fiff</a></li>
<li><a class="reference internal" href="#mne-change-baselines" id="id134">mne_change_baselines</a></li>
<li><a class="reference internal" href="#mne-change-nave" id="id135">mne_change_nave</a></li>
<li><a class="reference internal" href="#mne-check-eeg-locations" id="id136">mne_check_eeg_locations</a></li>
<li><a class="reference internal" href="#mne-check-surface" id="id137">mne_check_surface</a></li>
<li><a class="reference internal" href="#mne-collect-transforms" id="id138">mne_collect_transforms</a></li>
<li><a class="reference internal" href="#mne-compensate-data" id="id139">mne_compensate_data</a></li>
<li><a class="reference internal" href="#mne-copy-processing-history" id="id140">mne_copy_processing_history</a></li>
<li><a class="reference internal" href="#mne-convert-dig-data" id="id141">mne_convert_dig_data</a></li>
<li><a class="reference internal" href="#mne-convert-lspcov" id="id142">mne_convert_lspcov</a></li>
<li><a class="reference internal" href="#mne-convert-ncov" id="id143">mne_convert_ncov</a></li>
<li><a class="reference internal" href="#mne-convert-surface" id="id144">mne_convert_surface</a></li>
<li><a class="reference internal" href="#mne-cov2proj" id="id145">mne_cov2proj</a></li>
<li><a class="reference internal" href="#mne-create-comp-data" id="id146">mne_create_comp_data</a></li>
<li><a class="reference internal" href="#mne-ctf2fiff" id="id147">mne_ctf2fiff</a></li>
<li><a class="reference internal" href="#mne-ctf-dig2fiff" id="id148">mne_ctf_dig2fiff</a></li>
<li><a class="reference internal" href="#mne-dicom-essentials" id="id149">mne_dicom_essentials</a></li>
<li><a class="reference internal" href="#mne-edf2fiff" id="id150">mne_edf2fiff</a></li>
<li><a class="reference internal" href="#mne-epochs2mat" id="id151">mne_epochs2mat</a></li>
<li><a class="reference internal" href="#mne-evoked-data-summary" id="id152">mne_evoked_data_summary</a></li>
<li><a class="reference internal" href="#mne-eximia2fiff" id="id153">mne_eximia2fiff</a></li>
<li><a class="reference internal" href="#mne-fit-sphere-to-surf" id="id154">mne_fit_sphere_to_surf</a></li>
<li><a class="reference internal" href="#mne-fix-mag-coil-types" id="id155">mne_fix_mag_coil_types</a></li>
<li><a class="reference internal" href="#mne-fix-stim14" id="id156">mne_fix_stim14</a></li>
<li><a class="reference internal" href="#mne-flash-bem" id="id157">mne_flash_bem</a></li>
<li><a class="reference internal" href="#mne-insert-4d-comp" id="id158">mne_insert_4D_comp</a></li>
<li><a class="reference internal" href="#mne-kit2fiff" id="id159">mne_kit2fiff</a></li>
<li><a class="reference internal" href="#mne-list-bem" id="id160">mne_list_bem</a></li>
<li><a class="reference internal" href="#mne-list-coil-def" id="id161">mne_list_coil_def</a></li>
<li><a class="reference internal" href="#mne-list-proj" id="id162">mne_list_proj</a></li>
<li><a class="reference internal" href="#mne-list-source-space" id="id163">mne_list_source_space</a></li>
<li><a class="reference internal" href="#mne-list-versions" id="id164">mne_list_versions</a></li>
<li><a class="reference internal" href="#mne-make-cor-set" id="id165">mne_make_cor_set</a></li>
<li><a class="reference internal" href="#mne-make-derivations" id="id166">mne_make_derivations</a></li>
<li><a class="reference internal" href="#mne-make-eeg-layout" id="id167">mne_make_eeg_layout</a></li>
<li><a class="reference internal" href="#mne-make-morph-maps" id="id168">mne_make_morph_maps</a></li>
<li><a class="reference internal" href="#mne-make-uniform-stc" id="id169">mne_make_uniform_stc</a></li>
<li><a class="reference internal" href="#mne-mark-bad-channels" id="id170">mne_mark_bad_channels</a></li>
<li><a class="reference internal" href="#mne-morph-labels" id="id171">mne_morph_labels</a></li>
<li><a class="reference internal" href="#mne-organize-dicom" id="id172">mne_organize_dicom</a></li>
<li><a class="reference internal" href="#mne-prepare-bem-model" id="id173">mne_prepare_bem_model</a></li>
<li><a class="reference internal" href="#mne-process-stc" id="id174">mne_process_stc</a></li>
<li><a class="reference internal" href="#mne-raw2mat" id="id175">mne_raw2mat</a></li>
<li><a class="reference internal" href="#mne-rename-channels" id="id176">mne_rename_channels</a></li>
<li><a class="reference internal" href="#mne-sensitivity-map" id="id177">mne_sensitivity_map</a></li>
<li><a class="reference internal" href="#mne-sensor-locations" id="id178">mne_sensor_locations</a></li>
<li><a class="reference internal" href="#mne-show-fiff" id="id179">mne_show_fiff</a></li>
<li><a class="reference internal" href="#mne-simu" id="id180">mne_simu</a></li>
<li><a class="reference internal" href="#mne-smooth" id="id181">mne_smooth</a></li>
<li><a class="reference internal" href="#mne-surf2bem" id="id182">mne_surf2bem</a></li>
<li><a class="reference internal" href="#mne-toggle-skips" id="id183">mne_toggle_skips</a></li>
<li><a class="reference internal" href="#mne-transform-points" id="id184">mne_transform_points</a></li>
<li><a class="reference internal" href="#mne-tufts2fiff" id="id185">mne_tufts2fiff</a></li>
<li><a class="reference internal" href="#mne-view-manual" id="id186">mne_view_manual</a></li>
<li><a class="reference internal" href="#mne-volume-data2mri" id="id187">mne_volume_data2mri</a></li>
<li><a class="reference internal" href="#mne-volume-source-space" id="id188">mne_volume_source_space</a></li>
<li><a class="reference internal" href="#mne-watershed-bem" id="id189">mne_watershed_bem</a></li>
</ul>
</li>
</ul>
</div>
<div class="section" id="list-of-components">
<h2><a class="toc-backref" href="#id103">List of components</a><a class="headerlink" href="#list-of-components" title="Permalink to this headline">¶</a></h2>
<div class="section" id="software-components">
<h3><a class="toc-backref" href="#id104">Software components</a><a class="headerlink" href="#software-components" title="Permalink to this headline">¶</a></h3>
<table border="1" class="docutils">
<colgroup>
<col width="39%" />
<col width="61%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Name</th>
<th class="head">Purpose</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><a class="reference internal" href="#mne-analyze">mne_analyze</a></td>
<td>An interactive analysis tool for computing
source estimates, see
<a class="reference internal" href="gui/analyze.html#ch-interactive-analysis"><em>Interactive analysis with mne_analyze</em></a>.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-average-estimates">mne_average_estimates</a></td>
<td>Average data across subjects.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-browse-raw">mne_browse_raw</a></td>
<td>Interactive raw data browser. Includes
filtering, offline averaging, and
computation of covariance matrices,
see <a class="reference internal" href="gui/browse.html#ch-browse"><em>Browsing raw data with mne_browse_raw</em></a>.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-compute-mne">mne_compute_mne</a></td>
<td>Computes the minimum-norm estimates,
Most functionality is included in
<a class="reference internal" href="#mne-make-movie"><em>mne_make_movie</em></a>.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-compute-raw-inverse">mne_compute_raw_inverse</a></td>
<td>Compute the inverse solution from raw data
see <a class="reference internal" href="source_localization/inverse.html#computing-inverse"><em>Computing inverse from raw and evoked data</em></a>.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-convert-mne-data">mne_convert_mne_data</a></td>
<td>Convert MNE data files to other file
formats.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-do-forward-solution">mne_do_forward_solution</a></td>
<td>Convenience script to calculate the forward
solution matrix, see <a class="reference internal" href="cookbook.html#babchejd"><em>Computing the forward solution</em></a>.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-do-inverse-operator">mne_do_inverse_operator</a></td>
<td>Convenience script for inverse operator
decomposition, see <a class="reference internal" href="cookbook.html#cihcfjei"><em>Calculating the inverse operator</em></a>.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-forward-solution">mne_forward_solution</a></td>
<td>Calculate the forward solution matrix, see
<a class="reference internal" href="source_localization/forward.html#chddibah"><em>Computing the forward solution</em></a>.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-inverse-operator">mne_inverse_operator</a></td>
<td>Compute the inverse operator decomposition
see <a class="reference internal" href="source_localization/inverse.html#inverse-operator"><em>Inverse-operator decomposition</em></a>.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-make-movie">mne_make_movie</a></td>
<td>Make movies in batch mode, see
<a class="reference internal" href="source_localization/inverse.html#movies-and-snapshots"><em>Producing movies and snapshots</em></a>.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-make-source-space">mne_make_source_space</a></td>
<td>Create a <em>fif</em> source space description
file, see <a class="reference internal" href="source_localization/forward.html#behcgjdd"><em>Creating a surface-based source space</em></a>.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-process-raw">mne_process_raw</a></td>
<td>A batch-mode version of mne_browse_raw,
see <a class="reference internal" href="gui/browse.html#ch-browse"><em>Browsing raw data with mne_browse_raw</em></a>.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-redo-file">mne_redo_file</a></td>
<td>Many intermediate result files contain a
description of their
&#8216;production environment&#8217;. Such files can
be recreated easily with this utility.
This is convenient if, for example,
the selection of bad channels is changed
and the inverse operator decomposition has
to be recalculated.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-redo-file-nocwd">mne_redo_file_nocwd</a></td>
<td>Works like mne_redo_file but does not try
to change in to the working directory
specified in the &#8216;production environment&#8217;.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-setup-forward-model">mne_setup_forward_model</a></td>
<td>Set up the BEM-related fif files,
see <a class="reference internal" href="cookbook.html#cihdbfeg"><em>Setting up the boundary-element model</em></a>.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-setup-mri">mne_setup_mri</a></td>
<td>A convenience script to create the fif
files describing the anatomical MRI data,
see <a class="reference internal" href="appendix/c_misc.html#babccehf"><em>Setting up anatomical MR images for MRIlab</em></a></td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-setup-source-space">mne_setup_source_space</a></td>
<td>A convenience script to create source space
description file, see
<a class="reference internal" href="cookbook.html#setting-up-source-space"><em>Setting up the source space</em></a>.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-show-environment">mne_show_environment</a></td>
<td>Show information about the production
environment of a file.</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="utilities">
<span id="ch-misc"></span><h3><a class="toc-backref" href="#id105">Utilities</a><a class="headerlink" href="#utilities" title="Permalink to this headline">¶</a></h3>
<table border="1" class="docutils" id="babdjhgh">
<colgroup>
<col width="44%" />
<col width="56%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Name</th>
<th class="head">Purpose</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><a class="reference internal" href="#mne-add-patch-info">mne_add_patch_info</a></td>
<td>Add neighborhood information to a source
space file.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-add-to-meas-info">mne_add_to_meas_info</a></td>
<td>Utility to add new information to the
measurement info block of a fif file. The
source of information is another fif file.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-add-triggers">mne_add_triggers</a></td>
<td>Modify the trigger channel STI 014 in a raw
data file. The same effect can be reached
by using an event file for averaging in
<a class="reference internal" href="#mne-process-raw"><em>mne_process_raw</em></a> and
<a class="reference internal" href="#mne-browse-raw"><em>mne_browse_raw</em></a>.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-annot2labels">mne_annot2labels</a></td>
<td>Convert parcellation data into label files.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-anonymize">mne_anonymize</a></td>
<td>Remove subject-specific information from a
fif data file.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-average-forward-solutions">mne_average_forward_solutions</a></td>
<td>Calculate an average of forward solutions,
see <a class="reference internal" href="source_localization/forward.html#chdbbfca"><em>Averaging forward solutions</em></a>.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-brain-vision2fiff">mne_brain_vision2fiff</a></td>
<td>Convert EEG data from BrainVision format
to fif format.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-change-baselines">mne_change_baselines</a></td>
<td>Change the dc offsets according to
specifications given in a text file.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-change-nave">mne_change_nave</a></td>
<td>Change the number of averages in an
evoked-response data file. This is often
necessary if the file was derived from
several files.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-check-eeg-locations">mne_check_eeg_locations</a></td>
<td>Checks that the EEG electrode locations
have been correctly transferred from the
Polhemus data block to the channel
information tags</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-check-surface">mne_check_surface</a></td>
<td>Check the validity of a FreeSurfer surface
file or one of the surfaces within a BEM
file. This program simply checks for
topological errors in surface files.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-collect-transforms">mne_collect_transforms</a></td>
<td>Collect coordinate transformations from
several sources into a single fif file.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-compensate-data">mne_compensate_data</a></td>
<td>Change the applied software gradient
compensation in an evoked-response data
file, see <a class="reference internal" href="io.html#behddfbi"><em>Applying software gradient compensation</em></a>.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-copy-processing-history">mne_copy_processing_history</a></td>
<td>Copy the processing history between files.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-convert-dig-data">mne_convert_dig_data</a></td>
<td>Convert digitization data between
different formats.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-convert-lspcov">mne_convert_lspcov</a></td>
<td>Convert the LISP format noise covariance
matrix output by graph into fif.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-convert-ncov">mne_convert_ncov</a></td>
<td>Convert the ncov format noise covariance
file to fif.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-convert-surface">mne_convert_surface</a></td>
<td>Convert FreeSurfer and text format surface
files into Matlab mat files.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-cov2proj">mne_cov2proj</a></td>
<td>Pick eigenvectors from a covariance matrix
and create a signal-space projection (SSP)
file out of them.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-create-comp-data">mne_create_comp_data</a></td>
<td>Create a fif file containing software
gradient compensation information from a
text file.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-ctf2fiff">mne_ctf2fiff</a></td>
<td>Convert a CTF ds folder into a fif file.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-ctf-dig2fiff">mne_ctf_dig2fiff</a></td>
<td>Convert text format digitization data to
fif format.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-dicom-essentials">mne_dicom_essentials</a></td>
<td>List essential information from a
DICOM file.
This utility is used by the script
mne_organize_dicom, see <a class="reference internal" href="appendix/bem_model.html#babebjhi"><em>Organizing MRI data into directories</em></a>.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-edf2fiff">mne_edf2fiff</a></td>
<td>Convert EEG data from the EDF/EDF+/BDF
formats to the fif format.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-epochs2mat">mne_epochs2mat</a></td>
<td>Apply bandpass filter to raw data and
extract epochs for subsequent processing
in Matlab.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-evoked-data-summary">mne_evoked_data_summary</a></td>
<td>List summary of averaged data from a fif
file to the standard output.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-eximia2fiff">mne_eximia2fiff</a></td>
<td>Convert EEG data from the Nexstim eXimia
system to fif format.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-fit-sphere-to-surf">mne_fit_sphere_to_surf</a></td>
<td>Fit a sphere to a surface given in fif
or FreeSurfer format.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-fix-mag-coil-types">mne_fix_mag_coil_types</a></td>
<td>Update the coil types for magnetometers
in a fif file.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-fix-stim14">mne_fix_stim14</a></td>
<td>Fix coding errors of trigger channel
STI 014, see <a class="reference internal" href="appendix/c_misc.html#babcdbdi"><em>Cleaning the digital trigger channel</em></a>.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-flash-bem">mne_flash_bem</a></td>
<td>Create BEM tessellation using multi-echo
FLASH MRI data, see <a class="reference internal" href="appendix/bem_model.html#babfcdjh"><em>Using FLASH images</em></a>.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-insert-4d-comp">mne_insert_4D_comp</a></td>
<td>Read Magnes compensation channel data from
a text file and merge it with raw data
from other channels in a fif file.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-kit2fiff">mne_kit2fiff</a></td>
<td>Convert KIT data to FIF.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-list-bem">mne_list_bem</a></td>
<td>List BEM information in text format.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-list-coil-def">mne_list_coil_def</a></td>
<td>Create the coil description file. This
is run automatically at when the software
is set up, see <a class="reference internal" href="source_localization/forward.html#bjehhjij"><em>Creating the coil definition file</em></a>.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-list-proj">mne_list_proj</a></td>
<td>List signal-space projection data from a
fif file.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-list-source-space">mne_list_source_space</a></td>
<td>List source space information in text
format suitable for importing into
Neuromag MRIlab.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-list-versions">mne_list_versions</a></td>
<td>List versions and compilation dates of MNE
software modules.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-make-cor-set">mne_make_cor_set</a></td>
<td>Used by mne_setup_mri to create fif format
MRI description files from COR or mgh/mgz
format MRI data, see <a class="reference internal" href="appendix/c_misc.html#babccehf"><em>Setting up anatomical MR images for MRIlab</em></a>.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-make-derivations">mne_make_derivations</a></td>
<td>Create a channel derivation data file.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-make-eeg-layout">mne_make_eeg_layout</a></td>
<td>Make a topographical trace layout file
using the EEG electrode locations from
an actual measurement.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-make-morph-maps">mne_make_morph_maps</a></td>
<td>Precompute the mapping data needed for
morphing between subjects, see
<a class="reference internal" href="source_localization/morph.html#chdbbhdh"><em>Precomputing the morphing maps</em></a>.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-make-uniform-stc">mne_make_uniform_stc</a></td>
<td>Create a spatially uniform stc file for
testing purposes.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-mark-bad-channels">mne_mark_bad_channels</a></td>
<td>Update the list of unusable channels in
a data file</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-morph-labels">mne_morph_labels</a></td>
<td>Morph label file definitions between
subjects.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-organize-dicom">mne_organize_dicom</a></td>
<td>Organized DICOM MRI image files into
directories, see <a class="reference internal" href="appendix/bem_model.html#babebjhi"><em>Organizing MRI data into directories</em></a>.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-prepare-bem-model">mne_prepare_bem_model</a></td>
<td>Perform the geometry calculations for
BEM forward solutions, see <a class="reference internal" href="source_localization/forward.html#chdjfheb"><em>Computing the BEM geometry data</em></a>.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-process-stc">mne_process_stc</a></td>
<td>Manipulate stc files.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-raw2mat">mne_raw2mat</a></td>
<td>Convert raw data into a Matlab file.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-rename-channels">mne_rename_channels</a></td>
<td>Change the names and types of channels
in a fif file.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-sensitivity-map">mne_sensitivity_map</a></td>
<td>Compute a sensitivity map and output
the result in a w-file.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-sensor-locations">mne_sensor_locations</a></td>
<td>Create a file containing the sensor
locations in text format.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-show-fiff">mne_show_fiff</a></td>
<td>List contents of a fif file.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-simu">mne_simu</a></td>
<td>Simulate MEG and EEG data.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-smooth">mne_smooth</a></td>
<td>Smooth a w or stc file.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-surf2bem">mne_surf2bem</a></td>
<td>Create a <em>fif</em> file describing the
triangulated compartment boundaries for
the boundary-element model (BEM),
see <a class="reference internal" href="source_localization/forward.html#behcaccj"><em>Creating the BEM meshes</em></a>.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-toggle-skips">mne_toggle_skips</a></td>
<td>Change data skip tags in a raw file into
ignored skips or vice versa.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-transform-points">mne_transform_points</a></td>
<td>Transform between MRI and MEG head
coordinate frames.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-tufts2fiff">mne_tufts2fiff</a></td>
<td>Convert EEG data from the Tufts
University format to fif format.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-view-manual">mne_view_manual</a></td>
<td>Starts a PDF reader to show this manual
from its standard location.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-volume-data2mri">mne_volume_data2mri</a></td>
<td>Convert volumetric data defined in a
source space created with
mne_volume_source_space into an MRI
overlay.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#mne-volume-source-space">mne_volume_source_space</a></td>
<td>Make a volumetric source space,
see <a class="reference internal" href="source_localization/forward.html#bjefehji"><em>Creating a volumetric or discrete source space</em></a>.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#mne-watershed-bem">mne_watershed_bem</a></td>
<td>Do the segmentation for BEM using the
watershed algorithm, see <a class="reference internal" href="appendix/bem_model.html#babbdhag"><em>Using the watershed algorithm</em></a>.</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section" id="software-component-command-line-arguments">
<h2><a class="toc-backref" href="#id106">Software component command-line arguments</a><a class="headerlink" href="#software-component-command-line-arguments" title="Permalink to this headline">¶</a></h2>
<div class="section" id="mne-analyze">
<span id="id1"></span><h3><a class="toc-backref" href="#id107">mne_analyze</a><a class="headerlink" href="#mne-analyze" title="Permalink to this headline">¶</a></h3>
<p>Since mne_analyze is primarily an interactive analysis tool, there are only a
few command-line options:</p>
<p><tt class="docutils literal"><span class="pre">\---cd</span> <span class="pre">&lt;*dir*&gt;</span></tt></p>
<blockquote>
<div>Change to this directory before starting.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">\---subject</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specify the default subject name for surface loading.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">\---digtrig</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Name of the digital trigger channel. The default value is &#8216;STI
014&#8217;. Underscores in the channel name will be replaced
by spaces.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">\---digtrigmask</span> <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>Mask to be applied to the raw data trigger channel values before considering
them. This option is useful if one wants to set some bits in a don&#8217;t
care state. For example, some finger response pads keep the trigger
lines high if not in use, <em>i.e.</em>, a finger is
not in place. Yet, it is convenient to keep these devices permanently
connected to the acquisition system. The number can be given in
decimal or hexadecimal format (beginning with 0x or 0X). For example,
the value 255 (0xFF) means that only the lowest order byte (usually
trigger lines 1 - 8 or bits 0 - 7) will be considered.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">\---visualizehpi</span></tt></p>
<blockquote>
<div>Start mne_analyze in the restricted <em>head
position visualization</em> mode. For details, see <a class="reference internal" href="gui/analyze.html#chdedfae"><em>Visualizing the head position</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">\---dig</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>Specify a file containing the head shape digitization data. This option
is only usable if the <em>head position visualization</em> position
visualization mode has been first invoked with the &#8211;visualizehpi
option.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">\---hpi</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>Specify a file containing the transformation between the MEG device
and head coordinate frames. This option is only usable if the <em>head
position visualization</em> position visualization mode has
been first invoked with the <tt class="docutils literal"><span class="pre">--visualizehpi</span></tt> option.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">\---scalehead</span></tt></p>
<blockquote>
<div>In <em>head position visualization</em> mode, scale
the average scalp surface according to the head surface digitization
data before aligning  them to the scalp surface. This option is
recommended.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">\---rthelmet</span></tt></p>
<blockquote>
<div>Use the room-temperature helmet surface instead of the MEG sensor
surface when showing the relative position of the MEG sensors and
the head in the <em>head position visualization</em> mode.</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Before starting mne_analyze the <tt class="docutils literal"><span class="pre">SUBJECTS_DIR</span></tt> environment variable has to be set.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Strictly speaking, trigger mask value zero would mean that all trigger inputs are ignored. However, for convenience,    setting the mask to zero or not setting it at all has the same effect    as 0xFFFFFFFF, <em>i.e.</em>, all bits set.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The digital trigger channel can also be set with the MNE_TRIGGER_CH_NAME environment variable. Underscores in the variable    value will <em>not</em> be replaced with spaces by mne_analyze .    Using the <tt class="docutils literal"><span class="pre">--digtrig</span></tt> option supersedes the MNE_TRIGGER_CH_NAME    environment variable.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The digital trigger channel mask can also be set with the MNE_TRIGGER_CH_MASK environment variable. Using the <tt class="docutils literal"><span class="pre">--digtrigmask</span></tt> option    supersedes the MNE_TRIGGER_CH_MASK environment variable.</p>
</div>
</div>
<div class="section" id="mne-average-estimates">
<span id="id2"></span><h3><a class="toc-backref" href="#id108">mne_average_estimates</a><a class="headerlink" href="#mne-average-estimates" title="Permalink to this headline">¶</a></h3>
<p>This is a utility for averaging data in stc files. It requires that
all stc files represent data on one individual&#8217;s cortical
surface and contain identical sets of vertices. mne_average_estimates uses
linear interpolation to resample data in time as necessary. The
command line arguments are:</p>
<p><tt class="docutils literal"><span class="pre">---desc</span> <span class="pre">&lt;filenname&gt;</span></tt></p>
<blockquote>
<div>Specifies the description file for averaging. The format of this
file is described below.</div></blockquote>
<div class="section" id="the-description-file">
<h4>The description file<a class="headerlink" href="#the-description-file" title="Permalink to this headline">¶</a></h4>
<p>The description file for mne_average_estimates consists
of a sequence of tokens, separated by whitespace (space, tab, or
newline). If a token consists of several words it has to be enclosed
in quotes. One or more tokens constitute an phrase, which has a
meaning for the averaging definition. Any line starting with the
pound sign (#) is a considered to be a comment line. There are two
kinds of phrases in the description file: global and contextual.
The global phrases have the same meaning independent on their location
in the file while the contextual phrases have different effects depending
on their location in the file.</p>
<p>There are three types of contexts in the description file:
the global context, an input context,
and the output context. In the
beginning of the file the context is global for
defining global parameters. The input context
defines one of the input files (subjects) while the output context
specifies the destination for the average.</p>
<p>The global phrases are:</p>
<p><tt class="docutils literal"><span class="pre">tmin</span> <span class="pre">&lt;*value/ms*&gt;</span></tt></p>
<blockquote>
<div>The minimum time to be considered. The output stc file starts at
this time point if the time ranges of the stc files include this
time. Otherwise the output starts from the next later available
time point.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">tstep</span> <span class="pre">&lt;*step/ms*&gt;</span></tt></p>
<blockquote>
<div>Time step between consecutive movie frames, specified in milliseconds.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">tmax</span> <span class="pre">&lt;*value/ms*&gt;</span></tt></p>
<blockquote>
<div>The maximum time point to be considered. A multiple of tstep will be
added to the first time point selected until this value or the last time
point in one of the input stc files is reached.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">integ</span>&nbsp; <span class="pre">&lt;:math:`\Delta</span> <span class="pre">t`</span> <span class="pre">/*ms*&gt;</span></tt></p>
<blockquote>
<div>Integration time for each frame. Defaults to zero. The integration will
be performed on sensor data. If the time specified for a frame is <span class="math">\(t_0\)</span>,
the integration range will be <span class="math">\(t_0 - ^{\Delta t}/_2 \leq t \leq t_0 + ^{\Delta t}/_2\)</span>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">stc</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>Specifies an input stc file. The filename can be specified with
one of the <tt class="docutils literal"><span class="pre">-lh.stc</span></tt> and <tt class="docutils literal"><span class="pre">-rh.stc</span></tt> endings
or without them. This phrase ends the present context and starts
an input context.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">deststc</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>Specifies the output stc file. The filename can be specified with
one of the <tt class="docutils literal"><span class="pre">-lh.stc</span></tt> and <tt class="docutils literal"><span class="pre">-rh.stc</span></tt> endings
or without them. This phrase ends the present context and starts
the output context.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">lh</span></tt></p>
<blockquote>
<div>Process the left hemisphere. By default, both hemispheres are processed.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">rh</span></tt></p>
<blockquote>
<div>Process the left hemisphere. By default, both hemispheres are processed.</div></blockquote>
<p>The contextual phrases are:</p>
<p><tt class="docutils literal"><span class="pre">weight</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>Specifies the weight of the current data set. This phrase is valid
in the input and output contexts.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">abs</span></tt></p>
<blockquote>
<div>Specifies that the absolute value of the data should be taken. Valid
in all contexts. If specified in the global context, applies to
all subsequent input and output contexts. If specified in the input
or output contexts, applies only to the data associated with that
context.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">pow</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>Specifies that the data should raised to the specified power. For
negative values, the absolute value of the data will be taken and
the negative sign will be transferred to the result, unless abs is
specified. Valid in all contexts. Rules of application are identical
to abs .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">sqrt</span></tt></p>
<blockquote>
<div>Means pow 0.5</div></blockquote>
<p>The effects of the options can be summarized as follows.
Suppose that the description file includes <span class="math">\(P\)</span> contexts
and the temporally resampled data are organized in matrices <span class="math">\(S^{(p)}\)</span>,
where <span class="math">\(p = 1 \dotso P\)</span> is the subject index, and
the rows are the signals at different vertices of the cortical surface.
The average computed by mne_average_estimates is
then:</p>
<div class="math">
\[A_{jk} = |w[\newcommand\sgn{\mathop{\mathrm{sgn}}\nolimits}\sgn(B_{jk})]^{\alpha}|B_{jk}|^{\beta}\]</div>
<p>with</p>
<div class="math">
\[B_{jk} = \sum_{p = 1}^p {\bar{w_p}[\newcommand\sgn{\mathop{\mathrm{sgn}}\nolimits}\sgn(S_{jk}^{(p)})^{\alpha_p}|S_{jk}^{(p)}|^{\beta_p}}\]</div>
<p>and</p>
<div class="math">
\[\bar{w_p} = w_p / \sum_{p = 1}^p {|w_p|}\ .\]</div>
<p>In the above, <span class="math">\(\beta_p\)</span> and <span class="math">\(w_p\)</span> are
the powers and weights assigned to each of the subjects whereas <span class="math">\(\beta\)</span> and <span class="math">\(w\)</span> are
the output weight and power value, respectively. The sign is either
included (<span class="math">\(\alpha_p = 1\)</span>, <span class="math">\(\alpha = 1\)</span>)
or omitted (<span class="math">\(\alpha_p = 2\)</span>, <span class="math">\(\alpha = 2\)</span>)
depending on the presence of abs phrases in the description file.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">mne_average_estimates requires    that the number of vertices in the stc files are the same and that    the vertex numbers are identical. This will be the case if the files    have been produced in mne_make_movie using    the <tt class="docutils literal"><span class="pre">--morph</span></tt> option.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">It is straightforward to read and write stc    files using the MNE Matlab toolbox described in <a class="reference internal" href="matlab.html#ch-matlab"><em>Matlab toolbox</em></a> and    thus write custom Matlab functions to realize more complicated custom    group analysis tools.</p>
</div>
</div>
</div>
<div class="section" id="mne-browse-raw">
<span id="id3"></span><h3><a class="toc-backref" href="#id109">mne_browse_raw</a><a class="headerlink" href="#mne-browse-raw" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--cd</span> <span class="pre">&lt;*dir*&gt;</span></tt></p>
<blockquote>
<div>Change to this directory before starting.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--raw</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the raw data file to be opened. If a raw data file is not
specified, an empty interactive browser will open.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--grad</span> <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>Apply software gradient compensation of the given order to the data loaded
with the <tt class="docutils literal"><span class="pre">--raw</span></tt> option. This option is effective only
for data acquired with the CTF and 4D Magnes MEG systems. If orders
different from zero are requested for Neuromag data, an error message appears
and data are not loaded. Any compensation already existing in the
file can be undone or changed to another order by using an appropriate <tt class="docutils literal"><span class="pre">--grad</span></tt> options.
Possible orders are 0 (No compensation), 1 - 3 (CTF data), and 101
(Magnes data). This applies only to the data file loaded by specifying the <tt class="docutils literal"><span class="pre">--raw</span></tt> option.
For interactive data loading, the software gradient compensation
is specified in the corresponding file selection dialog, see <a class="reference internal" href="gui/browse.html#cacdchaj"><em>Open</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--filtersize</span> <span class="pre">&lt;*size*&gt;</span></tt></p>
<blockquote>
<div>Adjust the length of the FFT to be applied in filtering. The number will
be rounded up to the next power of two. If the size is <span class="math">\(N\)</span>,
the corresponding length of time is <span class="math">\(N/f_s\)</span>,
where <span class="math">\(f_s\)</span> is the sampling frequency
of your data. The filtering procedure includes overlapping tapers
of length <span class="math">\(N/2\)</span> so that the total FFT
length will actually be <span class="math">\(2N\)</span>. This
value cannot be changed after the program has been started.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--highpass</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>Highpass filter frequency limit. If this is too low with respect
to the selected FFT length and, the data will not be highpass filtered. It
is best to experiment with the interactive version to find the lowest applicable
filter for your data. This value can be adjusted in the interactive
version of the program. The default is 0, <em>i.e.</em>,
no highpass filter apart from that used during the acquisition will
be in effect.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--highpassw</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>The width of the transition band of the highpass filter. The default
is 6 frequency bins, where one bin is <span class="math">\(f_s / (2N)\)</span>. This
value cannot be adjusted in the interactive version of the program.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--lowpass</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>Lowpass filter frequency limit. This value can be adjusted in the interactive
version of the program. The default is 40 Hz.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--lowpassw</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>The width of the transition band of the lowpass filter. This value
can be adjusted in the interactive version of the program. The default
is 5 Hz.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eoghighpass</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>Highpass filter frequency limit for EOG. If this is too low with respect
to the selected FFT length and, the data will not be highpass filtered.
It is best to experiment with the interactive version to find the
lowest applicable filter for your data. This value can be adjusted in
the interactive version of the program. The default is 0, <em>i.e.</em>,
no highpass filter apart from that used during the acquisition will
be in effect.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eoghighpassw</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>The width of the transition band of the EOG highpass filter. The default
is 6 frequency bins, where one bin is <span class="math">\(f_s / (2N)\)</span>.
This value cannot be adjusted in the interactive version of the
program.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eoglowpass</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>Lowpass filter frequency limit for EOG. This value can be adjusted in
the interactive version of the program. The default is 40 Hz.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eoglowpassw</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>The width of the transition band of the EOG lowpass filter. This value
can be adjusted in the interactive version of the program. The default
is 5 Hz.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--filteroff</span></tt></p>
<blockquote>
<div>Do not filter the data. This initial value can be changed in the
interactive version of the program.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--digtrig</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Name of the composite digital trigger channel. The default value
is &#8216;STI 014&#8217;. Underscores in the channel name
will be replaced by spaces.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--digtrigmask</span> <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>Mask to be applied to the trigger channel values before considering them.
This option is useful if one wants to set some bits in a don&#8217;t care
state. For example, some finger response pads keep the trigger lines
high if not in use, <em>i.e.</em>, a finger is not in
place. Yet, it is convenient to keep these devices permanently connected
to the acquisition system. The number can be given in decimal or
hexadecimal format (beginning with 0x or 0X). For example, the value
255 (0xFF) means that only the lowest order byte (usually trigger
lines 1 - 8 or bits 0 - 7) will be considered.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--allowmaxshield</span></tt></p>
<blockquote>
<div>Allow loading of unprocessed Elekta-Neuromag data with MaxShield
on. These kind of data should never be used for source localization
without further processing with Elekta-Neuromag software.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--deriv</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of a derivation file. This overrides the use
of a standard derivation file, see <a class="reference internal" href="gui/browse.html#cacfhafh"><em>Load derivations</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--sel</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the channel selection file to be used. This overrides
the use of the standard channel selection files, see <a class="reference internal" href="gui/browse.html#caccjejd"><em>Selection</em></a>.</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Strictly speaking, trigger mask value zero would mean that all trigger inputs are ignored. However, for convenience,    setting the mask to zero or not setting it at all has the same effect    as 0xFFFFFFFF, <em>i.e.</em>, all bits set.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The digital trigger channel can also be set with the MNE_TRIGGER_CH_NAME environment variable. Underscores in the variable value will <em>not</em> be replaced with spaces. Using the <tt class="docutils literal"><span class="pre">--digtrig</span></tt> option supersedes the MNE_TRIGGER_CH_NAME    environment variable.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The digital trigger channel mask can also be set with the MNE_TRIGGER_CH_MASK environment variable. Using the <tt class="docutils literal"><span class="pre">--digtrigmask</span></tt> option    supersedes the MNE_TRIGGER_CH_MASK environment variable.</p>
</div>
</div>
<div class="section" id="mne-compute-mne">
<span id="id4"></span><h3><a class="toc-backref" href="#id110">mne_compute_mne</a><a class="headerlink" href="#mne-compute-mne" title="Permalink to this headline">¶</a></h3>
<p>This program is gradually becoming obsolete. All of its functions will
be eventually included to <a class="reference internal" href="#mne-make-movie"><em>mne_make_movie</em></a>,
see <a class="reference internal" href="source_localization/inverse.html#movies-and-snapshots"><em>Producing movies and snapshots</em></a>. At this time, <a class="reference internal" href="#mne-compute-mne"><em>mne_compute_mne</em></a> is
still needed to produce time-collapsed w files unless you are willing
to write a Matlab script of your own for this purpose.</p>
<p><tt class="docutils literal"><span class="pre">--inv</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Load the inverse operator decomposition from here.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--meas</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Load the MEG or EEG data from this file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--set</span> <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>The data set (condition) number to load. The list of data sets can
be seen, <em>e.g.</em>, in mne_analyze , mne_browse_raw ,
and xplotter .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--bmin</span> <span class="pre">&lt;*time/ms*&gt;</span></tt></p>
<blockquote>
<div>Specifies the starting time of the baseline. In order to activate
baseline correction, both <tt class="docutils literal"><span class="pre">--bmin</span></tt> and <tt class="docutils literal"><span class="pre">--bmax</span></tt> options
must be present.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--bmax</span> <span class="pre">&lt;*time/ms*&gt;</span></tt></p>
<blockquote>
<div>Specifies the finishing time of the baseline.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--nave</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>Specifies the number of averaged epochs in the input data. If the input
data file is one produced by mne_process_raw or mne_browse_raw ,
the number of averages is correct in the file. However, if subtractions
or some more complicated combinations of simple averages are produced, <em>e.g.</em>,
by using the xplotter software, the
number of averages should be manually adjusted. This is accomplished
either by employing this flag or by adjusting the number of averages
in the data file with help of mne_change_nave .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--snr</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>An estimate for the amplitude SNR. The regularization parameter will
be set as <span class="math">\(\lambda = ^1/_{\text{SNR}}\)</span>. If the SNR option is
absent, the regularization parameter will be estimated from the
data. The regularization parameter will be then time dependent.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--snronly</span></tt></p>
<blockquote>
<div>Only estimate SNR and output the result into a file called SNR. Each
line of the file contains three values: the time point in ms, the estimated
SNR + 1, and the regularization parameter estimated from the data
at this time point.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--abs</span></tt></p>
<blockquote>
<div>Calculate the absolute value of the current and the dSPM for fixed-orientation
data.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--spm</span></tt></p>
<blockquote>
<div>Calculate the dSPM instead of the expected current value.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--chi2</span></tt></p>
<blockquote>
<div>Calculate an approximate <span class="math">\(\chi_2^3\)</span> statistic
instead of the <em>F</em> statistic. This is simply
accomplished by multiplying the <em>F</em> statistic
by three.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--sqrtF</span></tt></p>
<blockquote>
<div>Take the square root of the <span class="math">\(\chi_2^3\)</span> or <em>F</em> statistic
before outputting the stc file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--collapse</span></tt></p>
<blockquote>
<div>Make all frames in the stc file (or the wfile) identical. The value
at each source location is the maximum value of the output quantity
at this location over the analysis period. This option is convenient
for determining the correct thresholds for the rendering of the
final brain-activity movies.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--collapse1</span></tt></p>
<blockquote>
<div>Make all frames in the stc file (or the wfile) identical. The value
at each source location is the <span class="math">\(L_1\)</span> norm
of the output quantity at this location over the analysis period.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--collapse2</span></tt></p>
<blockquote>
<div>Make all frames in the stc file (or the wfile) identical. The value
at each source location is the <span class="math">\(L_2\)</span> norm
of the output quantity at this location over the analysis period.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--SIcurrents</span></tt></p>
<blockquote>
<div>Output true current values in SI units (Am). By default, the currents are
scaled so that the maximum current value is set to 50 (Am).</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--out</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the output file name. This is the &#8216;stem&#8217; of
the output file name. The actual name is derived by removing anything up
to and including the last period from the end of &lt;<em>name</em>&gt; .
According to the hemisphere, <tt class="docutils literal"><span class="pre">-lh</span></tt> or <tt class="docutils literal"><span class="pre">-rh</span></tt> is
then appended. Finally, <tt class="docutils literal"><span class="pre">.stc</span></tt> or <tt class="docutils literal"><span class="pre">.w</span></tt> is added,
depending on the output file type.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--wfiles</span></tt></p>
<blockquote>
<div>Use binary w-files in the output whenever possible. The noise-normalization
factors can be always output in this format.  The current estimates
and dSPMs can be output as wfiles if one of the collapse options
is selected.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--pred</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Save the predicted data into this file. This is a fif file containing
the predicted data waveforms, see <a class="reference internal" href="source_localization/inverse.html#chdcacdc"><em>Predicted data</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--outputnorm</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Output noise-normalization factors to this file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--invnorm</span></tt></p>
<blockquote>
<div>Output inverse noise-normalization factors to the file defined by
the <tt class="docutils literal"><span class="pre">--outputnorm</span></tt> option.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--dip</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a dipole distribution snapshot file. This is a file containing the
current distribution at a time specified with the <tt class="docutils literal"><span class="pre">--diptime</span></tt> option.
The file format is the ASCII dip file format produced by the Neuromag
source modelling software (xfit). Therefore, the file can be loaded
to the Neuromag MRIlab MRI viewer to display the actual current
distribution. This option is only effective if the <tt class="docutils literal"><span class="pre">--spm</span></tt> option
is absent.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--diptime</span> <span class="pre">&lt;*time/ms*&gt;</span></tt></p>
<blockquote>
<div>Time for the dipole snapshot, see <tt class="docutils literal"><span class="pre">--dip</span></tt> option above.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--label</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Label to process. The label files are produced by tksurfer and specify
regions of interests (ROIs). A label file name should end with <tt class="docutils literal"><span class="pre">-lh.label</span></tt> for
left-hemisphere ROIs and with <tt class="docutils literal"><span class="pre">-rh.label</span></tt> for right-hemisphere
ones. The corresponding output files are tagged with <tt class="docutils literal"><span class="pre">-lh-</span></tt> &lt;<em>data type</em> <tt class="docutils literal"><span class="pre">.amp</span></tt> and <tt class="docutils literal"><span class="pre">-rh-</span></tt> &lt;<em>data type</em> <tt class="docutils literal"><span class="pre">.amp</span></tt> , respectively. &lt;<em>data type</em>&gt; equals <tt class="docutils literal"><span class="pre">MNE</span></tt> for expected current
data and <tt class="docutils literal"><span class="pre">spm</span></tt> for dSPM data. Each line of the output
file contains the waveform of the output quantity at one of the
source locations falling inside the ROI.</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The <tt class="docutils literal"><span class="pre">--tmin</span></tt> and <tt class="docutils literal"><span class="pre">--tmax</span></tt> options    which existed in previous versions of mne_compute_mne have    been removed. mne_compute_mne can now    process only the entire averaged epoch.</p>
</div>
</div>
<div class="section" id="mne-compute-raw-inverse">
<span id="id5"></span><h3><a class="toc-backref" href="#id111">mne_compute_raw_inverse</a><a class="headerlink" href="#mne-compute-raw-inverse" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--in</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>Specifies the input data file. This can be either an evoked data
file or a raw data file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--bmin</span> <span class="pre">&lt;*time/ms*&gt;</span></tt></p>
<blockquote>
<div>Specifies the starting time of the baseline. In order to activate
baseline correction, both <tt class="docutils literal"><span class="pre">--bmin</span></tt> and <tt class="docutils literal"><span class="pre">--bmax</span></tt> options
must be present. This option applies to evoked data only.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--bmax</span> <span class="pre">&lt;*time/ms*&gt;</span></tt></p>
<blockquote>
<div>Specifies the finishing time of the baseline. This option applies
to evoked data only.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--set</span> <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>The data set (condition) number to load. This is the sequential
number of the condition. You can easily see the association by looking
at the condition list in mne_analyze when
you load the file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--inv</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Load the inverse operator decomposition from here.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--nave</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>Specifies the effective number of averaged epochs in the input data, <span class="math">\(L_{eff}\)</span>,
as discussed in <a class="reference internal" href="source_localization/inverse.html#cbbdgiae"><em>Effective number of averages</em></a>. If the input data file is
one produced by mne_browse_raw or mne_process_raw ,
the number of averages is correct in the file. However, if subtractions
or some more complicated combinations of simple averages are produced,
e.g., by  using the xplotter software,
the number of averages should be manually adjusted along the guidelines
given in <a class="reference internal" href="source_localization/inverse.html#cbbdgiae"><em>Effective number of averages</em></a>. This is accomplished either by
employing this flag or by adjusting the number of averages in the
data file with help of the utility mne_change_nave .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--snr</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>An estimate for the amplitude SNR. The regularization parameter will
be set as <span class="math">\(\lambda^2 = 1/SNR^2\)</span>. The default value is
SNR = 1. Automatic selection of the regularization parameter is
currently not supported.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--spm</span></tt></p>
<blockquote>
<div>Calculate the dSPM instead of the expected current value.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--picknormalcomp</span></tt></p>
<blockquote>
<div>The components of the estimates corresponding to directions tangential
with the cortical mantle are zeroed out.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--mricoord</span></tt></p>
<blockquote>
<div>Provide source locations and orientations in the MRI coordinate frame
instead of the default head coordinate frame.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--label</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a label file to process. For each label file, the values
of the computed estimates stored in a fif file. For more details,
see <a class="reference internal" href="source_localization/inverse.html#implementation-details"><em>Implementation details</em></a>. The label files are produced by tksurfer
or mne_analyze and specify regions
of interests (ROIs). A label file name should end with <tt class="docutils literal"><span class="pre">-lh.label</span></tt> for
left-hemisphere ROIs and with <tt class="docutils literal"><span class="pre">-rh.label</span></tt> for right-hemisphere
ones. The corresponding output files are tagged with <tt class="docutils literal"><span class="pre">-lh-</span></tt> &lt;<em>data type</em>&gt; <tt class="docutils literal"><span class="pre">.fif</span></tt> and <tt class="docutils literal"><span class="pre">-rh-</span></tt> &lt;<em>data type</em>&gt; <tt class="docutils literal"><span class="pre">.fif</span></tt> , respectively. &lt;<em>data type</em>&gt; equals <tt class="docutils literal"><span class="pre">'mne</span></tt> &#8216; for expected
current data and <tt class="docutils literal"><span class="pre">'spm</span></tt> &#8216; for dSPM data.
For raw data, <tt class="docutils literal"><span class="pre">_raw.fif</span></tt> is employed instead of <tt class="docutils literal"><span class="pre">.fif</span></tt> .
The output files are stored in the same directory as the label files.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--labelselout</span></tt></p>
<blockquote>
<div>Produces additional label files for each label processed, containing only
those vertices within the input label which correspond to available
source space vertices in the inverse operator. These files have the
same name as the original label except that <tt class="docutils literal"><span class="pre">-lh</span></tt> and <tt class="docutils literal"><span class="pre">-rh</span></tt> are replaced
by <tt class="docutils literal"><span class="pre">-sel-lh</span></tt> and <tt class="docutils literal"><span class="pre">-sel-rh</span></tt> , respectively.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--align_z</span></tt></p>
<blockquote>
<div>Instructs the program to try to align the waveform signs within
the label. For more information, see <a class="reference internal" href="source_localization/inverse.html#implementation-details"><em>Implementation details</em></a>. This
flag will not have any effect if the inverse operator has been computed
with the strict orientation constraint active.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--labeldir</span> <span class="pre">&lt;*directory*&gt;</span></tt></p>
<blockquote>
<div>All previous <tt class="docutils literal"><span class="pre">--label</span></tt> options will be ignored when this
option is encountered. For each label in the directory, the output
file defined with the <tt class="docutils literal"><span class="pre">--out</span></tt> option will contain a summarizing
waveform which is the average of the waveforms in the vertices of
the label. The <tt class="docutils literal"><span class="pre">--labeldir</span></tt> option implies <tt class="docutils literal"><span class="pre">--align_z</span></tt> and <tt class="docutils literal"><span class="pre">--picknormalcomp</span></tt> options.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--orignames</span></tt></p>
<blockquote>
<div>This option is used with the <tt class="docutils literal"><span class="pre">--labeldir</span></tt> option, above.
With this option, the output file channel names will be the names
of the label files, truncated to 15 characters, instead of names
containing the vertex numbers.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--out</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Required with <tt class="docutils literal"><span class="pre">--labeldir</span></tt> . This is the output file for
the data.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--extra</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>By default, the output includes the current estimate signals and
the digital trigger channel, see <tt class="docutils literal"><span class="pre">--digtrig</span></tt> option,
below. With the <tt class="docutils literal"><span class="pre">--extra</span></tt> option, a custom set of additional
channels can be included. The extra channel text file should contain
the names of these channels, one channel name on each line. With
this option present, the digital trigger channel is not included
unless specified in the extra channel file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--noextra</span></tt></p>
<blockquote>
<div>No additional channels will be included with this option present.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--digtrig</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Name of the composite digital trigger channel. The default value
is &#8216;STI 014&#8217;. Underscores in the channel name
will be replaced by spaces.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--split</span> <span class="pre">&lt;*size/MB*&gt;</span></tt></p>
<blockquote>
<div>Specifies the maximum size of the raw data files saved. By default, the
output is split into files which are just below 2 GB so that the
fif file maximum size is not exceed.</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The digital trigger channel can also be set with    the MNE_TRIGGER_CH_NAME environment variable. Underscores in the variable    value will <em>not</em> be replaced with spaces by mne_compute_raw_inverse .    Using the <tt class="docutils literal"><span class="pre">--digtrig</span></tt> option supersedes the MNE_TRIGGER_CH_NAME    environment variable.</p>
</div>
</div>
<div class="section" id="mne-convert-mne-data">
<span id="id6"></span><h3><a class="toc-backref" href="#id112">mne_convert_mne_data</a><a class="headerlink" href="#mne-convert-mne-data" title="Permalink to this headline">¶</a></h3>
<p>This utility allows the conversion of various fif files related to the MNE
computations to other formats. The two principal purposes of this utility are
to facilitate development of new analysis approaches with Matlab
and conversion of the forward model and noise covariance matrix
data into evoked-response type fif files, which can be accessed
and displayed with the Neuromag source modelling software.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Most of the functions of mne_convert_mne_data are    now covered by the MNE Matlab toolbox covered in <a class="reference internal" href="matlab.html#ch-matlab"><em>Matlab toolbox</em></a>.    This toolbox is recommended to avoid creating additional files occupying    disk space.</p>
</div>
<p>The command-line options recognized by mne_convert_mne_data are:</p>
<p><tt class="docutils literal"><span class="pre">--fwd</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specity the name of the forward solution file to be converted. Channels
specified with the <tt class="docutils literal"><span class="pre">--bad</span></tt> option will be excluded from
the file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fixed</span></tt></p>
<blockquote>
<div>Convert the forward solution to the fixed-orientation mode before outputting
the converted file. With this option only the field patterns corresponding
to a dipole aligned with the estimated cortex surface normal are
output.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--surfsrc</span></tt></p>
<blockquote>
<div>When outputting a free-orientation forward model (three orthogonal dipole
components present) rotate the dipole coordinate system at each
source node so that the two tangential dipole components are output
first, followed by the field corresponding to the dipole aligned
with the estimated cortex surface normal. The orientation of the
first two dipole components in the tangential plane is arbitrarily selected
to create an orthogonal coordinate system.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--noiseonly</span></tt></p>
<blockquote>
<div>When creating a &#8216;measurement&#8217; fif file, do not
output a forward model file, just the noise-covariance matrix.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--senscov</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the fif file containing a sensor covariance matrix to
be included with the output. If no other input files are specified
only the covariance matrix is output</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--srccov</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the fif file containing the source covariance matrix to
be included with the output. Only diagonal source covariance files
can be handled at the moment.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--bad</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of the file containing the names of the channels to
be omitted, one channel name per line. This does not affect the output
of the inverse operator since the channels have been already selected
when the file was created.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fif</span></tt></p>
<blockquote>
<div>Output the forward model and the noise-covariance matrix into &#8216;measurement&#8217; fif
files. The forward model files are tagged with &lt;<em>modalities</em>&gt; <tt class="docutils literal"><span class="pre">-meas-fwd.fif</span></tt> and
the noise-covariance matrix files with &lt;<em>modalities</em>&gt; <tt class="docutils literal"><span class="pre">-meas-cov.fif</span></tt> .
Here, modalities is <tt class="docutils literal"><span class="pre">-meg</span></tt> if MEG is included, <tt class="docutils literal"><span class="pre">-eeg</span></tt> if
EEG is included, and <tt class="docutils literal"><span class="pre">-meg-eeg</span></tt> if both types of signals
are present. The inclusion of modalities is controlled by the <tt class="docutils literal"><span class="pre">--meg</span></tt> and <tt class="docutils literal"><span class="pre">--eeg</span></tt> options.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--mat</span></tt></p>
<blockquote>
<div>Output the data into MATLAB mat files. This is the default. The
forward model files are tagged with &lt;<em>modalities</em>&gt; <tt class="docutils literal"><span class="pre">-fwd.mat</span></tt> forward model
and noise-covariance matrix output, with <tt class="docutils literal"><span class="pre">-inv.mat</span></tt> for inverse
operator output, and with <tt class="docutils literal"><span class="pre">-inv-meas.mat</span></tt> for combined inverse
operator and measurement data output, respectively. The meaning
of &lt;<em>modalities</em>&gt; is the same
as in the fif output, described above.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--tag</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>By default, all variables in the matlab output files start with
<tt class="docutils literal"><span class="pre">mne\_</span></tt>. This option allows to change this prefix to &lt;<em>name</em>&gt; _.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--meg</span></tt></p>
<blockquote>
<div>Include MEG channels from the forward solution and noise-covariance
matrix.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eeg</span></tt></p>
<blockquote>
<div>Include EEG channels from the forward solution and noise-covariance
matrix.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--inv</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Output the inverse operator data from the specified file into a
mat file. The source and noise covariance matrices as well as active channels
have been previously selected when the inverse operator was created
with mne_inverse_operator . Thus
the options <tt class="docutils literal"><span class="pre">--meg</span></tt> , <tt class="docutils literal"><span class="pre">--eeg</span></tt> , <tt class="docutils literal"><span class="pre">--senscov</span></tt> , <tt class="docutils literal"><span class="pre">--srccov</span></tt> , <tt class="docutils literal"><span class="pre">--noiseonly</span></tt> ,
and <tt class="docutils literal"><span class="pre">--bad</span></tt> do not affect the output of the inverse operator.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--meas</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the file containing measurement data to be output together with
the inverse operator. The channels corresponding to the inverse operator
are automatically selected from the file if <tt class="docutils literal"><span class="pre">--inv</span></tt> .
option is present. Otherwise, the channel selection given with <tt class="docutils literal"><span class="pre">--sel</span></tt> option will
be taken into account.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--set</span> <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>Select the data set to be output from the measurement file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--bmin</span> <span class="pre">&lt;*value/ms*&gt;</span></tt></p>
<blockquote>
<div>Specifies the baseline minimum value setting for the measurement signal
output.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--bmax</span> <span class="pre">&lt;*value/ms*&gt;</span></tt></p>
<blockquote>
<div>Specifies the baseline maximum value setting for the measurement signal
output.</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The <tt class="docutils literal"><span class="pre">--tmin</span></tt> and <tt class="docutils literal"><span class="pre">--tmax</span></tt> options    which existed in previous versions of mne_converted_mne_data have    been removed. If output of measurement data is requested, the entire    averaged epoch is now included.</p>
</div>
<div class="section" id="guide-to-combining-options">
<h4>Guide to combining options<a class="headerlink" href="#guide-to-combining-options" title="Permalink to this headline">¶</a></h4>
<p>The combination of options is quite complicated. The <a class="reference internal" href="#behdciii"><em>Guide to combining mne_convert_mne_data options.</em></a> should be
helpful to determine the combination of options appropriate for your needs.</p>
<table border="1" class="docutils" id="behdciii">
<caption>Guide to combining mne_convert_mne_data options.</caption>
<colgroup>
<col width="39%" />
<col width="9%" />
<col width="27%" />
<col width="24%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td>Desired output</td>
<td>Format</td>
<td>Required options</td>
<td>Optional options</td>
</tr>
<tr class="row-even"><td>forward model</td>
<td>fif</td>
<td>&#8212;fwd &lt;<em>name</em>&gt;
&#8212;out &lt;<em>name</em>&gt;
&#8212;meg and/or &#8212;eeg
&#8212;fif</td>
<td>&#8212;bad &lt;<em>name</em>&gt;
&#8212;surfsrc</td>
</tr>
<tr class="row-odd"><td>forward model</td>
<td>mat</td>
<td>&#8212;fwd &lt;<em>name</em>&gt;
&#8212;out &lt;<em>name</em>&gt;
&#8212;meg and/or &#8211;eeg</td>
<td>&#8212;bad &lt;<em>name</em>&gt;
&#8212;surfsrc</td>
</tr>
<tr class="row-even"><td>forward model and sensor covariance</td>
<td>mat</td>
<td>&#8212;fwd &lt;<em>name</em>&gt;
&#8212;out &lt;<em>name</em>&gt;
&#8212;senscov &lt;<em>name</em>&gt;
&#8212;meg and/or &#8211;eeg</td>
<td>&#8212;bad &lt;<em>name</em>&gt;
&#8212;surfsrc</td>
</tr>
<tr class="row-odd"><td>sensor covariance</td>
<td>fif</td>
<td>&#8212;fwd &lt;<em>name</em>&gt;
&#8212;out &lt;<em>name</em>&gt;
&#8212;senscov &lt;<em>name</em>&gt;
&#8212;noiseonly
&#8212;fif
&#8212;meg and/or &#8211;eeg</td>
<td>&#8212;bad &lt;<em>name</em>&gt;</td>
</tr>
<tr class="row-even"><td>sensor covariance</td>
<td>mat</td>
<td>&#8212;senscov &lt;<em>name</em>&gt;
&#8212;out &lt;<em>name</em>&gt;</td>
<td>&#8212;bad &lt;<em>name</em>&gt;</td>
</tr>
<tr class="row-odd"><td>sensor covariance eigenvalues</td>
<td>text</td>
<td>&#8212;senscov &lt;<em>name</em>&gt;
&#8212;out &lt;<em>name</em>&gt;
&#8212;eig</td>
<td>&#8212;bad &lt;<em>name</em>&gt;</td>
</tr>
<tr class="row-even"><td>evoked MEG/EEG data</td>
<td>mat</td>
<td>&#8212;meas &lt;<em>name</em>&gt;
&#8212;out &lt;<em>name</em>&gt;</td>
<td>&#8212;sel &lt;<em>name</em>&gt;
&#8212;set &lt;<em>number</em>&gt;</td>
</tr>
<tr class="row-odd"><td>evoked MEG/EEG data forward model</td>
<td>mat</td>
<td>&#8212;meas &lt;<em>name</em>&gt;
&#8212;fwd &lt;<em>name</em>&gt;
&#8212;out &lt;<em>name</em>&gt;</td>
<td>&#8212;bad &lt;<em>name</em>&gt;
&#8212;set &lt;<em>number</em>&gt;</td>
</tr>
<tr class="row-even"><td>inverse operator data</td>
<td>mat</td>
<td>&#8212;inv &lt;<em>name</em>&gt;
&#8212;out &lt;<em>name</em>&gt;</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td>inverse operator data evoked
MEG/EEG data</td>
<td>mat</td>
<td>–&#8211;inv &lt;<em>name</em>&gt;
–&#8211;meas &lt;<em>name</em>&gt;
–&#8211;out &lt;<em>name</em>&gt;</td>
<td>&nbsp;</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="matlab-data-structures">
<h4>Matlab data structures<a class="headerlink" href="#matlab-data-structures" title="Permalink to this headline">¶</a></h4>
<p>The Matlab output provided by mne_convert_mne_data is
organized in structures, listed in <a class="reference internal" href="#behcicca"><em>Matlab structures produced by mne_convert_mne_data.</em></a>. The fields
occurring in these structures are listed in <a class="reference internal" href="#babcbigf"><em>The fields of Matlab structures.</em></a>.</p>
<p>The symbols employed in variable size descriptions are:</p>
<p><tt class="docutils literal"><span class="pre">nloc</span></tt></p>
<blockquote>
<div>Number
of source locations</div></blockquote>
<p><tt class="docutils literal"><span class="pre">nsource</span></tt></p>
<blockquote>
<div>Number
of sources. For fixed orientation sources nsource = nloc whereas nsource = 3*nloc for
free orientation sources</div></blockquote>
<p><tt class="docutils literal"><span class="pre">nchan</span></tt></p>
<blockquote>
<div>Number
of measurement channels.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">ntime</span></tt></p>
<blockquote>
<div>Number
of time points in the measurement data.</div></blockquote>
<table border="1" class="docutils" id="behcicca">
<caption>Matlab structures produced by mne_convert_mne_data.</caption>
<colgroup>
<col width="28%" />
<col width="72%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Structure</th>
<th class="head">Contents</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>&lt;<em>tag</em>&gt; _meas</td>
<td>Measured data</td>
</tr>
<tr class="row-odd"><td>&lt;<em>tag</em>&gt; _inv</td>
<td>The inverse operator decomposition</td>
</tr>
<tr class="row-even"><td>&lt;<em>tag</em>&gt; _fwd</td>
<td>The forward solution</td>
</tr>
<tr class="row-odd"><td>&lt;<em>tag</em>&gt; _noise</td>
<td>A standalone noise-covariance matrix</td>
</tr>
</tbody>
</table>
<p>The prefix given with the <tt class="docutils literal"><span class="pre">--tag</span></tt> option is indicated &lt;<em>tag</em>&gt; , see <a class="reference internal" href="#mne-convert-mne-data"><em>mne_convert_mne_data</em></a>. Its default value is MNE.</p>
<table border="1" class="docutils" id="babcbigf">
<caption>The fields of Matlab structures.</caption>
<colgroup>
<col width="23%" />
<col width="17%" />
<col width="60%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td>Variable</td>
<td>Size</td>
<td>Description</td>
</tr>
<tr class="row-even"><td>fwd</td>
<td>nsource x nchan</td>
<td>The forward solution, one source on each row. For free
orientation sources, the fields of the three orthogonal
dipoles for each location are listed consecutively.</td>
</tr>
<tr class="row-odd"><td>names ch_names</td>
<td>nchan (string)</td>
<td>String array containing the names of the channels included</td>
</tr>
<tr class="row-even"><td>ch_types</td>
<td>nchan x 2</td>
<td>The column lists the types of the channels (1 = MEG,
2 = EEG). The second column lists the coil types, see
<a class="reference internal" href="source_localization/forward.html#bgbbhgec"><em>Normal coil descriptions.</em></a> and <a class="reference internal" href="source_localization/forward.html#chdbdfje"><em>Accurate coil descriptions</em></a>. For EEG electrodes,
this value equals one.</td>
</tr>
<tr class="row-odd"><td>ch_pos</td>
<td>nchan x 3</td>
<td>The location information for each channel. The first three
values specify the origin of the sensor coordinate system
or the location of the electrode. For MEG channels, the
following nine number specify the <em>x</em>, <em>y</em>, and
<em>z</em>-direction unit vectors of the sensor coordinate system.
For EEG electrodes the first unit vector specifies the
location of the reference electrode. If the reference is
not specified this value is all zeroes. The remaining unit
vectors are irrelevant for EEG electrodes.</td>
</tr>
<tr class="row-even"><td>ch_lognos</td>
<td>nchan x 1</td>
<td>Logical channel numbers as listed in the fiff file</td>
</tr>
<tr class="row-odd"><td>ch_units</td>
<td>nchan x 2</td>
<td>Units and unit multipliers as listed in the fif file. The
unit of the data is listed in the first column (T = 112,
T/m = 201, V = 107). At present, the second column will be
always zero, <em>i.e.</em>, no unit multiplier.</td>
</tr>
<tr class="row-even"><td>ch_cals</td>
<td>nchan x 2</td>
<td>Even if the data comes from the conversion already
calibrated, the original calibration factors are included.
The first column is the range member of the fif data
structures and while the second is the cal member. To get
calibrated values in the units given in ch_units from the
raw data, the data must be multiplied with the product of
range and cal.</td>
</tr>
<tr class="row-odd"><td>sfreq</td>
<td>1</td>
<td>The sampling frequency in Hz.</td>
</tr>
<tr class="row-even"><td>lowpass</td>
<td>1</td>
<td>Lowpass filter frequency (Hz)</td>
</tr>
<tr class="row-odd"><td>highpass</td>
<td>1</td>
<td>Highpass filter frequency (Hz)</td>
</tr>
<tr class="row-even"><td>source_loc</td>
<td>nloc x 3</td>
<td>The source locations given in the coordinate frame
indicated by the coord_frame member.</td>
</tr>
<tr class="row-odd"><td>source_ori</td>
<td>nsource x 3</td>
<td>The source orientations</td>
</tr>
<tr class="row-even"><td>source_selection</td>
<td>nsource x 2</td>
<td>Indication of the sources selected from the complete source
spaces. Each row contains the number of the source in the
complete source space (starting with 0) and the source
space number (1 or 2). These numbers refer to the order the
two hemispheres where listed when mne_make_source_space was
invoked. mne_setup_source_space lists the left hemisphere
first.</td>
</tr>
<tr class="row-odd"><td>coord_frame</td>
<td>string</td>
<td>Name of the coordinate frame employed in the forward
calculations. Possible values are &#8216;head&#8217; and &#8216;mri&#8217;.</td>
</tr>
<tr class="row-even"><td>mri_head_trans</td>
<td>4 x 4</td>
<td>The coordinate frame transformation from mri the MEG &#8216;head&#8217;
coordinates.</td>
</tr>
<tr class="row-odd"><td>meg_head_trans</td>
<td>4 x 4</td>
<td>The coordinate frame transformation from the MEG device
coordinates to the MEG head coordinates</td>
</tr>
<tr class="row-even"><td>noise_cov</td>
<td>nchan x nchan</td>
<td>The noise covariance matrix</td>
</tr>
<tr class="row-odd"><td>source_cov</td>
<td>nsource</td>
<td>The elements of the diagonal source covariance matrix.</td>
</tr>
<tr class="row-even"><td>sing</td>
<td>nchan</td>
<td>The singular values of
<span class="math">\(A = C_0^{-^1/_2} G R^C = U \Lambda V^T\)</span>
with <span class="math">\(R\)</span> selected so that
<span class="math">\(\text{trace}(AA^T) / \text{trace}(I) = 1\)</span>
as discussed in <a class="reference internal" href="source_localization/inverse.html#chddhage"><em>Whitening and scaling</em></a></td>
</tr>
<tr class="row-odd"><td>eigen_fields</td>
<td>nchan x nchan</td>
<td>The rows of this matrix are the left singular vectors of
<span class="math">\(A\)</span>, i.e., the columns of <span class="math">\(U\)</span>, see above.</td>
</tr>
<tr class="row-even"><td>eigen_leads</td>
<td>nchan x nsource</td>
<td>The rows of this matrix are the right singular vectors of
<span class="math">\(A\)</span>, i.e., the columns of <span class="math">\(V\)</span>, see above.</td>
</tr>
<tr class="row-odd"><td>noise_eigenval</td>
<td>nchan</td>
<td>In terms of <a class="reference internal" href="source_localization/inverse.html#chddhage"><em>Whitening and scaling</em></a>, eigenvalues of <span class="math">\(C_0\)</span>,
i.e., not scaled with number of averages.</td>
</tr>
<tr class="row-even"><td>noise_eigenvec</td>
<td>nchan</td>
<td>Eigenvectors of the noise covariance matrix. In terms of
<a class="reference internal" href="source_localization/inverse.html#chddhage"><em>Whitening and scaling</em></a>, <span class="math">\(U_C^T\)</span>.</td>
</tr>
<tr class="row-odd"><td>data</td>
<td>nchan x ntime</td>
<td>The measured data. One row contains the data at one time
point.</td>
</tr>
<tr class="row-even"><td>times</td>
<td>ntime</td>
<td>The time points in the above matrix in seconds</td>
</tr>
<tr class="row-odd"><td>nave</td>
<td>1</td>
<td>Number of averages as listed in the data file.</td>
</tr>
<tr class="row-even"><td>meas_times</td>
<td>ntime</td>
<td>The time points in seconds.</td>
</tr>
</tbody>
</table>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The Matlab files can also be read in Python using <tt class="xref py py-func docutils literal"><span class="pre">scipy.io.loadmat()</span></tt></p>
</div>
</div>
</div>
<div class="section" id="mne-do-forward-solution">
<span id="id7"></span><h3><a class="toc-backref" href="#id113">mne_do_forward_solution</a><a class="headerlink" href="#mne-do-forward-solution" title="Permalink to this headline">¶</a></h3>
<p>This utility accepts the following options:</p>
<p><tt class="docutils literal"><span class="pre">--subject</span> <span class="pre">&lt;*subject*&gt;</span></tt></p>
<blockquote>
<div>Defines the name of the subject. This can be also accomplished
by setting the SUBJECT environment variable.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--src</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Source space name to use. This option overrides the <tt class="docutils literal"><span class="pre">--spacing</span></tt> option. The
source space is searched first from the current working directory
and then from <tt class="docutils literal"><span class="pre">$SUBJECTS_DIR/</span></tt> &lt;<em>subject</em>&gt; /bem.
The source space file must be specified exactly, including the <tt class="docutils literal"><span class="pre">fif</span></tt> extension.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--spacing</span> <span class="pre">&lt;*spacing/mm*&gt;</span>&nbsp; <span class="pre">or</span> <span class="pre">``ico-</span></tt> &lt;<em>number  or ``oct-`` &lt;*number</em>&gt;``</p>
<blockquote>
<div>This is an alternate way to specify the name of the source space
file. For example, if <tt class="docutils literal"><span class="pre">--spacing</span> <span class="pre">6</span></tt> is given on the command
line, the source space files searched for are./&lt;<em>subject</em>&gt; -6-src.fif
and <tt class="docutils literal"><span class="pre">$SUBJECTS_DIR/$SUBJECT/</span></tt> bem/&lt;<em>subject</em>&gt; -6-src.fif.
The first file found is used. Spacing defaults to 7 mm.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--bem</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the BEM to be used. The name of the file can be any of &lt;<em>name</em>&gt; , &lt;<em>name</em>&gt; -bem.fif, &lt;<em>name</em>&gt; -bem-sol.fif.
The file is searched for from the current working directory and
from <tt class="docutils literal"><span class="pre">bem</span></tt> . If this option is omitted, the most recent
BEM file in the <tt class="docutils literal"><span class="pre">bem</span></tt> directory is used.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--mri</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>The name of the MRI description file containing the MEG/MRI coordinate
transformation. This file was saved as part of the alignment procedure
outlined in <a class="reference internal" href="cookbook.html#chdbehdc"><em>Aligning coordinate frames</em></a>. The file is searched for from
the current working directory and from <tt class="docutils literal"><span class="pre">mri/T1-neuromag/sets</span></tt> .
The search order for MEG/MRI coordinate transformations is discussed
below.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--trans</span>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>The name of a text file containing the 4 x 4 matrix for the coordinate transformation
from head to mri coordinates, see below. If the option <tt class="docutils literal"><span class="pre">--trans</span></tt> is
present, the <tt class="docutils literal"><span class="pre">--mri</span></tt> option is not required. The search
order for MEG/MRI coordinate transformations is discussed below.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--meas</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>This file is the measurement fif file or an off-line average file
produced thereof. It is recommended that the average file is employed for
evoked-response data and the original raw data file otherwise. This
file provides the MEG sensor locations and orientations as well as
EEG electrode locations as well as the coordinate transformation between
the MEG device coordinates and MEG head-based coordinates.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fwd</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>This file will contain the forward solution as well as the coordinate transformations,
sensor and electrode location information, and the source space
data. A name of the form &lt;<em>name</em>&gt; <tt class="docutils literal"><span class="pre">-fwd.fif</span></tt> is
recommended. If this option is omitted the forward solution file
name is automatically created from the measurement file name and
the source space name.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--destdir</span> <span class="pre">&lt;*directory*&gt;</span></tt></p>
<blockquote>
<div>Optionally specifies a directory where the forward solution will
be stored.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--mindist</span> <span class="pre">&lt;*dist/mm*&gt;</span></tt></p>
<blockquote>
<div>Omit source space points closer than this value to the inner skull surface.
Any source space points outside the inner skull surface are automatically
omitted. The use of this option ensures that numerical inaccuracies
for very superficial sources do not cause unexpected effects in
the final current estimates. Suitable value for this parameter is
of the order of the size of the triangles on the inner skull surface.
If you employ the seglab software
to create the triangulations, this value should be about equal to
the wish for the side length of the triangles.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--megonly</span></tt></p>
<blockquote>
<div>Omit EEG forward calculations.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eegonly</span></tt></p>
<blockquote>
<div>Omit MEG forward calculations.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--all</span></tt></p>
<blockquote>
<div>Compute the forward solution for all vertices on the source space.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--overwrite</span></tt></p>
<blockquote>
<div>Overwrite the possibly existing forward model file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--help</span></tt></p>
<blockquote>
<div>Show usage information for the script.</div></blockquote>
<p>The MEG/MRI transformation is determined by the following
search sequence:</p>
<ul class="simple">
<li>If the <tt class="docutils literal"><span class="pre">--mri</span></tt> option was
present, the file is looked for literally as specified, in the directory
of the measurement file specified with the <tt class="docutils literal"><span class="pre">--meas</span></tt> option,
and in the directory $SUBJECTS_DIR/$SUBJECT/mri/T1-neuromag/sets.
If the file is not found, the script exits with an error message.</li>
<li>If the <tt class="docutils literal"><span class="pre">--trans</span></tt> option was present, the file is
looked up literally as specified. If the file is not found, the
script exists with an error message.</li>
<li>If neither <tt class="docutils literal"><span class="pre">--mri</span></tt> nor <tt class="docutils literal"><span class="pre">--trans</span></tt> option
was not present, the following default search sequence is engaged:<ul>
<li>The <tt class="docutils literal"><span class="pre">.fif</span></tt> ending in the
measurement file name is replaced by <tt class="docutils literal"><span class="pre">-trans.fif</span></tt> . If
this file is present, it will be used.</li>
<li>The newest file whose name ends with <tt class="docutils literal"><span class="pre">-trans.fif</span></tt> in
the directory of the measurement file is looked up. If such a file
is present, it will be used.</li>
<li>The newest file whose name starts with <tt class="docutils literal"><span class="pre">COR-</span></tt> in
directory $SUBJECTS_DIR/$SUBJECT/mri/T1-neuromag/sets is looked
up. If such a file is present, it will be used.</li>
<li>If all the above searches fail, the script exits with an error
message.</li>
</ul>
</li>
</ul>
<p>This search sequence is designed to work well with the MEG/MRI
transformation files output by mne_analyze ,
see <a class="reference internal" href="gui/analyze.html#cacehgcd"><em>Coordinate frame alignment</em></a>. It is recommended that -trans.fif file
saved with the Save default and Save... options in
the mne_analyze alignment dialog
are used because then the $SUBJECTS_DIR/$SUBJECT directory will
be composed of files which are dependent on the subjects&#8217;s
anatomy only, not on the MEG/EEG data to be analyzed.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">If the standard MRI description file and BEM    file selections are appropriate and the 7-mm source space grid spacing    is appropriate, only the <tt class="docutils literal"><span class="pre">--meas</span></tt> option is necessary.    If EEG data is not used <tt class="docutils literal"><span class="pre">--megonly</span></tt> option should be    included.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">If it is conceivable that the current-density    transformation will be incorporated into the inverse operator, specify    a source space with patch information for the forward computation.    This is not mandatory but saves a lot of time when the inverse operator    is created, since the patch information does not need to be created    at that stage.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The MEG head to MRI transformation matrix specified    with the <tt class="docutils literal"><span class="pre">--trans</span></tt> option should be a text file containing    a 4-by-4 matrix:</p>
</div>
<div class="math">
\[\begin{split}T = \begin{bmatrix}
R_{11} &amp; R_{12} &amp; R_{13} &amp; x_0 \\
R_{13} &amp; R_{13} &amp; R_{13} &amp; y_0 \\
R_{13} &amp; R_{13} &amp; R_{13} &amp; z_0 \\
0 &amp; 0 &amp; 0 &amp; 1
\end{bmatrix}\end{split}\]</div>
<p>defined so that if the augmented location vectors in MRI
head and MRI coordinate systems are denoted by <span class="math">\(r_{head}[x_{head}\ y_{head}\ z_{head}\ 1]\)</span> and <span class="math">\(r_{MRI}[x_{MRI}\ y_{MRI}\ z_{MRI}\ 1]\)</span>,
respectively,</p>
<div class="math">
\[r_{MRI} = T r_{head}\]</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">It is not possible to calculate an EEG forward    solution with a single-layer BEM.</p>
</div>
</div>
<div class="section" id="mne-do-inverse-operator">
<span id="id8"></span><h3><a class="toc-backref" href="#id114">mne_do_inverse_operator</a><a class="headerlink" href="#mne-do-inverse-operator" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--fwd</span> <span class="pre">&lt;*name</span> <span class="pre">of</span> <span class="pre">the</span> <span class="pre">forward</span> <span class="pre">solution</span> <span class="pre">file*&gt;</span></tt></p>
<blockquote>
<div>This is the forward solution file produced in the computations step described
in <a class="reference internal" href="cookbook.html#babchejd"><em>Computing the forward solution</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--meg</span></tt></p>
<blockquote>
<div>Employ MEG data in the inverse calculation. If neither <tt class="docutils literal"><span class="pre">--meg</span></tt> nor <tt class="docutils literal"><span class="pre">--eeg</span></tt> is
set only MEG channels are included.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eeg</span></tt></p>
<blockquote>
<div>Employ EEG data in the inverse calculation. If neither <tt class="docutils literal"><span class="pre">--meg</span></tt> nor <tt class="docutils literal"><span class="pre">--eeg</span></tt> is
set only MEG channels are included.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fixed</span></tt></p>
<blockquote>
<div>Use fixed source orientations normal to the cortical mantle. By default,
the source orientations are not constrained. If <tt class="docutils literal"><span class="pre">--fixed</span></tt> is specified,
the <tt class="docutils literal"><span class="pre">--loose</span></tt> flag is ignored.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--loose</span> <span class="pre">&lt;*amount*&gt;</span></tt></p>
<blockquote>
<div>Use a &#8216;loose&#8217; orientation constraint. This means
that the source covariance matrix entries corresponding to the current
component normal to the cortex are set equal to one and the transverse
components are set to &lt;<em>amount</em>&gt; .
Recommended value of amount is 0.1...0.6.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--depth</span></tt></p>
<blockquote>
<div>Employ depth weighting with the standard settings. For details,
see <a class="reference internal" href="source_localization/inverse.html#depth-weighting"><em>Depth weighting</em></a> and <a class="reference internal" href="source_localization/inverse.html#inverse-operator"><em>Inverse-operator decomposition</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--bad</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a text file to designate bad channels, listed one channel name
(like MEG 1933) on each line of the file. Be sure to include both
noisy and flat (non-functioning) channels in the list. If bad channels
were designated using mne_mark_bad_channels in
the measurement file which was specified with the <tt class="docutils literal"><span class="pre">--meas</span></tt> option when
the forward solution was computed, the bad channel information will
be automatically included. Also, any bad channel information in
the noise-covariance matrix file will be included.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--noisecov</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Name of the noise-covariance matrix file computed with one of the methods
described in <a class="reference internal" href="cookbook.html#babdeeeb"><em>Computing the noise-covariance matrix</em></a>. By default, the script looks
for a file whose name is derived from the forward solution file
by replacing its ending <tt class="docutils literal"><span class="pre">-</span></tt> &lt;<em>anything</em>&gt; <tt class="docutils literal"><span class="pre">-fwd.fif</span></tt> by <tt class="docutils literal"><span class="pre">-cov.fif</span></tt> .
If this file contains a projection operator, which will automatically
attached to the noise-covariance matrix by mne_browse_raw and mne_process_raw ,
no <tt class="docutils literal"><span class="pre">--proj</span></tt> option is necessary because mne_inverse_operator will
automatically include the projectors from the noise-covariance matrix
file. For backward compatibility, &#8211;senscov can be used as a synonym
for &#8211;noisecov.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--noiserank</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>Specifies the rank of the noise covariance matrix explicitly rather than
trying to reduce it automatically. This option is sheldom needed,</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--megreg</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>Regularize the MEG part of the noise-covariance matrix by this amount.
Suitable values are in the range 0.05...0.2. For details, see <a class="reference internal" href="source_localization/inverse.html#cov-regularization"><em>Regularization of the noise-covariance matrix</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eegreg</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>Like <tt class="docutils literal"><span class="pre">--megreg</span></tt> but applies to the EEG channels.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--diagnoise</span></tt></p>
<blockquote>
<div>Omit the off-diagonal terms of the noise covariance matrix. This option
is irrelevant to most users.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fmri</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>With help of this w file, an <em>a priori</em> weighting
can be applied to the source covariance matrix. The source of the weighting
is usually fMRI but may be also some other data, provided that the weighting can
be expressed as a scalar value on the cortical surface, stored in
a w file. It is recommended that this w file is appropriately smoothed (see <a class="reference internal" href="source_localization/morph.html#chdebahh"><em>About smoothing</em></a>)
in mne_analyze , tksurfer or
with mne_smooth_w to contain
nonzero values at all vertices of the triangular tessellation of
the cortical surface. The name of the file given is used as a stem of
the w files. The actual files should be called &lt;<em>name</em>&gt; <tt class="docutils literal"><span class="pre">-lh.pri</span></tt> and &lt;<em>name</em>&gt; <tt class="docutils literal"><span class="pre">-rh.pri</span></tt> for
the left and right hemisphere weight files, respectively. The application
of the weighting is discussed in <a class="reference internal" href="source_localization/inverse.html#mne-fmri-estimates"><em>fMRI-guided estimates</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fmrithresh</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>This option is mandatory and has an effect only if a weighting function
has been specified with the <tt class="docutils literal"><span class="pre">--fmri</span></tt> option. If the value
is in the <em>a priori</em> files falls below this value
at a particular source space point, the source covariance matrix
values are multiplied by the value specified with the <tt class="docutils literal"><span class="pre">--fmrioff</span></tt> option
(default 0.1). Otherwise it is left unchanged.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fmrioff</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>The value by which the source covariance elements are multiplied
if the <em>a priori</em> weight falls below the threshold
set with <tt class="docutils literal"><span class="pre">--fmrithresh</span></tt> , see above.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--srccov</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Use this diagonal source covariance matrix. By default the source covariance
matrix is a multiple of the identity matrix. This option is irrelevant
to most users.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--proj</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Include signal-space projection information from this file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--inv</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Save the inverse operator decomposition here. By default, the script looks
for a file whose name is derived from the forward solution file by
replacing its ending <tt class="docutils literal"><span class="pre">-fwd.fif</span></tt> by &lt;<em>options</em>&gt; <tt class="docutils literal"><span class="pre">-inv.fif</span></tt> , where
&lt;<em>options</em>&gt; includes options <tt class="docutils literal"><span class="pre">--meg</span></tt>, <tt class="docutils literal"><span class="pre">--eeg</span></tt>, and <tt class="docutils literal"><span class="pre">--fixed</span></tt> with the double
dashes replaced by single ones.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--destdir</span> <span class="pre">&lt;*directory*&gt;</span></tt></p>
<blockquote>
<div>Optionally specifies a directory where the inverse operator will
be stored.</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">If bad channels are included in the calculation,    strange results may ensue. Therefore, it is recommended that the    data to be analyzed is carefully inspected with to assign the bad    channels correctly.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">For convenience, the MNE software includes bad-channel    designation files which can be used to ignore all magnetometer or    all gradiometer channels in Vectorview measurements. These files are    called <tt class="docutils literal"><span class="pre">vv_grad_only.bad</span></tt> and <tt class="docutils literal"><span class="pre">vv_mag_only.bad</span></tt> , respectively.    Both files are located in <tt class="docutils literal"><span class="pre">$MNE_ROOT/share/mne/templates</span></tt> .</p>
</div>
</div>
<div class="section" id="mne-forward-solution">
<span id="id9"></span><h3><a class="toc-backref" href="#id115">mne_forward_solution</a><a class="headerlink" href="#mne-forward-solution" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--src</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Source space name to use. The name of the file must be specified exactly,
including the directory. Typically, the source space files reside
in $SUBJECTS_DIR/$SUBJECT/bem.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--bem</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the BEM to be used. These files end with bem.fif or bem-sol.fif and
reside in $SUBJECTS_DIR/$SUBJECT/bem. The former file contains only
the BEM surface information while the latter files contain the geometry
information precomputed with <a class="reference internal" href="#mne-prepare-bem-model"><em>mne_prepare_bem_model</em></a>,
see <a class="reference internal" href="source_localization/forward.html#chdjfheb"><em>Computing the BEM geometry data</em></a>. If precomputed geometry is not available,
the linear collocation solution will be computed by mne_forward_solution .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--origin</span> <span class="pre">&lt;*x/mm*&gt;</span> <span class="pre">:</span> <span class="pre">&lt;*x/mm*&gt;</span> <span class="pre">:</span> <span class="pre">&lt;*z/mm*&gt;</span></tt></p>
<blockquote>
<div>Indicates that the sphere model should be used in the forward calculations.
The origin is specified in MEG head coordinates unless the <tt class="docutils literal"><span class="pre">--mricoord</span></tt> option
is present. The MEG sphere model solution computed using the analytical
Sarvas formula. For EEG, an approximative solution described in</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eegmodels</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>This option is significant only if the sphere model is used and
EEG channels are present. The specified file contains specifications
of the EEG sphere model layer structures as detailed in <a class="reference internal" href="source_localization/forward.html#chdiafig"><em>The EEG sphere model definition file</em></a>. If this option is absent the file <tt class="docutils literal"><span class="pre">$HOME/.mne/EEG_models</span></tt> will
be consulted if it exists.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eegmodel</span> <span class="pre">&lt;*model</span> <span class="pre">name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of the sphere model to be used for EEG. If this option
is missing, the model Default will
be employed, see <a class="reference internal" href="source_localization/forward.html#chdiafig"><em>The EEG sphere model definition file</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eegrad</span> <span class="pre">&lt;*radius/mm*&gt;</span></tt></p>
<blockquote>
<div>Specifies the radius of the outermost surface (scalp) of the EEG sphere
model, see <a class="reference internal" href="source_localization/forward.html#chdiafig"><em>The EEG sphere model definition file</em></a>. The default value is 90 mm.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eegscalp</span></tt></p>
<blockquote>
<div>Scale the EEG electrode locations to the surface of the outermost sphere
when using the sphere model.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--accurate</span></tt></p>
<blockquote>
<div>Use accurate MEG sensor coil descriptions. This is the recommended
choice. More information</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fixed</span></tt></p>
<blockquote>
<div>Compute the solution for sources normal to the cortical mantle only. This
option should be used only for surface-based and discrete source
spaces.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--all</span></tt></p>
<blockquote>
<div>Compute the forward solution for all vertices on the source space.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--label</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Compute the solution only for points within the specified label. Multiple
labels can be present. The label files should end with <tt class="docutils literal"><span class="pre">-lh.label</span></tt> or <tt class="docutils literal"><span class="pre">-rh.label</span></tt> for
left and right hemisphere label files, respectively. If <tt class="docutils literal"><span class="pre">--all</span></tt> flag
is present, all surface points falling within the labels are included.
Otherwise, only decimated points with in the label are selected.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--mindist</span> <span class="pre">&lt;*dist/mm*&gt;</span></tt></p>
<blockquote>
<div>Omit source space points closer than this value to the inner skull surface.
Any source space points outside the inner skull surface are automatically
omitted. The use of this option ensures that numerical inaccuracies
for very superficial sources do not cause unexpected effects in
the final current estimates. Suitable value for this parameter is
of the order of the size of the triangles on the inner skull surface.
If you employ the seglab software to create the triangulations, this
value should be about equal to the wish for the side length of the
triangles.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--mindistout</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a file name to contain the coordinates of source space points
omitted due to the <tt class="docutils literal"><span class="pre">--mindist</span></tt> option.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--mri</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>The name of the MRI description file containing the MEG/MRI coordinate
transformation. This file was saved as part of the alignment procedure
outlined in <a class="reference internal" href="cookbook.html#chdbehdc"><em>Aligning coordinate frames</em></a>. These files typically reside in <tt class="docutils literal"><span class="pre">$SUBJECTS_DIR/$SUBJECT/mri/T1-neuromag/sets</span></tt> .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--trans</span>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>The name of a text file containing the 4 x 4 matrix for the coordinate transformation
from head to mri coordinates. With <tt class="docutils literal"><span class="pre">--trans</span></tt>, <tt class="docutils literal"><span class="pre">--mri</span></tt> option is not
required.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--notrans</span></tt></p>
<blockquote>
<div>The MEG/MRI coordinate transformation is taken as the identity transformation, <em>i.e.</em>,
the two coordinate systems are the same. This option is useful only
in special circumstances. If more than one of the <tt class="docutils literal"><span class="pre">--mri</span></tt> , <tt class="docutils literal"><span class="pre">--trans</span></tt> ,
and <tt class="docutils literal"><span class="pre">--notrans</span></tt> options are specified, the last one remains
in effect.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--mricoord</span></tt></p>
<blockquote>
<div>Do all computations in the MRI coordinate system. The forward solution
matrix is not affected by this option if the source orientations
are fixed to be normal to the cortical mantle. If all three source components
are included, the forward three source orientations parallel to
the coordinate axes is computed. If <tt class="docutils literal"><span class="pre">--mricoord</span></tt> is present, these
axes correspond to MRI coordinate system rather than the default
MEG head coordinate system. This option is useful only in special
circumstances.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--meas</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>This file is the measurement fif file or an off-line average file
produced thereof. It is recommended that the average file is employed for
evoked-response data and the original raw data file otherwise. This
file provides the MEG sensor locations and orientations as well as
EEG electrode locations as well as the coordinate transformation between
the MEG device coordinates and MEG head-based coordinates.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fwd</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>This file will contain the forward solution as well as the coordinate transformations,
sensor and electrode location information, and the source space
data. A name of the form &lt;<em>name</em>&gt;-fwd.fif is
recommended.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--meg</span></tt></p>
<blockquote>
<div>Compute the MEG forward solution.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eeg</span></tt></p>
<blockquote>
<div>Compute the EEG forward solution.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--grad</span></tt></p>
<blockquote>
<div>Include the derivatives of the fields with respect to the dipole
position coordinates to the output, see <a class="reference internal" href="source_localization/forward.html#bjefejjg"><em>Field derivatives</em></a>.</div></blockquote>
</div>
<div class="section" id="mne-inverse-operator">
<span id="id10"></span><h3><a class="toc-backref" href="#id116">mne_inverse_operator</a><a class="headerlink" href="#mne-inverse-operator" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--meg</span></tt></p>
<blockquote>
<div>Employ MEG data in the calculation of the estimates.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eeg</span></tt></p>
<blockquote>
<div>Employ EEG data in the calculation of the estimates. Note: The EEG
computations have not been thoroughly tested at this time.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fixed</span></tt></p>
<blockquote>
<div>Use fixed source orientations normal to the cortical mantle. By default,
the source orientations are not constrained.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--loose</span> <span class="pre">&lt;amount&gt;</span></tt></p>
<blockquote>
<div>Employ a loose orientation constraint (LOC). This means that the source
covariance matrix entries corresponding to the current component
normal to the cortex are set equal to one and the transverse components
are set to &lt;<em>amount</em>&gt; . Recommended
value of amount is 0.2...0.6.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--loosevar</span> <span class="pre">&lt;amount&gt;</span></tt></p>
<blockquote>
<div>Use an adaptive loose orientation constraint. This option can be
only employed if the source spaces included in the forward solution
have the patch information computed, see <a class="reference internal" href="cookbook.html#setting-up-source-space"><em>Setting up the source space</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fwd</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of the forward solution to use.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--noisecov</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of the noise-covariance matrix to use. If this
file contains a projection operator, attached by <a class="reference internal" href="#mne-browse-raw"><em>mne_browse_raw</em></a> and <a class="reference internal" href="#mne-process-raw"><em>mne_process_raw</em></a>,
no additional projection vectors can be added with the <tt class="docutils literal"><span class="pre">--proj</span></tt> option. For
backward compatibility, <tt class="docutils literal"><span class="pre">--senscov</span></tt> can be used as a synonym for <tt class="docutils literal"><span class="pre">--noisecov</span></tt>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--noiserank</span> <span class="pre">&lt;value&gt;</span></tt></p>
<blockquote>
<div>Specifies the rank of the noise covariance matrix explicitly rather than
trying to reduce it automatically. This option is seldom needed,</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--gradreg</span> <span class="pre">&lt;value&gt;</span></tt></p>
<blockquote>
<div>Regularize the planar gradiometer section (channels for which the unit
of measurement is T/m) of the noise-covariance matrix by the given
amount. The value is restricted to the range 0...1. For details, see <a class="reference internal" href="source_localization/inverse.html#cov-regularization"><em>Regularization of the noise-covariance matrix</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--magreg</span> <span class="pre">&lt;value&gt;</span></tt></p>
<blockquote>
<div>Regularize the magnetometer and axial gradiometer section (channels
for which the unit of measurement is T) of the noise-covariance matrix
by the given amount. The value is restricted to the range 0...1.
For details, see <a class="reference internal" href="source_localization/inverse.html#cov-regularization"><em>Regularization of the noise-covariance matrix</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eegreg</span> <span class="pre">&lt;value&gt;</span></tt></p>
<blockquote>
<div>Regularize the EEG section of the noise-covariance matrix by the given
amount. The value is restricted to the range 0...1. For details, see <a class="reference internal" href="source_localization/inverse.html#cov-regularization"><em>Regularization of the noise-covariance matrix</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--diagnoise</span></tt></p>
<blockquote>
<div>Omit the off-diagonal terms from the noise-covariance matrix in
the computations. This may be useful if the amount of signal-free
data has been insufficient to calculate a reliable estimate of the
full noise-covariance matrix.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--srccov</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of the diagonal source-covariance matrix to use.
By default the source covariance matrix is a multiple of the identity matrix.
This option can be employed to incorporate the fMRI constraint.
The software to create a source-covariance matrix file from fMRI
data will be provided in a future release of this software package.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--depth</span></tt></p>
<blockquote>
<div>Employ depth weighting. For details, see <a class="reference internal" href="source_localization/inverse.html#depth-weighting"><em>Depth weighting</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--weightexp</span> <span class="pre">&lt;value&gt;</span></tt></p>
<blockquote>
<div>This parameter determines the steepness of the depth weighting function
(default = 0.8). For details, see <a class="reference internal" href="source_localization/inverse.html#depth-weighting"><em>Depth weighting</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--weightlimit</span> <span class="pre">&lt;value&gt;</span></tt></p>
<blockquote>
<div>Maximum relative strength of the depth weighting (default = 10). For
details, see <a class="reference internal" href="source_localization/inverse.html#depth-weighting"><em>Depth weighting</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fmri</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>With help of this w file, an <em>a priori</em> weighting
can be applied to the source covariance matrix. The source of the
weighting is usually fMRI but may be also some other data, provided
that the weighting  can be expressed as a scalar value on the cortical
surface, stored in a w file. It is recommended that this w file
is appropriately smoothed (see <a class="reference internal" href="source_localization/morph.html#chdebahh"><em>About smoothing</em></a>) in mne_analyze , tksurfer or
with mne_smooth_w to contain
nonzero values at all vertices of the triangular tessellation of
the cortical surface. The name of the file given is used as a stem of
the w files. The actual files should be called &lt;<em>name</em>&gt; <tt class="docutils literal"><span class="pre">-lh.pri</span></tt> and &lt;<em>name</em>&gt; <tt class="docutils literal"><span class="pre">-rh.pri</span></tt> for
the left and right hemsphere weight files, respectively. The application
of the weighting is discussed in <a class="reference internal" href="source_localization/inverse.html#mne-fmri-estimates"><em>fMRI-guided estimates</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fmrithresh</span> <span class="pre">&lt;value&gt;</span></tt></p>
<blockquote>
<div>This option is mandatory and has an effect only if a weighting function
has been specified with the <tt class="docutils literal"><span class="pre">--fmri</span></tt> option. If the value
is in the <em>a priori</em> files falls below this value
at a particular source space point, the source covariance matrix
values are multiplied by the value specified with the <tt class="docutils literal"><span class="pre">--fmrioff</span></tt> option
(default 0.1). Otherwise it is left unchanged.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fmrioff</span> <span class="pre">&lt;value&gt;</span></tt></p>
<blockquote>
<div>The value by which the source covariance elements are multiplied
if the <em>a priori</em> weight falls below the threshold
set with <tt class="docutils literal"><span class="pre">--fmrithresh</span></tt> , see above.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--bad</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>A text file to designate bad channels, listed one channel name on each
line of the file. If the noise-covariance matrix specified with the <tt class="docutils literal"><span class="pre">--noisecov</span></tt> option
contains projections, bad channel lists can be included only if
they specify all channels containing non-zero entries in a projection
vector. For example, bad channels can usually specify all magnetometers
or all gradiometers since the projection vectors for these channel
types are completely separate. Similarly, it is possible to include
MEG data only or EEG data only by using only one of <tt class="docutils literal"><span class="pre">--meg</span></tt> or <tt class="docutils literal"><span class="pre">--eeg</span></tt> options
since the projection vectors for MEG and EEG are always separate.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--surfsrc</span></tt></p>
<blockquote>
<div>Use a source coordinate system based on the local surface orientation
at the source location. By default, the three dipole components are
pointing to the directions of the x, y, and z axis of the coordinate system
employed in the forward calculation (usually the MEG head coordinate
frame). This option changes the orientation so that the first two
source components lie in the plane normal to the surface normal
at the source location and the third component is aligned with it.
If patch information is available in the source space, the normal
is the average patch normal, otherwise the vertex normal at the source
location is used. If the <tt class="docutils literal"><span class="pre">--loose</span></tt> or <tt class="docutils literal"><span class="pre">--loosevar</span></tt> option
is employed, <tt class="docutils literal"><span class="pre">--surfsrc</span></tt> is implied.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--exclude</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Exclude the source space points defined by the given FreeSurfer &#8216;label&#8217; file
from the source reconstruction. This is accomplished by setting
the corresponding entries in the source-covariance matrix equal
to zero. The name of the file should end with <tt class="docutils literal"><span class="pre">-lh.label</span></tt>
if it refers to the left hemisphere and with <tt class="docutils literal"><span class="pre">-rh.label</span></tt> if
it lists points in the right hemisphere, respectively.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--proj</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Include signal-space projection (SSP) information from this file. For information
on SSP, see <a class="reference internal" href="gui/browse.html#cacchabi"><em>The Signal-Space Projection (SSP) method</em></a>. If the projections are present in
the noise-covariance matrix, the <tt class="docutils literal"><span class="pre">--proj</span></tt> option is
not allowed.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--csd</span></tt></p>
<blockquote>
<div>Compute the inverse operator for surface current densities instead
of the dipole source amplitudes. This requires the computation of patch
statistics for the source space. Since this computation is time consuming,
it is recommended that the patch statistics are precomputed and
the source space file containing the patch information is employed
already when the forward solution is computed, see <a class="reference internal" href="cookbook.html#setting-up-source-space"><em>Setting up the source space</em></a> and <a class="reference internal" href="cookbook.html#babchejd"><em>Computing the forward solution</em></a>.
For technical details of the patch information, please consult <a class="reference internal" href="source_localization/inverse.html#patch-stats"><em>Cortical patch statistics</em></a>. This option is considered experimental at
the moment.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--inv</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Save the inverse operator decomposition here.</div></blockquote>
</div>
<div class="section" id="mne-make-movie">
<span id="id11"></span><h3><a class="toc-backref" href="#id117">mne_make_movie</a><a class="headerlink" href="#mne-make-movie" title="Permalink to this headline">¶</a></h3>
<div class="section" id="input-files">
<h4>Input files<a class="headerlink" href="#input-files" title="Permalink to this headline">¶</a></h4>
<p><tt class="docutils literal"><span class="pre">--inv</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Load the inverse operator decomposition from here.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--meas</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Load the MEG or EEG data from this file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--set</span> <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>The data set (condition) number to load. This is the sequential
number of the condition. You can easily see the association by looking
at the condition list in mne_analyze when
you load the file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--stcin</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies an stc file to read as input.</div></blockquote>
</div>
<div class="section" id="times-and-baseline">
<h4>Times and baseline<a class="headerlink" href="#times-and-baseline" title="Permalink to this headline">¶</a></h4>
<p><tt class="docutils literal"><span class="pre">--tmin</span> <span class="pre">&lt;*time/ms*&gt;</span></tt></p>
<blockquote>
<div>Specifies the starting time employed in the analysis. If <tt class="docutils literal"><span class="pre">--tmin</span></tt> option
is missing the analysis starts from the beginning of the epoch.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--tmax</span> <span class="pre">&lt;*time/ms*&gt;</span></tt></p>
<blockquote>
<div>Specifies the finishing time employed in the analysis. If <tt class="docutils literal"><span class="pre">--tmax</span></tt> option
is missing the analysis extends to the end of the epoch.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--tstep</span> <span class="pre">&lt;*step/ms*&gt;</span></tt></p>
<blockquote>
<div>Time step between consequtive movie frames, specified in milliseconds.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--integ</span>&nbsp; <span class="pre">&lt;*:math:`\Delta`t/ms*&gt;</span></tt></p>
<blockquote>
<div>Integration time for each frame. Defaults to zero. The integration will
be performed on sensor data. If the time specified for a frame is <span class="math">\(t_0\)</span>,
the integration range will be <span class="math">\(t_0 - \Delta t/2 \leq t \leq t_0 + \Delta t/2\)</span>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--pick</span> <span class="pre">&lt;*time/ms*&gt;</span></tt></p>
<blockquote>
<div>Pick a time for the production of rgb, tif, jpg, png, or w files.
Several pick options may be present. The time must be with in the
analysis interval, indicated by the <tt class="docutils literal"><span class="pre">--tmin</span></tt> and <tt class="docutils literal"><span class="pre">--tmax</span></tt> options.
The <tt class="docutils literal"><span class="pre">--rgb</span></tt> , <tt class="docutils literal"><span class="pre">--tif</span></tt> , <tt class="docutils literal"><span class="pre">--jpg</span></tt> , <tt class="docutils literal"><span class="pre">--png</span></tt> , and <tt class="docutils literal"><span class="pre">--w</span></tt> options
control which file types are actually produced. When a <tt class="docutils literal"><span class="pre">--pick</span></tt> option
is encountered, the effect of any preceding <tt class="docutils literal"><span class="pre">--pickrange</span></tt> option
is ignored.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--pickrange</span></tt></p>
<blockquote>
<div>All previous <tt class="docutils literal"><span class="pre">-pick</span></tt> options will be ignored. Instead,
snapshots are produced as indicated by the <tt class="docutils literal"><span class="pre">--tmin</span></tt> , <tt class="docutils literal"><span class="pre">--tmax</span></tt> ,
and <tt class="docutils literal"><span class="pre">--tstep</span></tt> options. This is useful, <em>e.g.</em>,
for producing input for scripts merging the individual graphics
snapshots into a composite &#8220;filmstrip&#8221; reprensentation.
However, such scripts are not yet part of the MNE software.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--bmin</span> <span class="pre">&lt;*time/ms*&gt;</span></tt></p>
<blockquote>
<div>Specifies the starting time of the baseline. In order to activate
baseline correction, both <tt class="docutils literal"><span class="pre">--bmin</span></tt> and <tt class="docutils literal"><span class="pre">--bmax</span></tt> options
must be present.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--bmax</span> <span class="pre">&lt;*time/ms*&gt;</span></tt></p>
<blockquote>
<div>Specifies the finishing time of the baseline.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--baselines</span> <span class="pre">&lt;*file_name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a file which contains the baseline settings. Each line
of the file should contain a name of a channel, followed by the
baseline value, separated from the channel name by a colon. The
baseline values must be specified in basic units, i.e., Teslas/meter
for gradiometers, Teslas for magnetometers, and Volts for EEG channels.
If some channels are missing from the baseline file, warning messages are
issued: for these channels, the <tt class="docutils literal"><span class="pre">--bmin</span></tt> and <tt class="docutils literal"><span class="pre">--bmax</span></tt> settings will
be used.</div></blockquote>
</div>
<div class="section" id="options-controlling-the-estimates">
<h4>Options controlling the estimates<a class="headerlink" href="#options-controlling-the-estimates" title="Permalink to this headline">¶</a></h4>
<p><tt class="docutils literal"><span class="pre">--nave</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>Specifies the effective number of averaged epochs in the input data, <span class="math">\(L_{eff}\)</span>,
as discussed in <a class="reference internal" href="source_localization/inverse.html#cbbdgiae"><em>Effective number of averages</em></a>. If the input data file is
one produced by <a class="reference internal" href="#mne-browse-raw"><em>mne_browse_raw</em></a> or <a class="reference internal" href="#mne-process-raw"><em>mne_process_raw</em></a>, the
number of averages is correct in the file. However, if subtractions
or some more complicated combinations of simple averages are produced,
e.g., by  using the xplotter software,
the number of averages should be manually adjusted along the guidelines
given in <a class="reference internal" href="source_localization/inverse.html#cbbdgiae"><em>Effective number of averages</em></a>. This is accomplished either by
employing this flag or by adjusting the number of averages in the
data file with help of the utility mne_change_nave .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--snr</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>An estimate for the amplitude SNR. The regularization parameter will
be set as <span class="math">\(\lambda^2 = 1/SNR^2\)</span>. The default value is
SNR = 3. Automatic selection of the regularization parameter is
currently not supported.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--spm</span></tt></p>
<blockquote>
<div>Calculate the dSPM instead of the expected current value.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--sLORETA</span></tt></p>
<blockquote>
<div>Calculate the noise-normalized estimate using the sLORETA approach.
sLORETA solutions have in general a smaller location bias than either
the expected current (MNE) or the dSPM.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--signed</span></tt></p>
<blockquote>
<div>Indicate the current direction with respect to the cortex outer
normal by sign. Currents flowing out of the cortex are thus considered
positive (warm colors) and currents flowing into the cortex negative (cold
colors).</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--picknormalcomp</span></tt></p>
<blockquote>
<div>The components of the estimates corresponding to directions tangential
with the cortical mantle are zeroed out.</div></blockquote>
</div>
<div class="section" id="visualization-options">
<span id="cbbbbhif"></span><h4>Visualization options<a class="headerlink" href="#visualization-options" title="Permalink to this headline">¶</a></h4>
<p><tt class="docutils literal"><span class="pre">--subject</span> <span class="pre">&lt;*subject*&gt;</span></tt></p>
<blockquote>
<div>Specifies the subject whose MRI data is employed in the visualization.
This must be the same subject that was used for computing the current
estimates. The environment variable SUBJECTS_DIR must be set to
point to a locations where the subjects are to be found.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--morph</span> <span class="pre">&lt;*subject*&gt;</span></tt></p>
<blockquote>
<div>Morph the data to to the cortical surface of another subject. The Quicktime
movie, stc-file, graphics snapshot, and w-file outputs are affected
by this option, <em>i.e.</em>, they will take the morphing
into account and will represent the data on the cortical surface
of the subject defined with this option. The stc files morphed to
a single subject&#8217;s cortical surface are used by mne_average_estimates to
combine data from different subjects.
If morphing is selected appropriate smoothing must be specified
with the <tt class="docutils literal"><span class="pre">--smooth</span></tt> option. The morphing process can
be made faster by precomputing the necessary morphing maps with mne_make_morph_maps ,
see <a class="reference internal" href="source_localization/morph.html#chdbbhdh"><em>Precomputing the morphing maps</em></a>. More information about morphing and averaging
can be found in <a class="reference internal" href="source_localization/morph.html#ch-morph"><em>Morphing and averaging</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--morphgrade</span> <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>Adjusts the number of vertices in the stc files produced when morphing
is in effect. By default the number of vertices is 10242 corresponding
to &#8211;morphgrade value 5. Allowed values are 3, 4, 5, and 6 corresponding
to 642, 2562, 10242, and 40962 vertices, respectively.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--surface</span> <span class="pre">&lt;*surface</span> <span class="pre">name*&gt;</span></tt></p>
<blockquote>
<div>Name of the surface employed in the visualization. The default is inflated .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--curv</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specify a nonstandard curvature file name. The default curvature files
are <tt class="docutils literal"><span class="pre">lh.curv</span></tt> and <tt class="docutils literal"><span class="pre">rh.curv</span></tt> . With this option,
the names become <tt class="docutils literal"><span class="pre">lh.</span></tt> &lt;<em>name</em>&gt; and <tt class="docutils literal"><span class="pre">rh.</span></tt> &lt;<em>name</em>&gt; .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--patch</span> <span class="pre">&lt;*name*&gt;</span> <span class="pre">[:</span> <span class="pre">&lt;*angle/deg*&gt;</span> <span class="pre">]</span></tt></p>
<blockquote>
<div>Specify the name of a surface patch to be used for visualization instead
of the complete cortical surface. A complete name of a patch file
in the FreeSurface surf directory must be given. The name should
begin with lh or rh to allow association of the patch with a hemisphere.
Maximum of two <tt class="docutils literal"><span class="pre">--patch</span></tt> options can be in effect, one patch for each
hemisphere. If the name refers to a flat patch, the name can be
optionally followed by a colon and a rotation angle in degrees.
The flat patch will be then rotated counterclockwise by this amount
before display. You can check a suitable value for the rotation
angle by loading the patch interactively in mne_analyze .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--width</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>Width of the graphics output frames in pixels. The default width
is 600 pixels.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--height</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>Height of the graphics output frames in pixels. The default height
is 400 pixels.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--mag</span> <span class="pre">&lt;*factor*&gt;</span></tt></p>
<blockquote>
<div>Magnify the the visualized scene by this factor.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--lh</span></tt></p>
<blockquote>
<div>Select the left hemisphere for graphics output. By default, both hemisphere
are processed.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--rh</span></tt></p>
<blockquote>
<div>Select the right hemisphere for graphics output. By default, both hemisphere
are processed.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--view</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Select the name of the view for mov, rgb, and tif graphics output files.
The default viewnames, defined in <tt class="docutils literal"><span class="pre">$MNE_ROOT/share/mne/mne_analyze/eyes</span></tt> ,
are <em>lat</em> (lateral), <em>med</em> (medial), <em>ven</em> (ventral),
and <em>occ</em> (occipital). You can override these
defaults by creating the directory .mne under your home directory
and copying the eyes file there. Each line of the eyes file contais
the name of the view, the viewpoint for the left hemisphere, the
viewpoint for the right hemisphere, left hemisphere up vector, and
right hemisphere up vector. The entities are separated by semicolons.
Lines beginning with the pound sign (#) are considered to be comments.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--smooth</span> <span class="pre">&lt;*nstep*&gt;</span></tt></p>
<blockquote>
<div>Number of smoothsteps to take when producing the output frames. Depending
on the source space decimation, an appropriate number is 4 - 7.
Smoothing does not have any effect for the original brain if stc
files are produced. However, if morphing is selected smoothing is
mandatory even with stc output. For details of the smoothing procedure,
see <a class="reference internal" href="source_localization/morph.html#chdebahh"><em>About smoothing</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--nocomments</span></tt></p>
<blockquote>
<div>Do not include the comments in the image output files or movies.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--noscalebar</span></tt></p>
<blockquote>
<div>Do not include the scalebar in the image output files or movies.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--alpha</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>Adjust the opacity of maps shown on the cortical surface (0 = transparent,
1 = totally opaque). The default value is 1.</div></blockquote>
</div>
<div class="section" id="thresholding">
<h4>Thresholding<a class="headerlink" href="#thresholding" title="Permalink to this headline">¶</a></h4>
<p><tt class="docutils literal"><span class="pre">--fthresh</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>Specifies the threshold for the displayed colormaps. At the threshold,
the overlaid color will be equal to the background surface color.
For currents, the value will be multiplied by <span class="math">\(1^{-10}\)</span>.
The default value is 8.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fmid</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>Specifies the midpoint for the displayed colormaps. At this value, the
overlaid color will be read (positive values) or blue (negative values).
For currents, the value will be multiplied by <span class="math">\(1^{-10}\)</span>.
The default value is 15.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fmax</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>Specifies the maximum point for the displayed colormaps. At this value,
the overlaid color will bright yellow (positive values) or light
blue (negative values). For currents, the value will be multiplied
by <span class="math">\(1^{-10}\)</span>. The default value is 20.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fslope</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>Included for backwards compatibility. If this option is specified
and <tt class="docutils literal"><span class="pre">--fmax</span></tt> option is <em>not</em> specified, <span class="math">\(F_{max} = F_{mid} + 1/F_{slope}\)</span>.</div></blockquote>
</div>
<div class="section" id="output-files">
<h4>Output files<a class="headerlink" href="#output-files" title="Permalink to this headline">¶</a></h4>
<p><tt class="docutils literal"><span class="pre">--mov</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Produce QuickTime movie files. This is the &#8216;stem&#8217; of
the ouput file name. The actual name is derived by stripping anything
up to and including the last period from the end of &lt;<em>name</em>&gt; .
According to the hemisphere, <tt class="docutils literal"><span class="pre">-lh</span></tt> or <tt class="docutils literal"><span class="pre">-rh</span></tt> is
then appended. The name of the view is indicated with <tt class="docutils literal"><span class="pre">-</span></tt> &lt;<em>viename</em>&gt; .
Finally, <tt class="docutils literal"><span class="pre">.mov</span></tt> is added to indicate a QuickTime output
file. The movie is produced for all times as dictated by the <tt class="docutils literal"><span class="pre">--tmin</span></tt> , <tt class="docutils literal"><span class="pre">--tmax</span></tt> , <tt class="docutils literal"><span class="pre">--tstep</span></tt> ,
and <tt class="docutils literal"><span class="pre">--integ</span></tt> options.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--qual</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>Quality of the QuickTime movie output. The default quality is 80 and
allowed range is 25 - 100. The size of the movie files is a monotonously
increasing function of the movie quality.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--rate</span> <span class="pre">&lt;*rate*&gt;</span></tt></p>
<blockquote>
<div>Specifies the frame rate of the QuickTime movies. The default value is <span class="math">\(1/(10t_{step})\)</span>,
where <span class="math">\(t_{step}\)</span> is the time between subsequent
movie frames produced in seconds.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--rgb</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Produce rgb snapshots. This is the &#8216;stem&#8217; of the
ouput file name. The actual name is derived by stripping anything
up to and including the last period from the end of &lt;<em>name</em>&gt; .
According to the hemisphere, <tt class="docutils literal"><span class="pre">-lh</span></tt> or <tt class="docutils literal"><span class="pre">-rh</span></tt> is
then appended. The name of the view is indicated with <tt class="docutils literal"><span class="pre">-</span></tt> &lt;<em>viename</em>&gt; .
Finally, <tt class="docutils literal"><span class="pre">.rgb</span></tt> is added to indicate an rgb output file.
Files are produced for all picked times as dictated by the <tt class="docutils literal"><span class="pre">--pick</span></tt> and <tt class="docutils literal"><span class="pre">--integ</span></tt> options.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--tif</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Produce tif snapshots. This is the &#8216;stem&#8217; of the
ouput file name. The actual name is derived by stripping anything
up to and including the last period from the end of &lt;<em>name</em>&gt; .
According to the hemisphere, <tt class="docutils literal"><span class="pre">-lh</span></tt> or <tt class="docutils literal"><span class="pre">-rh</span></tt> is
then appended. The name of the view is indicated with <tt class="docutils literal"><span class="pre">-</span></tt> &lt;<em>viename</em>&gt; .
Finally, <tt class="docutils literal"><span class="pre">.tif</span></tt> is added to indicate an rgb output file.
Files are produced for all picked times as dictated by the <tt class="docutils literal"><span class="pre">--pick</span></tt> and <tt class="docutils literal"><span class="pre">--integ</span></tt> options.
The tif output files are <em>not</em> compressed. Pass
the files through an image processing program to compress them.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--jpg</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Produce jpg snapshots. This is the &#8216;stem&#8217; of the
ouput file name. The actual name is derived by stripping anything
up to and including the last period from the end of &lt;<em>name</em>&gt; .
According to the hemisphere, <tt class="docutils literal"><span class="pre">-lh</span></tt> or <tt class="docutils literal"><span class="pre">-rh</span></tt> is
then appended. The name of the view is indicated with <tt class="docutils literal"><span class="pre">-</span></tt> &lt;<em>viename</em>&gt; .
Finally, <tt class="docutils literal"><span class="pre">.jpg</span></tt> is added to indicate an rgb output file.
Files are produced for all picked times as dictated by the <tt class="docutils literal"><span class="pre">--pick</span></tt> and <tt class="docutils literal"><span class="pre">--integ</span></tt> options.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--png</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Produce png snapshots. This is the &#8216;stem&#8217; of the
ouput file name. The actual name is derived by stripping anything
up to and including the last period from the end of &lt;<em>name</em>&gt; .
According to the hemisphere, <tt class="docutils literal"><span class="pre">-lh</span></tt> or <tt class="docutils literal"><span class="pre">-rh</span></tt> is
then appended. The name of the view is indicated with <tt class="docutils literal"><span class="pre">-</span></tt> &lt;<em>viename</em>&gt; .
Finally, <tt class="docutils literal"><span class="pre">.png</span></tt> is added to indicate an rgb output file.
Files are produced for all picked times as dictated by the <tt class="docutils literal"><span class="pre">--pick</span></tt> and <tt class="docutils literal"><span class="pre">--integ</span></tt> options.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--w</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Produce w file snapshots. This is the &#8216;stem&#8217; of
the ouput file name. The actual name is derived by stripping anything
up to and including the last period from the end of &lt;<em>name</em>&gt; .
According to the hemisphere, <tt class="docutils literal"><span class="pre">-lh</span></tt> .w or <tt class="docutils literal"><span class="pre">-rh</span></tt> .w
is then appended. Files are produced for all picked times as dictated
by the <tt class="docutils literal"><span class="pre">--pick</span></tt> and <tt class="docutils literal"><span class="pre">--integ</span></tt> options.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--stc</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Produce stc files for either the original subject or the one selected with
the <tt class="docutils literal"><span class="pre">--morph</span></tt> option. These files will contain data only
for the decimated locations. If morphing is selected, appropriate
smoothing is mandatory. The morphed maps will be decimated with
help of a subdivided icosahedron so that the morphed stc files will
always contain 10242 vertices. These morphed stc files can be easily
averaged together, e.g., in Matlab since they always contain an
identical set of vertices.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--norm</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Indicates that a separate w file
containing the noise-normalization values will be produced. The
option <tt class="docutils literal"><span class="pre">--spm</span></tt> must also be present. Nevertheless, the
movies and stc files output will
contain MNE values. The noise normalization data files will be called &lt;<em>name</em>&gt;- &lt;<em>SNR</em>&gt; <tt class="docutils literal"><span class="pre">-lh.w</span></tt> and &lt;<em>name</em>&gt;- &lt;<em>SNR</em>&gt; <tt class="docutils literal"><span class="pre">-rh.w</span></tt> .</div></blockquote>
</div>
<div class="section" id="label-processing">
<span id="cbbhhcef"></span><h4>Label processing<a class="headerlink" href="#label-processing" title="Permalink to this headline">¶</a></h4>
<p><tt class="docutils literal"><span class="pre">--label</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a label file to process. For each label file, the values
of the computed estimates are listed in text files. The label files
are produced by tksurfer or mne_analyze and
specify regions of interests (ROIs). A label file name should end
with <tt class="docutils literal"><span class="pre">-lh.label</span></tt> for left-hemisphere ROIs and with <tt class="docutils literal"><span class="pre">-rh.label</span></tt> for
right-hemisphere ones. The corresponding output files are tagged
with <tt class="docutils literal"><span class="pre">-lh-</span></tt> &lt;<em>data type</em>&gt; <tt class="docutils literal"><span class="pre">.amp</span></tt> and <tt class="docutils literal"><span class="pre">-rh-</span></tt> &lt;<em>data type</em>&gt; <tt class="docutils literal"><span class="pre">.amp</span></tt>, respectively. &lt;<em>data type</em>&gt; equals <tt class="docutils literal"><span class="pre">'mne</span></tt> &#8216; for
expected current data and <tt class="docutils literal"><span class="pre">'spm</span></tt> &#8216; for
dSPM data. Each line of the output file contains the waveform of
the output quantity at one of the source locations falling inside
the ROI. For more information about the label output formats, see <a class="reference internal" href="gui/analyze.html#cacjjgfa"><em>Label timecourse files</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--labelcoords</span></tt></p>
<blockquote>
<div>Include coordinates of the vertices in the output. The coordinates will
be listed in millimeters in the coordinate system which was specified
for the forward model computations. This option cannot be used with
stc input files (<tt class="docutils literal"><span class="pre">--stcin</span></tt> ) because the stc files do
not contain the coordinates of the vertices.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--labelverts</span></tt></p>
<blockquote>
<div>Include vertex numbers in the output. The numbers refer to the complete
triangulation of the corresponding surface and are zero based. The
vertex numbers are by default on the first row or first column of the
output file depending on whether or not the <tt class="docutils literal"><span class="pre">--labeltimebytime</span></tt> option
is present.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--labeltimebytime</span></tt></p>
<blockquote>
<div>Output the label data time by time instead of the default vertex-by-vertex
output.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--labeltag</span> <span class="pre">&lt;*tag*&gt;</span></tt></p>
<blockquote>
<div>End the output files with the specified tag. By default, the output files
will end with <tt class="docutils literal"><span class="pre">-mne.amp</span></tt> or <tt class="docutils literal"><span class="pre">-spm.amp</span></tt> depending
on whether MNE or one of the noise-normalized estimates (dSPM or sLORETA)
was selected.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--labeloutdir</span> <span class="pre">&lt;*directory*&gt;</span></tt></p>
<blockquote>
<div>Specifies the directory where the output files will be located.
By default, they will be in the current working directory.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--labelcomments</span></tt></p>
<blockquote>
<div>Include comments in the output files. The comment lines begin with the
percent sign to make the files compatible with Matlab.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--scaleby</span> <span class="pre">&lt;*factor*&gt;</span></tt></p>
<blockquote>
<div>By default, the current values output to the files will be in the
actual physical units (Am). This option allows scaling of the current
values to other units. mne_analyze typically
uses 1e10 to bring the numbers to a human-friendly scale.</div></blockquote>
</div>
<div class="section" id="using-stc-file-input">
<h4>Using stc file input<a class="headerlink" href="#using-stc-file-input" title="Permalink to this headline">¶</a></h4>
<p>The <tt class="docutils literal"><span class="pre">--stcin</span></tt> option allows input of stc files.
This feature has several uses:</p>
<ul class="simple">
<li>QuickTime movies can be produced from
existing stc files without having to resort to EasyMeg.</li>
<li>Graphics snapshot can be produced from existing stc files.</li>
<li>Existing stc files can be temporally resampled with help of
the <tt class="docutils literal"><span class="pre">--tmin</span></tt> , <tt class="docutils literal"><span class="pre">--tmax</span></tt> , <tt class="docutils literal"><span class="pre">--tstep</span></tt> ,
and <tt class="docutils literal"><span class="pre">--integ</span></tt> options.</li>
<li>Existing stc files can be morphed to another cortical surface
by specifying the <tt class="docutils literal"><span class="pre">--morph</span></tt> option.</li>
<li>Timecourses can be inquired and stored into text files with
help of the <tt class="docutils literal"><span class="pre">--label</span></tt> options, see above.</li>
</ul>
</div>
</div>
<div class="section" id="mne-make-source-space">
<span id="id12"></span><h3><a class="toc-backref" href="#id118">mne_make_source_space</a><a class="headerlink" href="#mne-make-source-space" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--subject</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Name of the subject.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--morph</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Name of the subject to morph the source space to.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--spacing</span> <span class="pre">&lt;dist&gt;</span></tt></p>
<blockquote>
<div>Approximate source space spacing in mm.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--ico</span> <span class="pre">&lt;grade&gt;</span></tt></p>
<blockquote>
<div>Use the subdivided icosahedron or octahedron in downsampling instead of the &#8211;spacing option.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--oct</span> <span class="pre">&lt;grade&gt;</span></tt></p>
<blockquote>
<div>Same as &#8211;ico -grade.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--surf</span> <span class="pre">&lt;names&gt;</span></tt></p>
<blockquote>
<div>Surface file names (separated by colons)</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--src</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Name of the output file.</div></blockquote>
</div>
<div class="section" id="mne-process-raw">
<span id="id13"></span><h3><a class="toc-backref" href="#id119">mne_process_raw</a><a class="headerlink" href="#mne-process-raw" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--cd</span> <span class="pre">&lt;*dir*&gt;</span></tt></p>
<blockquote>
<div>Change to this directory before starting.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--raw</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the raw data file to be opened. This option is required.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--grad</span> <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>Apply software gradient compensation of the given order to the data loaded
with the <tt class="docutils literal"><span class="pre">--raw</span></tt> option. This option is effective only
for data acquired with the CTF and 4D Magnes MEG systems. If orders
different from zero are requested for Neuromag data, an error message appears
and data are not loaded. Any compensation already existing in the
file can be undone or changed to another order by using an appropriate <tt class="docutils literal"><span class="pre">--grad</span></tt> options.
Possible orders are 0 (No compensation), 1 - 3 (CTF data), and 101
(Magnes data). The same compensation will be applied to all loaded data
files.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--filtersize</span> <span class="pre">&lt;*size*&gt;</span></tt></p>
<blockquote>
<div>Adjust the length of the FFT to be applied in filtering. The number will
be rounded up to the next power of two. If the size is <span class="math">\(N\)</span>,
the corresponding length of time is <span class="math">\(N/f_s\)</span>,
where <span class="math">\(f_s\)</span> is the sampling frequency
of your data. The filtering procedure includes overlapping tapers
of length <span class="math">\(N/2\)</span> so that the total FFT
length will actually be <span class="math">\(2N\)</span>. This
value cannot be changed after the program has been started.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--highpass</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>Highpass filter frequency limit. If this is too low with respect
to the selected FFT length and, the data will not be highpass filtered. It
is best to experiment with the interactive version to find the lowest applicable
filter for your data. This value can be adjusted in the interactive
version of the program. The default is 0, <em>i.e.</em>,
no highpass filter apart from that used during the acquisition will
be in effect.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--highpassw</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>The width of the transition band of the highpass filter. The default
is 6 frequency bins, where one bin is <span class="math">\(f_s / (2N)\)</span>. This
value cannot be adjusted in the interactive version of the program.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--lowpass</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>Lowpass filter frequency limit. This value can be adjusted in the interactive
version of the program. The default is 40 Hz.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--lowpassw</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>The width of the transition band of the lowpass filter. This value
can be adjusted in the interactive version of the program. The default
is 5 Hz.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eoghighpass</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>Highpass filter frequency limit for EOG. If this is too low with respect
to the selected FFT length and, the data will not be highpass filtered.
It is best to experiment with the interactive version to find the
lowest applicable filter for your data. This value can be adjusted in
the interactive version of the program. The default is 0, <em>i.e.</em>,
no highpass filter apart from that used during the acquisition will
be in effect.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eoghighpassw</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>The width of the transition band of the EOG highpass filter. The default
is 6 frequency bins, where one bin is <span class="math">\(f_s / (2N)\)</span>.
This value cannot be adjusted in the interactive version of the
program.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eoglowpass</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>Lowpass filter frequency limit for EOG. This value can be adjusted in
the interactive version of the program. The default is 40 Hz.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eoglowpassw</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>The width of the transition band of the EOG lowpass filter. This value
can be adjusted in the interactive version of the program. The default
is 5 Hz.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--filteroff</span></tt></p>
<blockquote>
<div>Do not filter the data. This initial value can be changed in the
interactive version of the program.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--digtrig</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Name of the composite digital trigger channel. The default value
is &#8216;STI 014&#8217;. Underscores in the channel name
will be replaced by spaces.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--digtrigmask</span> <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>Mask to be applied to the trigger channel values before considering them.
This option is useful if one wants to set some bits in a don&#8217;t care
state. For example, some finger response pads keep the trigger lines
high if not in use, <em>i.e.</em>, a finger is not in
place. Yet, it is convenient to keep these devices permanently connected
to the acquisition system. The number can be given in decimal or
hexadecimal format (beginning with 0x or 0X). For example, the value
255 (0xFF) means that only the lowest order byte (usually trigger
lines 1 - 8 or bits 0 - 7) will be considered.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--proj</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specify the name of the file of the file containing a signal-space
projection (SSP) operator. If <tt class="docutils literal"><span class="pre">--proj</span></tt> options are present
the data file is not consulted for an SSP operator. The operator
corresponding to average EEG reference is always added if EEG data
are present.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--projon</span></tt></p>
<blockquote>
<div>Activate the projections loaded. One of the options <tt class="docutils literal"><span class="pre">--projon</span></tt> or <tt class="docutils literal"><span class="pre">--projoff</span></tt> must
be present on the mne_processs_raw command line.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--projoff</span></tt></p>
<blockquote>
<div>Deactivate the projections loaded. One of the options <tt class="docutils literal"><span class="pre">--projon</span></tt> or <tt class="docutils literal"><span class="pre">--projoff</span></tt> must
be present on the mne_processs_raw command line.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--makeproj</span></tt></p>
<blockquote>
<div>Estimate the noise subspace from the data and create a new signal-space
projection operator instead of using one attached to the data file
or those specified with the <tt class="docutils literal"><span class="pre">--proj</span></tt> option. The following
eight options define the parameters of the noise subspace estimation. More
information on the signal-space projection can be found in <a class="reference internal" href="gui/browse.html#cacchabi"><em>The Signal-Space Projection (SSP) method</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--projevent</span> <span class="pre">&lt;*no*&gt;</span></tt></p>
<blockquote>
<div>Specifies the events which identify the time points of interest
for projector calculation. When this option is present, <tt class="docutils literal"><span class="pre">--projtmin</span></tt> and <tt class="docutils literal"><span class="pre">--projtmax</span></tt> are
relative to the time point of the event rather than the whole raw
data file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--projtmin</span> <span class="pre">&lt;*time/s*&gt;</span></tt></p>
<blockquote>
<div>Specify the beginning time for the calculation of the covariance matrix
which serves as the basis for the new SSP operator. This option
is required with <tt class="docutils literal"><span class="pre">--projevent</span></tt> and defaults to the beginning
of the raw data file otherwise. This option is effective only if <tt class="docutils literal"><span class="pre">--makeproj</span></tt> or <tt class="docutils literal"><span class="pre">--saveprojtag</span></tt> options
are present.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--projtmax</span> <span class="pre">&lt;*time/s*&gt;</span></tt></p>
<blockquote>
<div>Specify the ending time for the calculation of the covariance matrix which
serves as the basis for the new SSP operator. This option is required
with <tt class="docutils literal"><span class="pre">--projevent</span></tt> and defaults to the end of the raw data
file otherwise. This option is effective only if <tt class="docutils literal"><span class="pre">--makeproj</span></tt> or <tt class="docutils literal"><span class="pre">--saveprojtag</span></tt> options
are present.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--projngrad</span> <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>Number of SSP components to include for planar gradiometers (default
= 5). This value is system dependent. For example, in a well-shielded
quiet environment, no planar gradiometer projections are usually
needed.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--projnmag</span> <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>Number of SSP components to include for magnetometers / axial gradiometers
(default = 8). This value is system dependent. For example, in a
well-shielded quiet environment, 3 - 4 components are need
while in a noisy environment with light shielding even more than
8 components may be necessary.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--projgradrej</span> <span class="pre">&lt;*value/</span> <span class="pre">fT/cm*&gt;</span></tt></p>
<blockquote>
<div>Rejection limit for planar gradiometers in the estimation of the covariance
matrix frfixom which the new SSP operator is derived. The default
value is 2000 fT/cm. Again, this value is system dependent.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--projmagrej</span> <span class="pre">&lt;*value/</span> <span class="pre">fT*&gt;</span></tt></p>
<blockquote>
<div>Rejection limit for planar gradiometers in the estimation of the covariance
matrix from which the new SSP operator is derived. The default value
is 3000 fT. Again, this value is system dependent.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--saveprojtag</span> <span class="pre">&lt;*tag*&gt;</span></tt></p>
<blockquote>
<div>This option defines the names of files to hold the SSP operator.
If this option is present the <tt class="docutils literal"><span class="pre">--makeproj</span></tt> option is
implied. The SSP operator file name is formed by removing the trailing <tt class="docutils literal"><span class="pre">.fif</span></tt> or <tt class="docutils literal"><span class="pre">_raw.fif</span></tt> from
the raw data file name by appending  &lt;<em>tag</em>&gt; .fif
to this stem. Recommended value for &lt;<em>tag</em>&gt; is <tt class="docutils literal"><span class="pre">-proj</span></tt> .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--saveprojaug</span></tt></p>
<blockquote>
<div>Specify this option if you want to use the projection operator file output
in the Elekta-Neuromag Signal processor (graph) software.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eventsout</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>List the digital trigger channel events to the specified file. By default,
only transitions from zero to a non-zero value are listed. If multiple
raw data files are specified, an equal number of <tt class="docutils literal"><span class="pre">--eventsout</span></tt> options
should be present. If the file name ends with .fif, the output will
be in fif format, otherwise a text event file will be output.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--allevents</span></tt></p>
<blockquote>
<div>List all transitions to file specified with the <tt class="docutils literal"><span class="pre">--eventsout</span></tt> option.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--events</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of a fif or text format event file (see <a class="reference internal" href="gui/browse.html#cacbcegc"><em>Event files</em></a>) to be associated with a raw data file to be
processed. If multiple raw data files are specified, the number
of <tt class="docutils literal"><span class="pre">--events</span></tt> options can be smaller or equal to the
number of raw data files. If it is equal, the event filenames will
be associated with the raw data files in the order given. If it
is smaller, the remaining raw data files for which an event file
is not specified will <em>not</em> have an event file associated
with them. The event file format is recognized from the file name:
if it ends with <tt class="docutils literal"><span class="pre">.fif</span></tt> , the file is assumed to be in
fif format, otherwise a text file is expected.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--ave</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of an off-line averaging description file. For details
of the format of this file, please consult <a class="reference internal" href="gui/browse.html#cacbbdgc"><em>Description files for off-line averaging</em></a>.
If multiple raw data files are specified, the number of <tt class="docutils literal"><span class="pre">--ave</span></tt> options
can be smaller or equal to the number of raw data files. If it is
equal, the averaging description file names will be associated with
the raw data files in the order given. If it is smaller, the last
description file will be used for the remaining raw data files.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--saveavetag</span> <span class="pre">&lt;*tag*&gt;</span></tt></p>
<blockquote>
<div>If this option is present and averaging is evoked with the <tt class="docutils literal"><span class="pre">--ave</span></tt> option,
the outfile and logfile options in the averaging description file
are ignored. Instead, trailing <tt class="docutils literal"><span class="pre">.fif</span></tt> or <tt class="docutils literal"><span class="pre">_raw.fif</span></tt> is
removed from the raw data file name and &lt;<em>tag</em>&gt; <tt class="docutils literal"><span class="pre">.fif</span></tt> or &lt;<em>tag</em>&gt; <tt class="docutils literal"><span class="pre">.log</span></tt> is appended
to create the output and log file names, respectively.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--gave</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>If multiple raw data files are specified as input and averaging
is requested, the grand average over all data files will be saved
to &lt;<em>name</em>&gt; .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--cov</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specify the name of a description file for covariance matrix estimation. For
details of the format of this file, please see <a class="reference internal" href="gui/browse.html#cacebacg"><em>Description files for covariance matrix estimation</em></a>.
If multiple raw data files are specified, the number of <tt class="docutils literal"><span class="pre">--cov</span></tt> options can
be smaller or equal to the number of raw data files. If it is equal, the
averaging description file names will be associated with the raw data
files in the order given. If it is smaller, the last description
file will be used for the remaining raw data files.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--savecovtag</span> <span class="pre">&lt;*tag*&gt;</span></tt></p>
<blockquote>
<div>If this option is present and covariance matrix estimation is evoked with
the <tt class="docutils literal"><span class="pre">--cov</span></tt> option, the outfile and logfile options in
the covariance estimation description file are ignored. Instead,
trailing <tt class="docutils literal"><span class="pre">.fif</span></tt> or <tt class="docutils literal"><span class="pre">_raw.fif</span></tt> is removed from
the raw data file name and &lt;<em>tag</em>&gt; .fif or &lt;<em>tag</em>&gt; .log
is appended to create the output and log file names, respectively.
For compatibility with other MNE software scripts, <tt class="docutils literal"><span class="pre">--savecovtag</span> <span class="pre">-cov</span></tt> is recommended.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--savehere</span></tt></p>
<blockquote>
<div>If the <tt class="docutils literal"><span class="pre">--saveavetag</span></tt> and <tt class="docutils literal"><span class="pre">--savecovtag</span></tt> options
are used to generate the file output file names, the resulting files
will go to the same directory as raw data by default. With this
option the output files will be generated in the current working
directory instead.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--gcov</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>If multiple raw data files are specified as input and covariance matrix estimation
is requested, the grand average over all data files will be saved
to &lt;<em>name</em>&gt; . The details of
the covariance matrix estimation are given in <a class="reference internal" href="gui/browse.html#cachaaeg"><em>Covariance matrix estimation</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--save</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Save a filtered and optionally down-sampled version of the data
file to &lt;<em>name</em>&gt; . If multiple
raw data files are specified, an equal number of <tt class="docutils literal"><span class="pre">--save</span></tt> options
should be present. If &lt;<em>filename</em>&gt; ends
with <tt class="docutils literal"><span class="pre">.fif</span></tt> or <tt class="docutils literal"><span class="pre">_raw.fif</span></tt> , these endings are
deleted. After these modifications, <tt class="docutils literal"><span class="pre">_raw.fif</span></tt> is inserted
after the remaining part of the file name. If the file is split
into multiple parts (see <tt class="docutils literal"><span class="pre">--split</span></tt> option below), the
additional parts will be called &lt;<em>name</em>&gt; <tt class="docutils literal"><span class="pre">-</span></tt> &lt;<em>number</em>&gt; <tt class="docutils literal"><span class="pre">_raw.fif</span></tt></div></blockquote>
<p><tt class="docutils literal"><span class="pre">--split</span> <span class="pre">&lt;*size/MB*&gt;</span></tt></p>
<blockquote>
<div>Specifies the maximum size of the raw data files saved with the <tt class="docutils literal"><span class="pre">--save</span></tt> option.
By default, the output is split into files which are just below
2 GB so that the fif file maximum size is not exceed.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--anon</span></tt></p>
<blockquote>
<div>Do not include any subject information in the output files created with
the <tt class="docutils literal"><span class="pre">--save</span></tt> option.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--decim</span> <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>The data are decimated by this factor before saving to the file
specified with the <tt class="docutils literal"><span class="pre">--save</span></tt> option. For decimation to
succeed, the data must be lowpass filtered to less than third of
the sampling frequency effective after decimation.</div></blockquote>
</div>
<div class="section" id="mne-redo-file">
<span id="id14"></span><h3><a class="toc-backref" href="#id120">mne_redo_file</a><a class="headerlink" href="#mne-redo-file" title="Permalink to this headline">¶</a></h3>
<p>Usage: <tt class="docutils literal"><span class="pre">mne_redo_file</span> <span class="pre">file-to-redo</span></tt></p>
</div>
<div class="section" id="mne-redo-file-nocwd">
<span id="id15"></span><h3><a class="toc-backref" href="#id121">mne_redo_file_nocwd</a><a class="headerlink" href="#mne-redo-file-nocwd" title="Permalink to this headline">¶</a></h3>
<p>Usage: <tt class="docutils literal"><span class="pre">mne_redo_file_nocwd</span> <span class="pre">file-to-redo</span></tt></p>
</div>
<div class="section" id="mne-setup-forward-model">
<span id="id16"></span><h3><a class="toc-backref" href="#id122">mne_setup_forward_model</a><a class="headerlink" href="#mne-setup-forward-model" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--subject</span> <span class="pre">&lt;*subject*&gt;</span></tt></p>
<blockquote>
<div>Defines the name of the subject. This can be also accomplished
by setting the SUBJECT environment variable.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--surf</span></tt></p>
<blockquote>
<div>Use the FreeSurfer surface files instead of the default ASCII triangulation
files. Please consult <a class="reference internal" href="cookbook.html#babdbbfc"><em>Setting up the head surface triangulation files</em></a> for the standard file
naming scheme.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--noswap</span></tt></p>
<blockquote>
<div>Traditionally, the vertices of the triangles in &#8216;tri&#8217; files
have been ordered so that, seen from the outside of the triangulation,
the vertices are ordered in clockwise fashion. The fif files, however,
employ the more standard convention with the vertices ordered counterclockwise.
Therefore, mne_setup_forward_model by
default reverses the vertex ordering before writing the fif file.
If, for some reason, you have counterclockwise-ordered tri files
available this behavior can be turned off by defining <tt class="docutils literal"><span class="pre">--noswap</span></tt> .
When the fif file is created, the vertex ordering is checked and
the process is aborted if it is incorrect after taking into account
the state of the swapping. Should this happen, try to run mne_setup_forward_model again including
the <tt class="docutils literal"><span class="pre">--noswap</span></tt> flag. In particular, if you employ the seglab software
to create the triangulations (see <a class="reference internal" href="appendix/bem_model.html#create-bem-model"><em>Creating the BEM meshes</em></a>), the <tt class="docutils literal"><span class="pre">--noswap</span></tt> flag
is required. This option is ignored if <tt class="docutils literal"><span class="pre">--surf</span></tt> is specified</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--ico</span> <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>This option is relevant (and required) only with the <tt class="docutils literal"><span class="pre">--surf</span></tt> option and
if the surface files have been produced by the watershed algorithm.
The watershed triangulations are isomorphic with an icosahedron,
which has been recursively subdivided six times to yield 20480 triangles.
However, this number of triangles results in a long computation
time even in a workstation with generous amounts of memory. Therefore,
the triangulations have to be decimated. Specifying <tt class="docutils literal"><span class="pre">--ico</span> <span class="pre">4</span></tt> yields 5120 triangles per surface while <tt class="docutils literal"><span class="pre">--ico</span> <span class="pre">3</span></tt> results
in 1280 triangles. The recommended choice is <tt class="docutils literal"><span class="pre">--ico</span> <span class="pre">4</span></tt> .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--homog</span></tt></p>
<blockquote>
<div>Use a single compartment model (brain only) instead a three layer one
(scalp, skull, and brain). Only the <tt class="docutils literal"><span class="pre">inner_skull.tri</span></tt> triangulation
is required. This model is usually sufficient for MEG but invalid
for EEG. If you are employing MEG data only, this option is recommended
because of faster computation times. If this flag is specified,
the options <tt class="docutils literal"><span class="pre">--brainc</span></tt> , <tt class="docutils literal"><span class="pre">--skullc</span></tt> , and <tt class="docutils literal"><span class="pre">--scalpc</span></tt> are irrelevant.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--brainc</span> <span class="pre">&lt;*conductivity/</span> <span class="pre">S/m*&gt;</span></tt></p>
<blockquote>
<div>Defines the brain compartment conductivity. The default value is 0.3 S/m.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--skullc</span> <span class="pre">&lt;*conductivity/</span> <span class="pre">S/m*&gt;</span></tt></p>
<blockquote>
<div>Defines the skull compartment conductivity. The default value is 0.006 S/m
corresponding to a conductivity ratio 1/50 between the brain and
skull compartments.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--scalpc</span> <span class="pre">&lt;*conductivity/</span> <span class="pre">S/m*&gt;</span></tt></p>
<blockquote>
<div>Defines the brain compartment conductivity. The default value is 0.3 S/m.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--innershift</span> <span class="pre">&lt;*value/mm*&gt;</span></tt></p>
<blockquote>
<div>Shift the inner skull surface outwards along the vertex normal directions
by this amount.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--outershift</span> <span class="pre">&lt;*value/mm*&gt;</span></tt></p>
<blockquote>
<div>Shift the outer skull surface outwards along the vertex normal directions
by this amount.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--scalpshift</span> <span class="pre">&lt;*value/mm*&gt;</span></tt></p>
<blockquote>
<div>Shift the scalp surface outwards along the vertex normal directions by
this amount.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--nosol</span></tt></p>
<blockquote>
<div>Omit the BEM model geometry dependent data preparation step. This
can be done later by running mne_setup_forward_model without the <tt class="docutils literal"><span class="pre">--nosol</span></tt> option.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--model</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Name for the BEM model geometry file. The model will be created into
the directory bem as &lt;<em>name</em>&gt;- <tt class="docutils literal"><span class="pre">bem.fif</span></tt> .        If
this option is missing, standard model names will be used (see below).</div></blockquote>
</div>
<div class="section" id="mne-setup-mri">
<span id="id17"></span><h3><a class="toc-backref" href="#id123">mne_setup_mri</a><a class="headerlink" href="#mne-setup-mri" title="Permalink to this headline">¶</a></h3>
<p>This command sets up the directories <tt class="docutils literal"><span class="pre">subjects/$SUBJECT/mri/T1-neuromag</span></tt> and
<tt class="docutils literal"><span class="pre">subjects/$SUBJECT/mri/brain-neuromag</span></tt> .</p>
</div>
<div class="section" id="mne-setup-source-space">
<span id="id18"></span><h3><a class="toc-backref" href="#id124">mne_setup_source_space</a><a class="headerlink" href="#mne-setup-source-space" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--subject</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Name of the subject in SUBJECTS_DIR. In the absence of this option,
the SUBJECT environment variable will be consulted. If it is not
defined, mne_setup_source_space exits
with an error.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--morph</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Name of a subject in SUBJECTS_DIR. If this option is present, the source
space will be first constructed for the subject defined by the &#8211;subject
option or the SUBJECT environment variable and then morphed to this
subject. This option is useful if you want to create a source spaces
for several subjects and want to directly compare the data across
subjects at the source space vertices without any morphing procedure
afterwards. The drawback of this approach is that the spacing between
source locations in the &#8220;morph&#8221; subject is not going
to be as uniform as it would be without morphing.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--surf</span> <span class="pre">&lt;*name1*&gt;:</span> <span class="pre">&lt;*name2*&gt;:...</span></tt></p>
<blockquote>
<div>FreeSurfer surface file names specifying the source surfaces, separated
by colons.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--spacing</span> <span class="pre">&lt;*spacing/mm*&gt;</span></tt></p>
<blockquote>
<div>Specifies the approximate grid spacing of the source space in mm.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--ico</span> <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>Instead of using the traditional method for cortical surface decimation
it is possible to create the source space using the topology of
a recursively subdivided icosahedron ( &lt;<em>number</em>&gt; &gt; 0)
or an octahedron ( &lt;<em>number</em>&gt;  &lt; 0).
This method uses the cortical surface inflated to a sphere as a
tool to find the appropriate vertices for the source space. The
benefit of the <tt class="docutils literal"><span class="pre">--ico</span></tt> option is that the source space will have triangulation
information between the decimated vertices included, which some
future versions of MNE software may be able to utilize. The number
of triangles increases by a factor of four in each subdivision,
starting from 20 triangles in an icosahedron and 8 triangles in
an octahedron. Since the number of vertices on a closed surface
is <span class="math">\(n_{vert} = (n_{tri} + 4) / 2\)</span>, the number of vertices in
the <em>k</em> th subdivision of an icosahedron and an
octahedron are <span class="math">\(10 \cdot 4^k +2\)</span> and <span class="math">\(4_{k + 1} + 2\)</span>,
respectively. The recommended values for &lt;<em>number</em>&gt; and
the corresponding number of source space locations are listed in Table 3.1.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--all</span></tt></p>
<blockquote>
<div>Include all nodes to the output. The active dipole nodes are identified
in the fif file by a separate tag. If tri files were used as input
the output file will also contain information about the surface
triangulation. This option is always recommended to include complete
information.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--src</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Output file name. Use a name &lt;<em>dir</em>&gt;/&lt;<em>name</em>&gt;-src.fif</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">If both <tt class="docutils literal"><span class="pre">--ico</span></tt> and <tt class="docutils literal"><span class="pre">--spacing</span></tt> options    are present the later one on the command line takes precedence.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Due to the differences between the FreeSurfer    and MNE libraries, the number of source space points generated with    the <tt class="docutils literal"><span class="pre">--spacing</span></tt> option may be different between the current    version of MNE and versions 2.5 or earlier (using <tt class="docutils literal"><span class="pre">--spacing</span></tt> option    to mne_setup_source_space ) if    the FreeSurfer surfaces employ the (old) quadrangle format or if    there are topological defects on the surfaces. All new FreeSurfer    surfaces are specified as triangular tessellations and are e of    defects.</p>
</div>
</div>
<div class="section" id="mne-show-environment">
<span id="id19"></span><h3><a class="toc-backref" href="#id125">mne_show_environment</a><a class="headerlink" href="#mne-show-environment" title="Permalink to this headline">¶</a></h3>
<p>Usage: <tt class="docutils literal"><span class="pre">mne_show_environment</span> <span class="pre">files</span></tt></p>
</div>
</div>
<div class="section" id="utility-command-line-arguments">
<h2><a class="toc-backref" href="#id126">Utility command-line arguments</a><a class="headerlink" href="#utility-command-line-arguments" title="Permalink to this headline">¶</a></h2>
<div class="section" id="mne-add-patch-info">
<span id="id20"></span><h3><a class="toc-backref" href="#id127">mne_add_patch_info</a><a class="headerlink" href="#mne-add-patch-info" title="Permalink to this headline">¶</a></h3>
<div class="section" id="purpose">
<h4>Purpose<a class="headerlink" href="#purpose" title="Permalink to this headline">¶</a></h4>
<p>The utility mne_add_patch_info uses
the detailed cortical surface geometry information to add data about
cortical patches corresponding to each source space point. A new
copy of the source space(s) included in the input file is created
with the patch information included. In addition to the patch information, mne_add_patch_info can
optionally calculate distances, along the cortical surface, between
the vertices selected to the source space.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Depending on the speed of your computer and the options selected, mne_add_patch_info takes 5 - 30 minutes to run.</p>
</div>
</div>
<div class="section" id="command-line-options">
<span id="cjagcdcc"></span><h4>Command line options<a class="headerlink" href="#command-line-options" title="Permalink to this headline">¶</a></h4>
<p>mne_add_patch_info accepts
the following command-line options:</p>
<p><tt class="docutils literal"><span class="pre">--verbose</span></tt></p>
<blockquote>
<div>Provide verbose output during the calculations.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--dist</span>&nbsp; <span class="pre">&lt;*dist/mm*&gt;</span></tt></p>
<blockquote>
<div>Invokes the calculation of distances between vertices included in
the source space along the cortical surface. Only pairs whose distance in
the three-dimensional volume is less than the specified distance are
considered. For details, see <a class="reference internal" href="#cjaifjdd"><em>Computational details</em></a>, below.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--src</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>The input source space file. The source space files usually end
with <tt class="docutils literal"><span class="pre">-src.fif</span></tt> .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--srcp</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>The output source space file which will contain the patch information.
If the file exists it will overwritten without asking for permission.
A recommended naming convention is to add the letter <tt class="docutils literal"><span class="pre">p</span></tt> after the
source spacing included in the file name. For example, if the input
file is <tt class="docutils literal"><span class="pre">mh-7-src.fif</span></tt> , a recommended output file name
is <tt class="docutils literal"><span class="pre">mh-7p-src.fif</span></tt> .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--w</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Name of a w file, which will contain the patch area information. Two
files will be created:  &lt;<em>name</em>&gt; <tt class="docutils literal"><span class="pre">-lh.w</span></tt> and  &lt;<em>name</em>&gt; <tt class="docutils literal"><span class="pre">-rh.w</span></tt> .
The numbers in the files are patch areas in <span class="math">\(\text{mm}^2\)</span>.
The source space vertices are marked with value 150.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--labeldir</span>&nbsp; <span class="pre">&lt;*directory*&gt;</span></tt></p>
<blockquote>
<div>Create a label file corresponding to each of the patches in the
given directory. The directory must be created before running mne_add_patch_info .</div></blockquote>
</div>
<div class="section" id="computational-details">
<span id="cjaifjdd"></span><h4>Computational details<a class="headerlink" href="#computational-details" title="Permalink to this headline">¶</a></h4>
<p>By default, mne_add_patch_info creates
a copy of the source space(s) with the following additional information
for each vertex in the original dense triangulation of the cortex:</p>
<ul class="simple">
<li>The number of the closest active source
space vertex and</li>
<li>The distance to this vertex.</li>
</ul>
<p>This information can be used to determine, <em>e.g.</em>,
the sizes of the patches, their average normals, and the standard
deviation of the normal directions. This information is also returned
by the mne_read_source_space Matlab function as described in Table 10.28.</p>
<p>The <tt class="docutils literal"><span class="pre">--dist</span></tt> option to mne_add_patch_info invokes
the calculation of inter-vertex distances. These distances are computed
along the the cortical surface (usually the white matter) on which
the source space vertices are located.</p>
<p>Since the calculation of all possible distances would take
a very long time, the distance given with the <tt class="docutils literal"><span class="pre">--dist</span></tt> option allows
restriction to the neighborhood of each source space vertex. This
neighborhood is defined as the sphere around each source space vertex,
with radius given by the <tt class="docutils literal"><span class="pre">--dist</span></tt> option. Because the distance calculation
is done along the folded cortical surface whose details are given
by the dense triangulation of the cortical surface produced by FreeSurfer,
some of the distances computed will be larger than the value give
with &#8211;dist.</p>
</div>
</div>
<div class="section" id="mne-add-to-meas-info">
<span id="id21"></span><h3><a class="toc-backref" href="#id128">mne_add_to_meas_info</a><a class="headerlink" href="#mne-add-to-meas-info" title="Permalink to this headline">¶</a></h3>
<p>Add new data to meas info.</p>
<p><tt class="docutils literal"><span class="pre">--add</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>The file to add.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--dest</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>the destination file.</div></blockquote>
</div>
<div class="section" id="mne-add-triggers">
<span id="id22"></span><h3><a class="toc-backref" href="#id129">mne_add_triggers</a><a class="headerlink" href="#mne-add-triggers" title="Permalink to this headline">¶</a></h3>
<div class="section" id="id23">
<h4>Purpose<a class="headerlink" href="#id23" title="Permalink to this headline">¶</a></h4>
<p>The utility mne_add_triggers modifies
the digital trigger channel (STI 014) in raw data files
to include additional transitions. Since the raw data file is modified,
it is possible to make irreversible changes. Use this utility with
caution. It is recommended that you never run mne_add_triggers on
an original raw data file.</p>
</div>
<div class="section" id="id24">
<h4>Command line options<a class="headerlink" href="#id24" title="Permalink to this headline">¶</a></h4>
<p>mne_add_triggers accepts
the following command-line options:</p>
<p><tt class="docutils literal"><span class="pre">--raw</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the raw data file to be modified.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--trg</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the trigger line modification list. This text file should
contain two entries per line: the sample number and the trigger
number to be added into the file. The number of the first sample
in the file is zero. It is recommended that trigger numbers whose
binary equivalent has lower eight bits equal to zero are used to
avoid conflicts with the ordinary triggers occurring in the file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--delete</span></tt></p>
<blockquote>
<div>Delete the triggers defined by the trigger file instead of adding
them. This enables changing the file to its original state, provided
that the trigger file is preserved.</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Since <a class="reference internal" href="#mne-browse-raw"><em>mne_browse_raw</em></a> and <a class="reference internal" href="#mne-process-raw"><em>mne_process_raw</em></a> can employ an event file which effectively adds new trigger instants, mne_add_triggers is    for the most part obsolete but it has been retained in the MNE software    suite for backward compatibility.</p>
</div>
</div>
</div>
<div class="section" id="mne-annot2labels">
<span id="id25"></span><h3><a class="toc-backref" href="#id130">mne_annot2labels</a><a class="headerlink" href="#mne-annot2labels" title="Permalink to this headline">¶</a></h3>
<p>The utility mne_annot2labels converts
cortical parcellation data into a set of labels. The parcellation
data are read from the directory <tt class="docutils literal"><span class="pre">$SUBJECTS_DIR/$SUBJECT/label</span></tt> and
the resulting labels are written to the current directory. mne_annot2labels requires
that the environment variable <tt class="docutils literal"><span class="pre">$SUBJECTS_DIR</span></tt> is set.
The command line options for mne_annot2labels are:</p>
<p><tt class="docutils literal"><span class="pre">--subject</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of the subject. If this option is not present
the <tt class="docutils literal"><span class="pre">$SUBJECT</span></tt> environment variable is consulted. If
the subject name cannot be determined, the program quits.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--parc</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the parcellation name to convert. The corresponding parcellation
file names will be <tt class="docutils literal"><span class="pre">$SUBJECTS_DIR/$SUBJECT/label/</span></tt>  &lt;<em>hemi</em>&gt; <tt class="docutils literal"><span class="pre">h.</span></tt>  &lt;<em>name</em>&gt; <tt class="docutils literal"><span class="pre">.annot</span></tt> where  &lt;<em>hemi</em>&gt; is <tt class="docutils literal"><span class="pre">l</span></tt> or <tt class="docutils literal"><span class="pre">r</span></tt> for the
left and right hemisphere, respectively.</div></blockquote>
</div>
<div class="section" id="mne-anonymize">
<span id="id26"></span><h3><a class="toc-backref" href="#id131">mne_anonymize</a><a class="headerlink" href="#mne-anonymize" title="Permalink to this headline">¶</a></h3>
<p>Depending no the settings during acquisition in the Elekta-Neuromag EEG/MEG
systems the data files may contain subject identifying information
in unencrypted form. The utility mne_anonymize was
written to clear tags containing such information from a fif file.
Specifically, this utility removes the following tags from the fif
file:</p>
<table border="1" class="docutils" id="chdehbcg">
<caption>Tags cleared by mne_anonymize .</caption>
<colgroup>
<col width="34%" />
<col width="66%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Tag</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>FIFF_SUBJ_FIRST_NAME</td>
<td>First name of the subject</td>
</tr>
<tr class="row-odd"><td>FIFF_SUBJ_MIDDLE_NAME</td>
<td>Middle name of the subject</td>
</tr>
<tr class="row-even"><td>FIFF_SUBJ_LAST_NAME</td>
<td>Last name of the subject</td>
</tr>
<tr class="row-odd"><td>FIFF_SUBJ_BIRTH_DAY</td>
<td>Birthday of the subject (Julian day number)</td>
</tr>
<tr class="row-even"><td>FIFF_SUBJ_SEX</td>
<td>The sex of the subject</td>
</tr>
<tr class="row-odd"><td>FIFF_SUBJ_HAND</td>
<td>Handedness of the subject</td>
</tr>
<tr class="row-even"><td>FIFF_SUBJ_WEIGHT</td>
<td>Weight of the subject in kg</td>
</tr>
<tr class="row-odd"><td>FIFF_SUBJ_HEIGHT</td>
<td>Height of the subject in m</td>
</tr>
<tr class="row-even"><td>FIFF_SUBJ_COMMENT</td>
<td>Comment about the subject</td>
</tr>
</tbody>
</table>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">mne_anonymize normally    keeps the FIFF_SUBJ_HIS_ID tag which can be used to identify the    subjects uniquely after the information listed in <a class="reference internal" href="#chdehbcg"><em>Tags cleared by mne_anonymize .</em></a> have    been removed. If the <tt class="docutils literal"><span class="pre">--his</span></tt> option is specified on the command line,    the FIFF_SUBJ_HIS_ID tag will be removed as well. The data of the    tags listed in <a class="reference internal" href="#chdehbcg"><em>Tags cleared by mne_anonymize .</em></a> and the optional FIFF_SUBJ_HIS_ID    tag are overwritten with zeros and the space claimed by omitting    these tags is added to the free space list of the file. Therefore, after mne_anonymize has    processed a data file there is no way to recover the removed information.    Use this utility with caution.</p>
</div>
<p>mne_anonymize recognizes
the following command-line options:</p>
<p><tt class="docutils literal"><span class="pre">--his</span></tt></p>
<blockquote>
<div>Remove the FIFF_SUBJ_HIS_ID tag as well, see above.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--file</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of the file to be modified.</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">You need write permission to the file to be    processed.</p>
</div>
</div>
<div class="section" id="mne-average-forward-solutions">
<span id="id27"></span><h3><a class="toc-backref" href="#id132">mne_average_forward_solutions</a><a class="headerlink" href="#mne-average-forward-solutions" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--fwd</span> <span class="pre">&lt;*name*&gt;</span> <span class="pre">:[</span> <span class="pre">&lt;*weight*&gt;</span> <span class="pre">]</span></tt></p>
<blockquote>
<div>Specifies a forward solution to include. If no weight is specified,
1.0 is assumed. In the averaging process the weights are divided
by their sum. For example, if two forward solutions are averaged
and their specified weights are 2 and 3, the average is formed with
a weight of 2/5 for the first solution and 3/5 for the second one.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--out</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the output file which will contain the averaged forward solution.</div></blockquote>
</div>
<div class="section" id="mne-brain-vision2fiff">
<span id="id28"></span><h3><a class="toc-backref" href="#id133">mne_brain_vision2fiff</a><a class="headerlink" href="#mne-brain-vision2fiff" title="Permalink to this headline">¶</a></h3>
<p>The utility mne_brain_vision2fiff was
created to import BrainVision EEG data. This utility also helps
to import the eXimia (Nexstim) TMS-compatible EEG system data to
the MNE software. The utility uses an optional fif file containing
the head digitization data to allow source modeling. The MNE Matlab
toolbox contains the function fiff_write_dig_file to
write a digitization file based on digitization data available in
another format, see <a class="reference internal" href="matlab.html#ch-matlab"><em>Matlab toolbox</em></a>.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">mne_brain_vision2fiff reads events from the <tt class="docutils literal"><span class="pre">vmrk</span></tt> file referenced in the
<tt class="docutils literal"><span class="pre">vhdr</span></tt> file, but it only includes events whose &#8220;Type&#8221; is <tt class="docutils literal"><span class="pre">Stimulus</span></tt> and
whose &#8220;description&#8221; is given by <tt class="docutils literal"><span class="pre">S&lt;number&gt;</span></tt>. All other events are ignored.</p>
</div>
<p>The command-line options of mne_brain_vision2fiff are:</p>
<p><tt class="docutils literal"><span class="pre">--header</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>The name of the BrainVision header file. The extension of this file
is <tt class="docutils literal"><span class="pre">vhdr</span></tt> . The header file typically refers to a marker
file (<tt class="docutils literal"><span class="pre">vmrk</span></tt> ) which is automatically processed and a
digital trigger channel (STI 014) is formed from the marker information.
The <tt class="docutils literal"><span class="pre">vmrk</span></tt> file is ignored if the <tt class="docutils literal"><span class="pre">--eximia</span></tt> option
is present.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--dig</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>The name of the fif file containing the digitization data.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--orignames</span></tt></p>
<blockquote>
<div>Use the original EEG channel labels. If this option is absent the EEG
channels will be automatically renamed to EEG 001, EEG 002, <em>etc.</em></div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eximia</span></tt></p>
<blockquote>
<div>Interpret this as an eXimia data file. The first three channels
will be thresholded and interpreted as trigger channels. The composite
digital trigger channel will be composed in the same way as in the
<a class="reference internal" href="#mne-kit2fiff"><em>mne_kit2fiff</em></a> utility. In addition, the fourth channel
will be assigned as an EOG channel. This option is normally used
by the <a class="reference internal" href="#mne-eximia2fiff"><em>mne_eximia2fiff</em></a> script.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--split</span> <span class="pre">&lt;*size/MB*&gt;</span></tt></p>
<blockquote>
<div>Split the output data into several files which are no more than &lt;<em>size</em>&gt; MB.
By default, the output is split into files which are just below
2 GB so that the fif file maximum size is not exceeded.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--out</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of the output fif format data file. If &lt;<em>filename</em>&gt; ends
with <tt class="docutils literal"><span class="pre">.fif</span></tt> or <tt class="docutils literal"><span class="pre">_raw.fif</span></tt> , these endings are
deleted. After these modifications, <tt class="docutils literal"><span class="pre">_raw.fif</span></tt> is inserted
after the remaining part of the file name. If the file is split
into multiple parts, the additional parts will be called
&lt;<em>name</em>&gt; <tt class="docutils literal"><span class="pre">-</span></tt> &lt;<em>number</em>&gt; <tt class="docutils literal"><span class="pre">_raw.fif</span></tt> .</div></blockquote>
</div>
<div class="section" id="mne-change-baselines">
<span id="id29"></span><h3><a class="toc-backref" href="#id134">mne_change_baselines</a><a class="headerlink" href="#mne-change-baselines" title="Permalink to this headline">¶</a></h3>
<p>The utility mne_change_baselines computes
baseline values and applies them to an evoked-response data file.
The command-line options are:</p>
<p><tt class="docutils literal"><span class="pre">--in</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the input data file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--set</span>&nbsp; <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>The data set number to compute baselines from or to apply baselines
to. If this option is omitted, all average data sets in the input file
are processed.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--out</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>The output file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--baselines</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a text file which contains the baseline values to be applied. Each
line should contain a channel name, colon, and the baseline value
given in &#8216;native&#8217; units (T/m, T, or V). If this
option is encountered, the limits specified by previous <tt class="docutils literal"><span class="pre">--bmin</span></tt> and <tt class="docutils literal"><span class="pre">--bmax</span></tt> options will not
have an effect.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--list</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a text file to contain the baseline values. Listing is
provided only if a specific data set is selected with the <tt class="docutils literal"><span class="pre">--set</span></tt> option.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--bmin</span>&nbsp; <span class="pre">&lt;*value/ms*&gt;</span></tt></p>
<blockquote>
<div>Lower limit of the baseline. Effective only if <tt class="docutils literal"><span class="pre">--baselines</span></tt> option is
not present. Both <tt class="docutils literal"><span class="pre">--bmin</span></tt> and <tt class="docutils literal"><span class="pre">--bmax</span></tt> must
be present to compute the baseline values. If either <tt class="docutils literal"><span class="pre">--bmin</span></tt> or <tt class="docutils literal"><span class="pre">--bmax</span></tt> is
encountered, previous <tt class="docutils literal"><span class="pre">--baselines</span></tt> option will be ignored.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--bmax</span>&nbsp; <span class="pre">&lt;*value/ms*&gt;</span></tt></p>
<blockquote>
<div>Upper limit of the baseline.</div></blockquote>
</div>
<div class="section" id="mne-change-nave">
<span id="id30"></span><h3><a class="toc-backref" href="#id135">mne_change_nave</a><a class="headerlink" href="#mne-change-nave" title="Permalink to this headline">¶</a></h3>
<p>Usage: <tt class="docutils literal"><span class="pre">mne_change_nave</span> <span class="pre">--nave</span> <span class="pre">&lt;number&gt;</span> <span class="pre">&lt;meas</span> <span class="pre">file&gt;</span> <span class="pre">...</span></tt></p>
</div>
<div class="section" id="mne-check-eeg-locations">
<span id="id31"></span><h3><a class="toc-backref" href="#id136">mne_check_eeg_locations</a><a class="headerlink" href="#mne-check-eeg-locations" title="Permalink to this headline">¶</a></h3>
<p>Some versions of the Neuromag acquisition software did not
copy the EEG channel location information properly from the Polhemus
digitizer information data block to the EEG channel information
records if the number of EEG channels exceeds 60. The purpose of mne_check_eeg_locations is
to detect this problem and fix it, if requested. The command-line
options are:</p>
<p><tt class="docutils literal"><span class="pre">--file</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specify the measurement data file to be checked or modified.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--dig</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Name of the file containing the Polhemus digitizer information. Default
is the data file name.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fix</span></tt></p>
<blockquote>
<div>By default mne_check_eeg_locations only
checks for missing EEG locations (locations close to the origin).
With &#8211;fix mne_check_eeg_locations reads
the Polhemus data from the specified file and copies the EEG electrode
location information to the channel information records in the measurement
file. There is no harm running mne_check_eeg_locations on
a data file even if the EEG channel locations were correct in the
first place.</div></blockquote>
</div>
<div class="section" id="mne-check-surface">
<span id="id32"></span><h3><a class="toc-backref" href="#id137">mne_check_surface</a><a class="headerlink" href="#mne-check-surface" title="Permalink to this headline">¶</a></h3>
<p>This program just reads a surface file to check whether it is valid.</p>
<p><tt class="docutils literal"><span class="pre">--surf</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>The input file (FreeSurfer surface format).</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--bem</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>The input file (a BEM fif file)</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--id</span> <span class="pre">&lt;id&gt;</span></tt></p>
<blockquote>
<div><p>Surface id to list (default : 4)</p>
<blockquote>
<div><ul class="simple">
<li>4 for outer skin (scalp) surface
* 3 for outer skull surface
* 1 for inner skull surface</li>
</ul>
</div></blockquote>
</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--checkmore</span></tt></p>
<blockquote>
<div>Do more thorough testing</div></blockquote>
</div>
<div class="section" id="mne-collect-transforms">
<span id="id33"></span><h3><a class="toc-backref" href="#id138">mne_collect_transforms</a><a class="headerlink" href="#mne-collect-transforms" title="Permalink to this headline">¶</a></h3>
<p>The utility mne_collect_transforms collects
coordinate transform information from various sources and saves
them into a single fif file. The coordinate transformations used
by MNE software are summarized in Figure 5.1. The output
of mne_collect_transforms may
include all transforms referred to therein except for the sensor
coordinate system transformations <span class="math">\(T_{s_1} \dotso T_{s_n}\)</span>.
The command-line options are:</p>
<p><tt class="docutils literal"><span class="pre">--meas</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a measurement data file which provides <span class="math">\(T_1\)</span>.
A forward solution or an inverse operator file can also be specified
as implied by Table 5.1.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--mri</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies an MRI description or a standalone coordinate transformation
file produced by mne_analyze which
provides <span class="math">\(T_2\)</span>. If the <tt class="docutils literal"><span class="pre">--mgh</span></tt> option
is not present mne_collect_transforms also
tries to find <span class="math">\(T_3\)</span>, <span class="math">\(T_4\)</span>, <span class="math">\(T_-\)</span>,
and <span class="math">\(T_+\)</span> from this file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--mgh</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>An MRI volume volume file in mgh or mgz format.
This file provides <span class="math">\(T_3\)</span>. The transformation <span class="math">\(T_4\)</span> will
be read from the talairach.xfm file referred to in the MRI volume.
The fixed transforms <span class="math">\(T_-\)</span> and <span class="math">\(T_+\)</span> will
also be created.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--out</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the output file. If this option is not present, the collected transformations
will be output on screen but not saved.</div></blockquote>
</div>
<div class="section" id="mne-compensate-data">
<span id="id34"></span><h3><a class="toc-backref" href="#id139">mne_compensate_data</a><a class="headerlink" href="#mne-compensate-data" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--in</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the input data file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--out</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the output data file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--grad</span> <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>Specifies the desired compensation grade in the output file. The value
can be 1, 2, 3, or 101. The values starting from 101 will be used
for 4D Magnes compensation matrices.</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Only average data is included in the output. Evoked-response data files produced with mne_browse_raw or mne_process_raw may    include standard errors of mean, which can not be re-compensated    using the above method and are thus omitted.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Raw data cannot be compensated using mne_compensate_data . For this purpose, load the data to mne_browse_raw or mne_process_raw , specify    the desired compensation grade, and save a new raw data file.</p>
</div>
</div>
<div class="section" id="mne-copy-processing-history">
<span id="id35"></span><h3><a class="toc-backref" href="#id140">mne_copy_processing_history</a><a class="headerlink" href="#mne-copy-processing-history" title="Permalink to this headline">¶</a></h3>
<p>In order for the inverse operator calculation to work correctly
with data processed with the Elekta-Neuromag Maxfilter (TM) software,
the so-called <em>processing history</em> block must
be included in data files. Previous versions of the MNE Matlab functions
did not copy processing history to files saved. As of March 30,
2009, the Matlab toolbox routines fiff_start_writing_raw and fiff_write_evoked have
been enchanced to include these data to the output file as appropriate.
If you have older raw data files created in Matlab from input which
has been processed Maxfilter, it is necessary to copy the <em>processing
history</em> block from the original to modified raw data
file using the mne_copy_processing_history utility described
below. The raw data processing programs mne_browse_raw and mne_process_raw have
handled copying of the processing history since revision 2.5 of
the MNE software.</p>
<p>mne_copy_processing_history is
simple to use:</p>
<p><tt class="docutils literal"><span class="pre">mne_copy_processing_history</span> <span class="pre">--from</span></tt>  &lt;<em>from</em>&gt; <tt class="docutils literal"><span class="pre">--to</span></tt>  &lt;<em>to</em>&gt; ,</p>
<p>where  &lt;<em>from</em>&gt; is an
original raw data file containing the processing history and  &lt;<em>to</em>&gt; is
a file output with older MNE Matlab routines. Be careful: this operation
cannot be undone. If the  &lt;<em>from</em>&gt; file
does not have the processing history block or the  &lt;<em>to</em>&gt; file
already has it, the destination file remains unchanged.</p>
</div>
<div class="section" id="mne-convert-dig-data">
<span id="id36"></span><h3><a class="toc-backref" href="#id141">mne_convert_dig_data</a><a class="headerlink" href="#mne-convert-dig-data" title="Permalink to this headline">¶</a></h3>
<p>Converts Polhemus digitization data between different file formats.
The input formats are:</p>
<p><tt class="docutils literal"><span class="pre">fif</span></tt></p>
<blockquote>
<div>The
standard format used in MNE. The digitization data are typically
present in the measurement files.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">hpts</span></tt></p>
<blockquote>
<div>A text format which is a translation
of the fif format data, see <a class="reference internal" href="#cjadjebh"><em>The hpts format</em></a> below.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">elp</span></tt></p>
<blockquote>
<div>A text format produced by the <em>Source
Signal Imaging, Inc.</em> software. For description of this &#8220;probe&#8221; format,
see <a class="reference external" href="http://www.sourcesignal.com/formats_probe.html">http://www.sourcesignal.com/formats_probe.html</a>.</div></blockquote>
<p>The data can be output in fif and hpts formats.
Only the last command-line option specifying an input file will
be honored. Zero or more output file options can be present on the
command line.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The elp and hpts input    files may contain textual EEG electrode labels. They will not be    copied to the fif format output.</p>
</div>
<p>The command-line options of mne_convert_dig_data are:</p>
<p><tt class="docutils literal"><span class="pre">--fif</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of an input fif file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--hpts</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of an input hpts file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--elp</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of an input elp file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fifout</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of an output fif file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--hptsout</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of an output hpts file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--headcoord</span></tt></p>
<blockquote>
<div>The fif and hpts input
files are assumed to contain data in the  MNE head coordinate system,
see <a class="reference internal" href="source_localization/forward.html#bjebibai"><em>The head and device coordinate systems</em></a>. With this option present, the data are
transformed to the MNE head coordinate system with help of the fiducial
locations in the data. Use this option if this is not the case or
if you are unsure about the definition of the coordinate system
of the fif and hpts input
data. This option is implied with elp input
files. If this option is present, the fif format output file will contain
the transformation between the original digitizer data coordinates
the MNE head coordinate system.</div></blockquote>
<div class="section" id="the-hpts-format">
<span id="cjadjebh"></span><h4>The hpts format<a class="headerlink" href="#the-hpts-format" title="Permalink to this headline">¶</a></h4>
<p>The hpts format digitzer
data file may contain comment lines starting with the pound sign
(#) and data lines of the form:</p>
<blockquote>
<div>&lt;<em>category</em>&gt; &lt;<em>identifier</em>&gt; &lt;<em>x/mm</em>&gt; &lt;<em>y/mm</em>&gt; &lt;<em>z/mm</em>&gt;</div></blockquote>
<p>where</p>
<p>`` &lt;<em>category</em>&gt;``</p>
<blockquote>
<div>defines the type of points. Allowed categories are: hpi , cardinal (fiducial ),eeg ,
and extra corresponding to head-position
indicator coil locations, cardinal landmarks, EEG electrode locations,
and additional head surface points, respectively. Note that tkmedit does not
recognize the fiducial as an
alias for cardinal .</div></blockquote>
<p>`` &lt;<em>identifier</em>&gt;``</p>
<blockquote>
<div>identifies the point. The identifiers are usually sequential numbers. For
cardinal landmarks, 1 = left auricular point, 2 = nasion, and 3
= right auricular point. For EEG electrodes, identifier = 0 signifies
the reference electrode. Some programs (not tkmedit )
accept electrode labels as identifiers in the eeg category.</div></blockquote>
<p>`` &lt;<em>x/mm</em>&gt; , &lt;<em>y/mm</em>&gt; , &lt;<em>z/mm</em>&gt;``</p>
<blockquote>
<div>Location of the point, usually in the MEG head coordinate system, see <a class="reference internal" href="source_localization/forward.html#bjebibai"><em>The head and device coordinate systems</em></a>.
Some programs have options to accept coordinates in meters instead
of millimeters. With <tt class="docutils literal"><span class="pre">--meters</span></tt> option, mne_transform_points lists
the coordinates in meters.</div></blockquote>
</div>
</div>
<div class="section" id="mne-convert-lspcov">
<span id="id37"></span><h3><a class="toc-backref" href="#id142">mne_convert_lspcov</a><a class="headerlink" href="#mne-convert-lspcov" title="Permalink to this headline">¶</a></h3>
<p>The utility mne_convert_lspcov converts a LISP-format noise-covariance file,
produced by the Neuromag signal processor, graph into fif format.</p>
<p>The command-line options are:</p>
<p><tt class="docutils literal"><span class="pre">--lspcov</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>The LISP noise-covariance matrix file to be converted.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--meas</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>A fif format measurement file used to assign channel names to the noise-covariance
matrix elements. This file should have precisely the same channel
order within MEG and EEG as the LISP-format covariance matrix file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--out</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>The name of a fif format output file. The file name should end with
-cov.fif.text format output file. No information about the channel names
is included. The covariance matrix file is listed row by row. This
file can be loaded to MATLAB, for example</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--outasc</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>The name of a text format output file. No information about the channel
names is included. The covariance matrix file is listed row by row.
This file can be loaded to MATLAB, for example</div></blockquote>
</div>
<div class="section" id="mne-convert-ncov">
<span id="id38"></span><h3><a class="toc-backref" href="#id143">mne_convert_ncov</a><a class="headerlink" href="#mne-convert-ncov" title="Permalink to this headline">¶</a></h3>
<p>The ncov file format was used to store the noise-covariance
matrix file. The MNE software requires that the covariance matrix
files are in fif format. The utility mne_convert_ncov converts
ncov files to fif format.</p>
<p>The command-line options are:</p>
<p><tt class="docutils literal"><span class="pre">--ncov</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>The ncov file to be converted.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--meas</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>A fif format measurement file used to assign channel names to the noise-covariance
matrix elements. This file should have precisely the same channel
order within MEG and EEG as the ncov file. Typically, both the ncov
file and the measurement file are created by the now mature off-line
averager, meg_average.</div></blockquote>
</div>
<div class="section" id="mne-convert-surface">
<span id="id39"></span><h3><a class="toc-backref" href="#id144">mne_convert_surface</a><a class="headerlink" href="#mne-convert-surface" title="Permalink to this headline">¶</a></h3>
<p>The utility mne_convert_surface converts
surface data files between different formats.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The MNE Matlab toolbox functions enable    reading of FreeSurfer surface files directly. Therefore, the <tt class="docutils literal"><span class="pre">--mat</span></tt>   option has been removed. The dfs file format conversion functionality    has been moved here from mne_convert_dfs .    Consequently, mne_convert_dfs has    been removed from MNE software.</p>
</div>
<div class="section" id="babeabaa">
<span id="id40"></span><h4>command-line options<a class="headerlink" href="#babeabaa" title="Permalink to this headline">¶</a></h4>
<p>mne_convert_surface accepts
the following command-line options:</p>
<p><tt class="docutils literal"><span class="pre">--fif</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a fif format input file. The first surface (source space)
from this file will be read.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--tri</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a text format input file. The format of this file is described in <a class="reference internal" href="source_localization/forward.html#behdefcd"><em>Tessellation file format</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--meters</span></tt></p>
<blockquote>
<div>The unit of measure for the vertex locations in a text input files
is meters instead of the default millimeters. This option does not
have any effect on the interpretation of the FreeSurfer surface
files specified with the <tt class="docutils literal"><span class="pre">--surf</span></tt> option.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--swap</span></tt></p>
<blockquote>
<div>Swap the ordering or the triangle vertices. The standard convention in
the MNE software is to have the vertices in text format files ordered
so that the vector cross product of the vectors from vertex 1 to
2 and 1 to 3 gives the direction of the outward surface normal. This
is also called the counterclockwise ordering. If your text input file
does not comply with this right-hand rule, use the <tt class="docutils literal"><span class="pre">--swap</span></tt> option.
This option does not have any effect on the interpretation of the FreeSurfer surface
files specified with the <tt class="docutils literal"><span class="pre">--surf</span></tt> option.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--surf</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a FreeSurfer format
input file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--dfs</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of a dfs file to be converted. The surfaces produced
by BrainSuite are in the dfs format.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--mghmri</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a mgh/mgz format MRI data file which will be used to define
the coordinate transformation to be applied to the data read from
a dfs file to bring it to the FreeSurfer MRI
coordinates, <em>i.e.</em>, the coordinate system of
the MRI stack in the file. In addition, this option can be used
to insert &#8220;volume geometry&#8221; information to the FreeSurfer
surface file output (<tt class="docutils literal"><span class="pre">--surfout</span></tt> option). If the input file already
contains the volume geometry information, &#8211;replacegeom is needed
to override the input volume geometry and to proceed to writing
the data.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--replacegeom</span></tt></p>
<blockquote>
<div>Replaces existing volume geometry information. Used in conjunction
with the <tt class="docutils literal"><span class="pre">--mghmri</span></tt> option described above.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fifmri</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a fif format MRI destription file which will be used to define
the coordinate transformation to be applied to the data read from
a dfs file to bring it to the same coordinate system as the MRI stack
in the file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--trans</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of a text file which contains the coordinate
transformation to be applied to the data read from the dfs file
to bring it to the MRI coordinates, see below. This option is rarely
needed.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--flip</span></tt></p>
<blockquote>
<div>By default, the dfs surface nodes are assumed to be in a right-anterior-superior
(RAS) coordinate system with its origin at the left-posterior-inferior
(LPI) corner of the MRI stack. Sometimes the dfs file has left and
right flipped. This option reverses this flip, <em>i.e.</em>,
assumes the surface coordinate system is left-anterior-superior
(LAS) with its origin in the right-posterior-inferior (RPI) corner
of the MRI stack.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--shift</span> <span class="pre">&lt;*value/mm*&gt;</span></tt></p>
<blockquote>
<div>Shift the surface vertices to the direction of the surface normals
by this amount before saving the surface.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--surfout</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a FreeSurfer format output file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fifout</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a fif format output file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--triout</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies an ASCII output file that will contain the surface data
in the triangle file format desribed in <a class="reference internal" href="source_localization/forward.html#behdefcd"><em>Tessellation file format</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--pntout</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a ASCII output file which will contain the vertex numbers only.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--metersout</span></tt></p>
<blockquote>
<div>With this option the ASCII output will list the vertex coordinates
in meters instead of millimeters.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--swapout</span></tt></p>
<blockquote>
<div>Defines the vertex ordering of ASCII triangle files to be output.
For details, see <tt class="docutils literal"><span class="pre">--swap</span></tt> option, above.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--smfout</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a smf (Simple Model Format) output file. For details of this
format, see <a class="reference external" href="http://people.sc.fsu.edu/~jburkardt/data/smf/smf.txt">http://people.sc.fsu.edu/~jburkardt/data/smf/smf.txt</a>.</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Multiple output options can be specified to    produce outputs in several different formats with a single invocation    of mne_convert_surface .</p>
</div>
<p>The coordinate transformation file specified with the <tt class="docutils literal"><span class="pre">--trans</span></tt> should contain
a 4 x 4 coordinate transformation matrix:</p>
<div class="math">
\[\begin{split}T = \begin{bmatrix}
R_{11} &amp; R_{12} &amp; R_{13} &amp; x_0 \\
R_{13} &amp; R_{13} &amp; R_{13} &amp; y_0 \\
R_{13} &amp; R_{13} &amp; R_{13} &amp; z_0 \\
0 &amp; 0 &amp; 0 &amp; 1
\end{bmatrix}\end{split}\]</div>
<p>defined so that if the augmented location vectors in the
dfs file and MRI coordinate systems are denoted by <span class="math">\(r_{dfs} = [x_{dfs} y_{dfs} z_{dfs} 1]^T\)</span> and <span class="math">\(r_{MRI} = [x_{MRI} y_{MRI} z_{MRI} 1]^T\)</span>,
respectively,</p>
<div class="math">
\[r_{MRI} = Tr_{dfs}\]</div>
</div>
</div>
<div class="section" id="mne-cov2proj">
<span id="id41"></span><h3><a class="toc-backref" href="#id145">mne_cov2proj</a><a class="headerlink" href="#mne-cov2proj" title="Permalink to this headline">¶</a></h3>
<div class="section" id="id42">
<h4>Purpose<a class="headerlink" href="#id42" title="Permalink to this headline">¶</a></h4>
<p>The utility mne_cov2proj picks
eigenvectors from a covariance matrix and outputs them as a signal-space
projection (SSP) file.</p>
</div>
<div class="section" id="id43">
<h4>Command line options<a class="headerlink" href="#id43" title="Permalink to this headline">¶</a></h4>
<p>mne_cov2proj accepts the
following command-line options:</p>
<p><tt class="docutils literal"><span class="pre">--cov</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>The covariance matrix file to be used a source. The covariance matrix
files usually end with <tt class="docutils literal"><span class="pre">-cov.fif</span></tt> .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--proj</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>The output file to contain the projection. It is recommended that
the file name ends with <tt class="docutils literal"><span class="pre">-proj.fif</span></tt> .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--bad</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specify channels not to be included when an eigenvalue decomposition
of the covariance matrix is computed.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--include</span>&nbsp; <span class="pre">&lt;*val1*&gt;</span> <span class="pre">[:</span> <span class="pre">&lt;*val2*&gt;</span> <span class="pre">]</span></tt></p>
<blockquote>
<div>Select an eigenvector or a range of eigenvectors to include. It
is recommended that magnetometers, gradiometers, and EEG data are handled
separately with help of the <tt class="docutils literal"><span class="pre">--bad</span></tt> , <tt class="docutils literal"><span class="pre">--meg</span></tt> , <tt class="docutils literal"><span class="pre">--megmag</span></tt> , <tt class="docutils literal"><span class="pre">--meggrad</span></tt> ,
and <tt class="docutils literal"><span class="pre">--eeg</span></tt> options.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--meg</span></tt></p>
<blockquote>
<div>After loading the covariance matrix, modify it so that only elements corresponding
to MEG channels are included.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eeg</span></tt></p>
<blockquote>
<div>After loading the covariance matrix, modify it so that only elements corresponding
to EEG channels are included.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--megmag</span></tt></p>
<blockquote>
<div>After loading the covariance matrix, modify it so that only elements corresponding
to MEG magnetometer channels are included.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--meggrad</span></tt></p>
<blockquote>
<div>After loading the covariance matrix, modify it so that only elements corresponding
to MEG planar gradiometer channels are included.</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The <tt class="docutils literal"><span class="pre">--megmag</span></tt> and <tt class="docutils literal"><span class="pre">--meggrad</span></tt> employ    the Vectorview channel numbering scheme to recognize MEG magnetometers    (channel names ending with &#8216;1&#8217;) and planar gradiometers    (other channels). Therefore, these options are only meaningful in    conjunction with data acquired with a Neuromag Vectorview system.</p>
</div>
</div>
</div>
<div class="section" id="mne-create-comp-data">
<span id="id44"></span><h3><a class="toc-backref" href="#id146">mne_create_comp_data</a><a class="headerlink" href="#mne-create-comp-data" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--in</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the input text file containing the compensation data.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--kind</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>The compensation type to be stored in the output file with the data. This
value defaults to 101 for the Magnes compensation and does not need
to be changed.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--out</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the output fif file containing the compensation channel weight
matrix <span class="math">\(C_{(k)}\)</span>, see <a class="reference internal" href="io.html#behddfbi"><em>Applying software gradient compensation</em></a>.</div></blockquote>
<p>The format of the text-format compensation data file is:</p>
<blockquote>
<div>&lt;<em>number of MEG helmet channels</em>&gt; &lt;<em>number of compensation channels included</em>&gt;
&lt;<em>cname_1</em>&gt; &lt;<em>cname_2</em>&gt; ...
&lt;<em>name_1</em>&gt; &lt;<em>weights</em>&gt;
&lt;<em>name_2</em>&gt; &lt;<em>weights</em>&gt; ...</div></blockquote>
<p>In the above &lt;<em>name_k</em>&gt; denote
names of MEG helmet channels and &lt;<em>cname_k</em>&gt;
those of the compensation channels, respectively. If the channel
names contain spaces, they must be surrounded by quotes, for example, <tt class="docutils literal"><span class="pre">&quot;MEG</span> <span class="pre">0111&quot;</span></tt> .</p>
</div>
<div class="section" id="mne-ctf2fiff">
<span id="id45"></span><h3><a class="toc-backref" href="#id147">mne_ctf2fiff</a><a class="headerlink" href="#mne-ctf2fiff" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--verbose</span></tt></p>
<blockquote>
<div>Produce a verbose listing of the conversion process to stdout.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--ds</span> <span class="pre">&lt;*directory*&gt;</span></tt></p>
<blockquote>
<div>Read the data from this directory</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--omit</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>Read the names of channels to be omitted from this text file. Enter one
channel name per line. The names should match exactly with those
listed in the CTF data structures. By default, all channels are included.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fif</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>The name of the output file. If the length of the raw data exceeds
the 2-GByte fif file limit, several output files will be produced.
These additional &#8216;extension&#8217; files will be tagged
with <tt class="docutils literal"><span class="pre">_001.fif</span></tt> , <tt class="docutils literal"><span class="pre">_002.fif</span></tt> , etc.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--evoked</span></tt></p>
<blockquote>
<div>Produce and evoked-response fif file instead of a raw data file.
Each trial in the CTF data file is included as a separate category
(condition). The maximum number of samples in each trial is limited
to 25000.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--infoonly</span></tt></p>
<blockquote>
<div>Write only the measurement info to the output file, do not include data.</div></blockquote>
<p>During conversion, the following files are consulted from
the ds directory:</p>
<p>`` &lt;<em>name</em>&gt; .res4``</p>
<blockquote>
<div>This file contains most of the header information pertaining the acquisition.</div></blockquote>
<p>`` &lt;<em>name</em>&gt; .hc``</p>
<blockquote>
<div>This file contains the HPI coil locations in sensor and head coordinates.</div></blockquote>
<p>`` &lt;<em>name</em>&gt; .meg4``</p>
<blockquote>
<div>This file contains the actual MEG data. If the data are split across several
files due to the 2-GByte file size restriction, the &#8216;extension&#8217; files
are called &lt;<em>name</em>&gt; <tt class="docutils literal"><span class="pre">.</span></tt> &lt;<em>number</em>&gt; <tt class="docutils literal"><span class="pre">_meg4</span></tt> .</div></blockquote>
<p>`` &lt;<em>name</em>&gt; .eeg``</p>
<blockquote>
<div>This is an optional input file containing the EEG electrode locations. More
details are given below.</div></blockquote>
<p>If the &lt;<em>name</em>&gt; <tt class="docutils literal"><span class="pre">.eeg</span></tt> file,
produced from the Polhemus data file with CTF software, is present,
it is assumed to contain lines with the format:</p>
<blockquote>
<div>&lt;<em>number</em>&gt; &lt;<em>name</em>&gt; &lt;<em>x/cm</em>&gt; &lt;<em>y/cm</em>&gt; &lt;<em>z/cm</em>&gt;</div></blockquote>
<p>The field &lt;<em>number</em>&gt; is
a sequential number to be assigned to the converted data point in
the fif file. &lt;<em>name</em>&gt; is either
a name of an EEG channel, one of <tt class="docutils literal"><span class="pre">left</span></tt> , <tt class="docutils literal"><span class="pre">right</span></tt> ,
or <tt class="docutils literal"><span class="pre">nasion</span></tt> to indicate a fiducial landmark, or any word
which is not a name of any channel in the data. If &lt;<em>name</em>&gt; is
a name of an EEG channel available in the data, the location is
included in the Polhemus data as an EEG electrode locations and
inserted as the location of the EEG electrode. If the name is one
of the fiducial landmark names, the point is included in the Polhemus
data as a fiducial landmark. Otherwise, the point is included as
an additional head surface points.</p>
<p>The standard <tt class="docutils literal"><span class="pre">eeg</span></tt> file produced by CTF software
does not contain the fiducial locations. If desired, they can be
manually copied from the <tt class="docutils literal"><span class="pre">pos</span></tt> file which was the source
of the <tt class="docutils literal"><span class="pre">eeg</span></tt> file.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">In newer CTF data the EEG position information maybe present in the <tt class="docutils literal"><span class="pre">res4</span></tt> file. If the <tt class="docutils literal"><span class="pre">eeg</span></tt> file    is present, the positions given there take precedence over the information in the <tt class="docutils literal"><span class="pre">res4</span></tt> file.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">mne_ctf2fiff converts both epoch mode and continuous raw data file into raw data fif files. It is not advisable to use epoch mode files with time gaps between the epochs because the data will be discontinuous in the resulting fif file with jumps at the junctions between epochs. These discontinuities    produce artefacts if the raw data is filtered in mne_browse_raw , mne_process_raw ,    or graph .</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The conversion process includes a transformation from the CTF head coordinate system convention to that used in the Neuromag systems.</p>
</div>
</div>
<div class="section" id="mne-ctf-dig2fiff">
<span id="id46"></span><h3><a class="toc-backref" href="#id148">mne_ctf_dig2fiff</a><a class="headerlink" href="#mne-ctf-dig2fiff" title="Permalink to this headline">¶</a></h3>
<p>The input data to mne_ctf_dig2fiff is
a text file, which contains the coordinates of the digitization
points in centimeters. The first line should contain a single number
which is the number of points listed in the file. Each of the following
lines contains a sequential number of the point, followed by the
three coordinates. mne_ctf_dig2fiff ignores
any text following the <span class="math">\(z\)</span> coordinate
on each line. If the <tt class="docutils literal"><span class="pre">--numfids</span></tt> option is specified,
the first three points indicate the three fiducial locations (1
= nasion, 2 = left auricular point, 3 = right auricular point).
Otherwise, the input file must end with three lines beginning with <tt class="docutils literal"><span class="pre">left</span></tt> , <tt class="docutils literal"><span class="pre">right</span></tt> ,
or <tt class="docutils literal"><span class="pre">nasion</span></tt> to indicate the locations of the fiducial
landmarks, respectively.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The sequential numbers should be unique within a file. I particular, the numbers 1, 2, and 3 must not be appear more than once if the <tt class="docutils literal"><span class="pre">--numfids</span></tt> options is used.</p>
</div>
<p>The command-line options for mne_ctf_dig2fiff are:</p>
<p><tt class="docutils literal"><span class="pre">--dig</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the input data file in CTF output format.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--numfids</span></tt></p>
<blockquote>
<div>Fiducial locations are numbered instead of labeled, see above.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--hpts</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the output hpts file. The format of this text file is
described in <a class="reference internal" href="#cjadjebh"><em>The hpts format</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fif</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the output fif file.</div></blockquote>
</div>
<div class="section" id="mne-dicom-essentials">
<span id="id47"></span><h3><a class="toc-backref" href="#id149">mne_dicom_essentials</a><a class="headerlink" href="#mne-dicom-essentials" title="Permalink to this headline">¶</a></h3>
<p>Print essential information about a dicom file.</p>
<p><tt class="docutils literal"><span class="pre">--in</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>The input file.</div></blockquote>
</div>
<div class="section" id="mne-edf2fiff">
<span id="id48"></span><h3><a class="toc-backref" href="#id150">mne_edf2fiff</a><a class="headerlink" href="#mne-edf2fiff" title="Permalink to this headline">¶</a></h3>
<p>The mne_edf2fiff allows
conversion of EEG data from EDF, EDF+, and BDF formats to the fif
format. Documentation for these three input formats can be found
at:</p>
<p><tt class="docutils literal"><span class="pre">EDF:</span></tt></p>
<blockquote>
<div><a class="reference external" href="http://www.edfplus.info/specs/edf.html">http://www.edfplus.info/specs/edf.html</a></div></blockquote>
<p><tt class="docutils literal"><span class="pre">EDF+:</span></tt></p>
<blockquote>
<div><a class="reference external" href="http://www.edfplus.info/specs/edfplus.html">http://www.edfplus.info/specs/edfplus.html</a></div></blockquote>
<p><tt class="docutils literal"><span class="pre">BDF:</span></tt></p>
<blockquote>
<div><a class="reference external" href="http://www.biosemi.com/faq/file_format.htm">http://www.biosemi.com/faq/file_format.htm</a></div></blockquote>
<p>EDF (European Data Format) and EDF+ are 16-bit formats while
BDF is a 24-bit variant of this format used by the EEG systems manufactured
by a company called BioSemi.</p>
<p>None of these formats support electrode location information
and  head shape digitization information. Therefore, this information
has to be provided separately. Presently hpts and elp file formats
are supported to include digitization data. For information on these
formats, see <a class="reference internal" href="#cjadjebh"><em>The hpts format</em></a> and <a class="reference external" href="http://www.sourcesignal.com/formats_probe.html">http://www.sourcesignal.com/formats_probe.html</a>.
Note that it is mandatory to have the three fiducial locations (nasion
and the two auricular points) included in the digitization data.
Using the locations of the fiducial points the digitization data
are converted to the MEG head coordinate system employed in the
MNE software, see <a class="reference internal" href="source_localization/forward.html#bjebibai"><em>The head and device coordinate systems</em></a>. In the comparison of the
channel names only the initial segment up to the first &#8216;-&#8216; (dash)
in the EDF/EDF+/BDF channel name is significant.</p>
<p>The EDF+ files may contain an annotation channel which can
be used to store trigger information. The Time-stamped Annotation
Lists (TALs) on the annotation  data can be converted to a trigger
channel (STI 014) using an annotation map file which associates
an annotation label with a number on the trigger channel. The TALs
can be listed with the <tt class="docutils literal"><span class="pre">--listtal</span></tt> option,
see below.</p>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">The data samples in a BDF file    are represented in a 3-byte (24-bit) format. Since 3-byte raw data    buffers are not presently supported in the fif format    these data will be changed to 4-byte integers in the conversion.    Since the maximum size of a fif file is 2 GBytes, the maximum size of    a BDF file to be converted is approximately 1.5 GBytes</p>
</div>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">The EDF/EDF+/BDF formats support channel    dependent sampling rates. This feature is not supported by mne_edf2fiff .    However, the annotation channel in the EDF+ format can have a different    sampling rate. The annotation channel data is not included in the    fif files output.</p>
</div>
<div class="section" id="using-mne-edf2fiff">
<h4>Using mne_edf2fiff<a class="headerlink" href="#using-mne-edf2fiff" title="Permalink to this headline">¶</a></h4>
<p>The command-line options of mne_edf2fiff are:</p>
<p><tt class="docutils literal"><span class="pre">--edf</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of the raw data file to process.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--tal</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>List the time-stamped annotation list (TAL) data from an EDF+ file here.
This output is useful to assist in creating the annotation map file,
see the <tt class="docutils literal"><span class="pre">--annotmap</span></tt> option, below.
This output file is an event file compatible with mne_browse_raw and mne_process_raw ,
see <a class="reference internal" href="gui/browse.html#ch-browse"><em>Browsing raw data with mne_browse_raw</em></a>. In addition, in the mapping between TAL
labels and trigger numbers provided by the <tt class="docutils literal"><span class="pre">--annotmap</span></tt> option is
employed to assign trigger numbers in the event file produced. In
the absence of the <tt class="docutils literal"><span class="pre">--annotmap</span></tt> option default trigger number 1024
is used.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--annotmap</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>Specify a file which maps the labels of the TALs to numbers on a trigger
channel (STI 014) which will be added to the output file if this
option is present. This annotation map file
may contain comment lines starting with the &#8216;%&#8217; or &#8216;#&#8217; characters.
The data lines contain a label-number pair, separated by a colon.
For example, a line &#8216;Trigger-1:9&#8217; means that each
annotation labeled with the text &#8216;Trigger-1&#8217; will
be translated to the number 9 on the trigger channel.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--elp</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of the an electrode location file. This file
is in the &#8220;probe&#8221; file format used by the <em>Source
Signal Imaging, Inc.</em> software. For description of the
format, see <a class="reference external" href="http://www.sourcesignal.com/formats_probe.html">http://www.sourcesignal.com/formats_probe.html</a>. Note
that some other software packages may produce electrode-position
files with the elp ending not
conforming to the above specification. As discussed above, the fiducial
marker locations, optional in the &#8220;probe&#8221; file
format specification are mandatory for mne_edf2fiff .
When this option is encountered on the command line any previously
specified hpts file will be ignored.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--hpts</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of an electrode position file in  the hpts format discussed
in <a class="reference internal" href="#cjadjebh"><em>The hpts format</em></a>. The mandatory entries are the fiducial marker
locations and the EEG electrode locations. It is recommended that
electrode (channel) names instead of numbers are used to label the
EEG electrode locations. When this option is encountered on the
command line any previously specified elp file
will be ignored.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--meters</span></tt></p>
<blockquote>
<div>Assumes that the digitization data in an hpts file
is given in meters instead of millimeters.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fif</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of the fif file to be output.</div></blockquote>
</div>
<div class="section" id="post-conversion-tasks">
<h4>Post-conversion tasks<a class="headerlink" href="#post-conversion-tasks" title="Permalink to this headline">¶</a></h4>
<p>This section outlines additional steps to be taken to use
the EDF/EDF+/BDF file is converted to the fif format in MNE:</p>
<ul class="simple">
<li>Some of the channels may not have a
digitized electrode location associated with them. If these channels
are used for EOG or EMG measurements, their channel types should
be changed to the correct ones using the <a class="reference internal" href="#mne-rename-channels"><em>mne_rename_channels</em></a> utility,
EEG channels which do not have a location
associated with them should be assigned to be MISC channels.</li>
<li>After the channel types are correctly defined, a topographical
layout file can be created for mne_browse_raw and mne_analyze using
the <a class="reference internal" href="#mne-make-eeg-layout"><em>mne_make_eeg_layout</em></a> utility.</li>
<li>The trigger channel name in BDF files is &#8220;Status&#8221;.
This must be specified with the <tt class="docutils literal"><span class="pre">--digtrig</span></tt> option or with help of
the MNE_TRIGGER_CH_NAME environment variable when <a class="reference internal" href="#mne-browse-raw"><em>mne_browse_raw</em></a> or
<a class="reference internal" href="#mne-process-raw"><em>mne_process_raw</em></a> is invoked.</li>
<li>Only the two least significant bytes on the &#8220;Status&#8221; channel
of BDF files are significant as trigger information the <tt class="docutils literal"><span class="pre">--digtrigmask</span></tt>
0xff option MNE_TRIGGER_CH_MASK environment variable should be used
to specify this to <a class="reference internal" href="#mne-browse-raw"><em>mne_browse_raw</em></a> and <a class="reference internal" href="#mne-process-raw"><em>mne_process_raw</em></a>,</li>
</ul>
</div>
</div>
<div class="section" id="mne-epochs2mat">
<span id="id49"></span><h3><a class="toc-backref" href="#id151">mne_epochs2mat</a><a class="headerlink" href="#mne-epochs2mat" title="Permalink to this headline">¶</a></h3>
<p>The utility mne_epochs2mat converts
epoch data including all or selected channels from a raw data file
to a simple binary file with an associated description file in Matlab
mat file format. With help of the description file, a matlab program
can easily read the epoch data from the simple binary file. Signal
space projection and bandpass filtering can be optionally applied
to the raw data prior to saving the epochs.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The MNE Matlab toolbox described in <a class="reference internal" href="matlab.html#ch-matlab"><em>Matlab toolbox</em></a> provides direct    access to raw fif files without conversion with mne_epochs2mat first.    Therefore, it is recommended that you use the Matlab toolbox rather than mne_epochs2mat which    creates large files occupying disk space unnecessarily. An exception    to this is the case where you apply a filter to the data and save    the band-pass filtered epochs.</p>
</div>
<div class="section" id="id50">
<h4>Command-line options<a class="headerlink" href="#id50" title="Permalink to this headline">¶</a></h4>
<p>mne_epochs2mat accepts
the following command-line options are:</p>
<p><tt class="docutils literal"><span class="pre">--raw</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of the raw data fif file to use as input.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--mat</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of the destination file. Anything following the last
period in the file name will be removed before composing the output
file name. The binary epoch file will be called &lt;<em>trimmed name</em>&gt; <tt class="docutils literal"><span class="pre">.epochs</span></tt> and
the corresponding Matlab description file will be &lt;<em>trimmed name</em>&gt; <tt class="docutils literal"><span class="pre">_desc.mat</span></tt> .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--tag</span> <span class="pre">&lt;*tag*&gt;</span></tt></p>
<blockquote>
<div>By default, all Matlab variables included in the description file
start with <tt class="docutils literal"><span class="pre">mne\_</span></tt>. This option changes the prefix to &lt;<em>tag</em>&gt; _.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--events</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>The file containing the event definitions. This can be a text or
fif format file produced by <a class="reference internal" href="#mne-process-raw"><em>mne_process_raw</em></a> or
<a class="reference internal" href="#mne-browse-raw"><em>mne_browse_raw</em></a>. With help of this file it is possible
to select virtually any data segment from the raw data file. If
this option is missing, the digital trigger channel in the raw data
file or a fif format event file produced automatically by mne_process_raw or mne_browse_raw is
consulted for event information.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--event</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Event number identifying the epochs of interest.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--tmin</span> <span class="pre">&lt;*time/ms*&gt;</span></tt></p>
<blockquote>
<div>The starting point of the epoch with respect to the event of interest.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--tmax</span> <span class="pre">&lt;*time/ms*&gt;</span></tt></p>
<blockquote>
<div>The endpoint of the epoch with respect to the event of interest.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--sel</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a text file which contains the names of the channels to include
in the output file, one channel name per line. If the <tt class="docutils literal"><span class="pre">--inv</span></tt> option
is specified, <tt class="docutils literal"><span class="pre">--sel</span></tt> is ignored. If neither <tt class="docutils literal"><span class="pre">--inv</span></tt> nor <tt class="docutils literal"><span class="pre">--sel</span></tt> is
present, all MEG and EEG channels are included. The digital trigger
channel can be included with the <tt class="docutils literal"><span class="pre">--includetrig</span></tt> option, described
below.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--inv</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies an inverse operator, which will be employed in two ways. First,
the channels included to output will be those included in the inverse
operator. Second, any signal-space projection operator present in
the inverse operator file will be applied to the data. This option
cancels the effect of <tt class="docutils literal"><span class="pre">--sel</span></tt> and <tt class="docutils literal"><span class="pre">--proj</span></tt> options.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--digtrig</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Name of the composite digital trigger channel. The default value
is &#8216;STI 014&#8217;. Underscores in the channel name
will be replaced by spaces.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--digtrigmask</span> <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>Mask to be applied to the trigger channel values before considering them.
This option is useful if one wants to set some bits in a don&#8217;t care
state. For example, some finger response pads keep the trigger lines
high if not in use, <em>i.e.</em>, a finger is not in
place. Yet, it is convenient to keep these devices permanently connected
to the acquisition system. The number can be given in decimal or
hexadecimal format (beginning with 0x or 0X). For example, the value
255 (0xFF) means that only the lowest order byte (usually trigger
lines 1 - 8 or bits 0 - 7) will be considered.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--includetrig</span></tt></p>
<blockquote>
<div>Add the digital trigger channel to the list of channels to output.
This option should not be used if the trigger channel is already
included in the selection specified with the <tt class="docutils literal"><span class="pre">--sel</span></tt> option.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--filtersize</span> <span class="pre">&lt;*size*&gt;</span></tt></p>
<blockquote>
<div>Adjust the length of the FFT to be applied in filtering. The number will
be rounded up to the next power of two. If the size is <span class="math">\(N\)</span>,
the corresponding length of time is <span class="math">\(^N/_{f_s}\)</span>,
where <span class="math">\(f_s\)</span> is the sampling frequency
of your data. The filtering procedure includes overlapping tapers
of length <span class="math">\(^N/_2\)</span> so that the total FFT
length will actually be <span class="math">\(2N\)</span>. The default
value is 4096.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--highpass</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>Highpass filter frequency limit. If this is too low with respect
to the selected FFT length and data file sampling frequency, the
data will not be highpass filtered. You can experiment with the
interactive version to find the lowest applicable filter for your
data. This value can be adjusted in the interactive version of the
program. The default is 0, i.e., no highpass filter in effect.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--highpassw</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>The width of the transition band of the highpass filter. The default
is 6 frequency bins, where one bin is <span class="math">\(^{f_s}/_{(2N)}\)</span>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--lowpass</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>Lowpass filter frequency limit. This value can be adjusted in the interactive
version of the program. The default is 40 Hz.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--lowpassw</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>The width of the transition band of the lowpass filter. This value
can be adjusted in the interactive version of the program. The default
is 5 Hz.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--filteroff</span></tt></p>
<blockquote>
<div>Do not filter the data.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--proj</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Include signal-space projection (SSP) information from this file.
If the <tt class="docutils literal"><span class="pre">--inv</span></tt> option is present, <tt class="docutils literal"><span class="pre">--proj</span></tt> has
no effect.</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Baseline has not been subtracted from the epochs. This has to be done in subsequent processing with Matlab if so desired.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Strictly speaking, trigger mask value zero would mean that all trigger inputs are ignored. However, for convenience,    setting the mask to zero or not setting it at all has the same effect    as 0xFFFFFFFF, <em>i.e.</em>, all bits set.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The digital trigger channel can also be set with the MNE_TRIGGER_CH_NAME environment variable. Underscores in the variable    value will <em>not</em> be replaced with spaces by mne_browse_raw or mne_process_raw .    Using the <tt class="docutils literal"><span class="pre">--digtrig</span></tt> option supersedes the MNE_TRIGGER_CH_NAME    environment variable.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The digital trigger channel mask can also be    set with the MNE_TRIGGER_CH_MASK environment variable. Using the <tt class="docutils literal"><span class="pre">--digtrigmask</span></tt> option    supersedes the MNE_TRIGGER_CH_MASK environment variable.</p>
</div>
</div>
<div class="section" id="the-binary-epoch-data-file">
<h4>The binary epoch data file<a class="headerlink" href="#the-binary-epoch-data-file" title="Permalink to this headline">¶</a></h4>
<p>mne_epochs2mat saves the
epoch data extracted from the raw data file is a simple binary file.
The data are stored as big-endian single-precision floating point
numbers. Assuming that each of the total of <span class="math">\(p\)</span> epochs
contains <span class="math">\(n\)</span> channels and <span class="math">\(m\)</span> time
points, the data <span class="math">\(s_{jkl}\)</span> are ordered
as</p>
<div class="math">
\[s_{111} \dotso s_{1n1} s_{211} \dotso s_{mn1} \dotso s_{mnp}\ ,\]</div>
<p>where the first index stands for the time point, the second
for the channel, and the third for the epoch number, respectively.
The data are not calibrated, i.e., the calibration factors present
in the Matlab description file have to be applied to get to physical
units as described below.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The maximum size of an epoch data file is 2 Gbytes, <em>i.e.</em>, 0.5 Gsamples.</p>
</div>
</div>
<div class="section" id="id51">
<h4>Matlab data structures<a class="headerlink" href="#id51" title="Permalink to this headline">¶</a></h4>
<p>The Matlab description files output by mne_epochs2mat contain
a data structure &lt;<em>tag</em>&gt;_epoch_info .
The fields of the this structure are listed in <a class="reference internal" href="#behfdcih"><em>The fields of the raw data info structure.</em></a>.
Further explanation of the epochs member
is provided in <a class="reference internal" href="#behhaghe"><em>The epochs member of the raw data info structure.</em></a>.</p>
<table border="1" class="docutils" id="behifjij">
<caption>The fields of the raw data info structure.</caption>
<colgroup>
<col width="23%" />
<col width="17%" />
<col width="60%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td>Variable</td>
<td>Size</td>
<td>Description</td>
</tr>
<tr class="row-even"><td>orig_file</td>
<td>string</td>
<td>The name of the original fif file specified with the
<tt class="docutils literal"><span class="pre">--raw</span></tt> option.</td>
</tr>
<tr class="row-odd"><td>epoch_file</td>
<td>string</td>
<td>The name of the epoch data file produced by mne_epocs2mat.</td>
</tr>
<tr class="row-even"><td>nchan</td>
<td>1</td>
<td>Number of channels.</td>
</tr>
<tr class="row-odd"><td>nepoch</td>
<td>1</td>
<td>Total number of epochs.</td>
</tr>
<tr class="row-even"><td>epochs</td>
<td>nepoch x 5</td>
<td>Description of the content of the epoch data file,
see <a class="reference internal" href="#behhaghe"><em>The epochs member of the raw data info structure.</em></a>.</td>
</tr>
<tr class="row-odd"><td>sfreq</td>
<td>1</td>
<td>The sampling frequency in Hz.</td>
</tr>
<tr class="row-even"><td>lowpass</td>
<td>1</td>
<td>Lowpass filter frequency (Hz)</td>
</tr>
<tr class="row-odd"><td>highpass</td>
<td>1</td>
<td>Highpass filter frequency (Hz)</td>
</tr>
<tr class="row-even"><td>ch_names</td>
<td>nchan (string)</td>
<td>String array containing the names of the channels included</td>
</tr>
<tr class="row-odd"><td>ch_types</td>
<td>nchan x 2</td>
<td>The column lists the types of the channels (1 = MEG, 2 =
EEG). The second column lists the coil types, see
<a class="reference internal" href="source_localization/forward.html#bgbbhgec"><em>Normal coil descriptions.</em></a> and <a class="reference internal" href="source_localization/forward.html#chdbdfje"><em>Accurate coil descriptions</em></a>. For EEG electrodes,
this value equals one.</td>
</tr>
<tr class="row-even"><td>ch_lognos</td>
<td>nchan x 1</td>
<td>Logical channel numbers as listed in the fiff file</td>
</tr>
<tr class="row-odd"><td>ch_units</td>
<td>nchan x 2</td>
<td>Units and unit multipliers as listed in the fif file.
The unit of the data is listed in the first column
(T = 112, T/m = 201, V = 107). At present, the second
column will be always zero, <em>i.e.</em>, no unit multiplier.</td>
</tr>
<tr class="row-even"><td>ch_pos</td>
<td>nchan x 12</td>
<td>The location information for each channel. The first three
values specify the origin of the sensor coordinate system
or the location of the electrode. For MEG channels, the
following nine number specify the <em>x</em>, <em>y</em>, and
<em>z</em>-direction unit vectors of the sensor coordinate
system. For EEG electrodes the first vector after the
electrode location specifies the location of the reference
electrode. If the reference is not specified this value is
all zeroes. The remaining unit vectors are irrelevant for
EEG electrodes.</td>
</tr>
<tr class="row-odd"><td>ch_cals</td>
<td>nchan x 2</td>
<td>The raw data output by mne_raw2mat are not calibrated.
The first column is the range member of the fiff data
structures and while the second is the cal member. To
get calibrated data values in the units given in
ch_units from the raw data, the data must be multiplied
with the product of range and cal .</td>
</tr>
<tr class="row-even"><td>meg_head_trans</td>
<td>4 x 4</td>
<td>The coordinate frame transformation from the MEG device
coordinates to the MEG head coordinates.</td>
</tr>
</tbody>
</table>
<table border="1" class="docutils" id="behhaghe">
<caption>The epochs member of the raw data info structure.</caption>
<colgroup>
<col width="19%" />
<col width="81%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td>Column</td>
<td>Contents</td>
</tr>
<tr class="row-even"><td>1</td>
<td>The raw data type (2 or 16 = 2-byte signed integer, 3 = 4-byte
signed integer, 4 = single-precision float). The epoch data are
written using the big-endian byte order. The data are stored
sample by sample.</td>
</tr>
<tr class="row-odd"><td>2</td>
<td>Byte location of this epoch in the binary epoch file.</td>
</tr>
<tr class="row-even"><td>3</td>
<td>First sample of this epoch in the original raw data file.</td>
</tr>
<tr class="row-odd"><td>4</td>
<td>First sample of the epoch with respect to the event.</td>
</tr>
<tr class="row-even"><td>5</td>
<td>Number of samples in the epoch.</td>
</tr>
</tbody>
</table>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">For source modelling purposes, it is recommended    that the MNE Matlab toolbox, see <a class="reference internal" href="matlab.html#ch-matlab"><em>Matlab toolbox</em></a> is employed    to read the measurement info instead of using the channel information    in the raw data info structure described in <a class="reference internal" href="#behifjij"><em>The fields of the raw data info structure.</em></a>.</p>
</div>
</div>
</div>
<div class="section" id="mne-evoked-data-summary">
<span id="id52"></span><h3><a class="toc-backref" href="#id152">mne_evoked_data_summary</a><a class="headerlink" href="#mne-evoked-data-summary" title="Permalink to this headline">¶</a></h3>
<p>Print a summary of evoked-response data sets in a file (averages only).</p>
<p><tt class="docutils literal"><span class="pre">--in</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>The input file.</div></blockquote>
</div>
<div class="section" id="mne-eximia2fiff">
<span id="id53"></span><h3><a class="toc-backref" href="#id153">mne_eximia2fiff</a><a class="headerlink" href="#mne-eximia2fiff" title="Permalink to this headline">¶</a></h3>
<p>Usage:</p>
<p><tt class="docutils literal"><span class="pre">mne_eximia2fiff</span></tt> [<tt class="docutils literal"><span class="pre">--dig</span></tt> dfile ] [<tt class="docutils literal"><span class="pre">--orignames</span></tt> ] file1 file2 ...</p>
<p>where file1 file2 ...
are eXimia <tt class="docutils literal"><span class="pre">nxe</span></tt> files and the <tt class="docutils literal"><span class="pre">--orignames</span></tt> option
is passed on to <a class="reference internal" href="#mne-brain-vision2fiff"><em>mne_brain_vision2fiff</em></a>.
If you want to convert all data files in a directory, say</p>
<p><tt class="docutils literal"><span class="pre">mne_eximia2fiff</span> <span class="pre">*.nxe</span></tt></p>
<p>The optional file specified with the <tt class="docutils literal"><span class="pre">--dig</span></tt> option is assumed
to contain digitizer data from the recording in the Nexstim format.
The resulting fif data file will contain these data converted to
the fif format as well as the coordinate transformation between
the eXimia digitizer and MNE head coordinate systems.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This script converts raw data files only.</p>
</div>
</div>
<div class="section" id="mne-fit-sphere-to-surf">
<span id="id54"></span><h3><a class="toc-backref" href="#id154">mne_fit_sphere_to_surf</a><a class="headerlink" href="#mne-fit-sphere-to-surf" title="Permalink to this headline">¶</a></h3>
<div class="section" id="id55">
<h4>Purpose<a class="headerlink" href="#id55" title="Permalink to this headline">¶</a></h4>
<p>The utility mne_fit_sphere_to_surf finds
the sphere which best fits a given surface.</p>
</div>
<div class="section" id="id56">
<h4>Command line options<a class="headerlink" href="#id56" title="Permalink to this headline">¶</a></h4>
<p>mne_fit_sphere_to_surf accepts
the following command-line options:</p>
<p><tt class="docutils literal"><span class="pre">--bem</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>A BEM file to use. The names of these files usually end with <tt class="docutils literal"><span class="pre">bem.fif</span></tt> or <tt class="docutils literal"><span class="pre">bem-sol.fif</span></tt> .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--surf</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>A FreeSurfer surface file to read. This is an alternative to using
a surface from the BEM file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--scalp</span></tt></p>
<blockquote>
<div>Use the scalp surface instead of the inner skull surface in sphere
fitting. If the surface is specified with the <tt class="docutils literal"><span class="pre">--surf</span></tt> option,
this one is irrelevant.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--mritrans</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>A file containing a transformation matrix between the MEG head coordinates
and MRI coordinates. With this option, the sphere origin will be
output in MEG head coordinates. Otherwise the output will be in MRI
coordinates.</div></blockquote>
</div>
</div>
<div class="section" id="mne-fix-mag-coil-types">
<span id="id57"></span><h3><a class="toc-backref" href="#id155">mne_fix_mag_coil_types</a><a class="headerlink" href="#mne-fix-mag-coil-types" title="Permalink to this headline">¶</a></h3>
<p>The purpose of mne_fix_mag_coil_types is
to change coil type 3022 to 3024 in the MEG channel definition records
in the data files specified on the command line.</p>
<p>As shown in Tables 5.2 and 5.3, the Neuromag Vectorview systems
can contain magnetometers with two different coil sizes (coil types
3022 and 3023 vs. 3024). The systems incorporating coils of type
3024 were introduced last. At some sites the data files have still
defined the magnetometers to be of type 3022 to ensure compatibility
with older versions of Neuromag software. In the MNE software as
well as in the present version of Neuromag software coil type 3024
is fully supported. Therefore, it is now safe to upgrade the data
files to use the true coil type.</p>
<p>If the <tt class="docutils literal"><span class="pre">--magnes</span></tt> option is specified, the 4D
Magnes magnetometer coil type (4001) is changed to 4D Magnes gradiometer
coil type (4002). Use this option always and <em>only
if</em> your Magnes data comes from a system with axial gradiometers
instead of magnetometers. The fif converter included with the Magnes
system does not assign the gradiometer coil type correctly.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The effect of the difference between the coil sizes of magnetometer types 3022 and 3024 on the current estimates computed by the MNE software is very small. Therefore the use of mne_fix_mag_coil_types is not mandatory.</p>
</div>
</div>
<div class="section" id="mne-fix-stim14">
<span id="id58"></span><h3><a class="toc-backref" href="#id156">mne_fix_stim14</a><a class="headerlink" href="#mne-fix-stim14" title="Permalink to this headline">¶</a></h3>
<p>Some earlier versions of the Neuromag acquisition software
had a problem with the encoding of the eighth bit on the digital
stimulus channel STI 014. This problem has been now fixed. Old data
files can be fixed with mne_fix_stim14 ,
which takes raw data file names as arguments. mne_fix_stim14 also
changes the calibration of STI 014 to unity. If the encoding of
STI 014 is already correct, running mne_fix_stim14 will
not have any effect on the raw data.</p>
<p>In newer Neuromag Vectorview systems with 16-bit digital
inputs the upper two bytes of the samples may be incorrectly set
when stimulus input 16 is used and the data are acquired in the
32-bit  mode. This problem can be fixed by running mne_fix_stim14 on
a raw data file with the <tt class="docutils literal"><span class="pre">--32</span></tt> option:</p>
<p><tt class="docutils literal"><span class="pre">mne_fix_stim14</span> <span class="pre">--32</span></tt>  &lt;<em>raw data file</em>&gt;</p>
<p>In this case, the correction will be applied to the stimulus
channels &#8216;STI101&#8217; and &#8216;STI201&#8217;.</p>
</div>
<div class="section" id="mne-flash-bem">
<span id="id59"></span><h3><a class="toc-backref" href="#id157">mne_flash_bem</a><a class="headerlink" href="#mne-flash-bem" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--help</span></tt></p>
<blockquote>
<div>Prints the usage information.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--usage</span></tt></p>
<blockquote>
<div>Prints the usage information.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--noconvert</span></tt></p>
<blockquote>
<div>Skip conversion of the original MRI data. The original data are
not needed and the preparatory steps 1.-3. listed below
are thus not needed.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--noflash30</span></tt></p>
<blockquote>
<div>The 30-degree flip angle data are not used.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--unwarp</span>&nbsp; <span class="pre">&lt;*type*&gt;</span></tt></p>
<blockquote>
<div>Run grad_unwarp with <tt class="docutils literal"><span class="pre">--unwarp</span></tt>  &lt;<em>type</em>&gt; option on each of the converted
data sets.</div></blockquote>
</div>
<div class="section" id="mne-insert-4d-comp">
<span id="id60"></span><h3><a class="toc-backref" href="#id158">mne_insert_4D_comp</a><a class="headerlink" href="#mne-insert-4d-comp" title="Permalink to this headline">¶</a></h3>
<p>Import Magnes WH3600 reference channel data from a text file.</p>
<p><tt class="docutils literal"><span class="pre">--in</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>The name of the fif file containing the helmet data.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--ref</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>The name of the text file containing the reference channel data.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--out</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>The output fif file.</div></blockquote>
</div>
<div class="section" id="mne-kit2fiff">
<span id="id61"></span><h3><a class="toc-backref" href="#id159">mne_kit2fiff</a><a class="headerlink" href="#mne-kit2fiff" title="Permalink to this headline">¶</a></h3>
<p>The utility mne_kit2fiff was
created in collaboration with Alec Maranz and Asaf Bachrach to import
their MEG data acquired with the 160-channel KIT MEG system to MNE
software.</p>
<p>To import the data, the following input files are mandatory:</p>
<ul class="simple">
<li>The Polhemus data file (elp file)
containing the locations of the fiducials and the head-position
indicator (HPI) coils. These data are usually given in the CTF/4D
head coordinate system. However, mne_kit2fiff does
not rely on this assumption. This file can be exported directly from
the KIT system.</li>
<li>A file containing the locations of the HPI coils in the MEG
device coordinate system. These data are used together with the elp file
to establish the coordinate transformation between the head and
device coordinate systems. This file can be produced easily by manually
editing one of the files exported by the KIT system.</li>
<li>A sensor data file (sns file)
containing the locations and orientations of the sensors. This file
can be exported directly from the KIT system.</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The output fif file will use the Neuromag head    coordinate system convention, see <a class="reference internal" href="source_localization/forward.html#bjebibai"><em>The head and device coordinate systems</em></a>. A coordinate    transformation between the CTF/4D head coordinates and the Neuromag    head coordinates is included. This transformation can be read with    MNE Matlab Toolbox routines, see <a class="reference internal" href="matlab.html#ch-matlab"><em>Matlab toolbox</em></a>.</p>
</div>
<p>The following input files are optional:</p>
<ul class="simple">
<li>A head shape data file (hsp file)
containing locations of additional points from the head surface.
These points must be given in the same coordinate system as that
used for the elp file and the
fiducial locations must be within 1 mm from those in the elp file.</li>
<li>A raw data file containing the raw data values, sample by
sample, as text. If this file is not specified, the output fif file
will only contain the measurement info block.</li>
</ul>
<p>By default mne_kit2fiff includes
the first 157 channels, assumed to be the MEG channels, in the output
file. The compensation channel data are not converted by default
but can be added, together with other channels, with the <tt class="docutils literal"><span class="pre">--type</span></tt> .
The channels from 160 onwards are designated as miscellaneous input
channels (MISC 001, MISC 002, etc.). The channel names and types
of these channels can be afterwards changed with the <a class="reference internal" href="#mne-rename-channels"><em>mne_rename_channels</em></a>
utility. In addition, it is possible to synthesize
the digital trigger channel (STI 014) from available analog
trigger channel data, see the <tt class="docutils literal"><span class="pre">--stim</span></tt> option, below.
The synthesized trigger channel data value at sample <span class="math">\(k\)</span> will
be:</p>
<div class="math">
\[s(k) = \sum_{p = 1}^n {t_p(k) 2^{p - 1}}\ ,\]</div>
<p>where <span class="math">\(t_p(k)\)</span> are the thresholded
from the input channel data d_p(k):</p>
<div class="math">
\[\begin{split}t_p(k) = \Bigg\{ \begin{array}{l}
    0 \text{  if  } d_p(k) \leq t\\
    1 \text{  if  } d_p(k) &gt; t
\end{array}\ .\end{split}\]</div>
<p>The threshold value <span class="math">\(t\)</span> can
be adjusted with the <tt class="docutils literal"><span class="pre">--stimthresh</span></tt> option, see below.</p>
<p>mne_kit2fiff accepts the following command-line options:</p>
<p><tt class="docutils literal"><span class="pre">--elp</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>The name of the file containing the locations of the fiducials and
the HPI coils. This option is mandatory.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--hsp</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>The name of the file containing the locations of the fiducials and additional
points on the head surface. This file is optional.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--sns</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>The name of file containing the sensor locations and orientations. This
option is mandatory.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--hpi</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>The name of a text file containing the locations of the HPI coils
in the MEG device coordinate frame, given in millimeters. The order of
the coils in this file does not have to be the same as that in the elp file.
This option is mandatory.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--raw</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of the raw data file. If this file is not specified, the
output fif file will only contain the measurement info block.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--sfreq</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>The sampling frequency of the data. If this option is not specified, the
sampling frequency defaults to 1000 Hz.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--lowpass</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>The lowpass filter corner frequency used in the data acquisition.
If not specified, this value defaults to 200 Hz.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--highpass</span> <span class="pre">&lt;*value/Hz*&gt;</span></tt></p>
<blockquote>
<div>The highpass filter corner frequency used in the data acquisition.
If not specified, this value defaults to 0 Hz (DC recording).</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--out</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of the output fif format data file. If this file
is not specified, no output is produced but the elp , hpi ,
and hsp files are processed normally.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--stim</span> <span class="pre">&lt;*chs*&gt;</span></tt></p>
<blockquote>
<div>Specifies a colon-separated list of numbers of channels to be used
to synthesize a digital trigger channel. These numbers refer to
the scanning order channels as listed in the sns file,
starting from one. The digital trigger channel will be the last
channel in the file. If this option is absent, the output file will
not contain a trigger channel.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--stimthresh</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>The threshold value used when synthesizing the digital trigger channel,
see above. Defaults to 1.0.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--add</span> <span class="pre">&lt;*chs*&gt;</span></tt></p>
<blockquote>
<div>Specifies a colon-separated list of numbers of channels to include between
the 157 default MEG channels and the digital trigger channel. These
numbers refer to the scanning order channels as listed in the sns file,
starting from one.</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The mne_kit2fiff utility    has not been extensively tested yet.</p>
</div>
</div>
<div class="section" id="mne-list-bem">
<span id="id62"></span><h3><a class="toc-backref" href="#id160">mne_list_bem</a><a class="headerlink" href="#mne-list-bem" title="Permalink to this headline">¶</a></h3>
<p>The utility mne_list_bem outputs
the BEM meshes in text format. The default output data contains
the <em>x</em>, <em>y</em>, and <em>z</em> coordinates
of the vertices, listed in millimeters, one vertex per line.</p>
<p>The command-line options are:</p>
<p><tt class="docutils literal"><span class="pre">--bem</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>The BEM file to be listed. The file name normally ends with -bem.fif or -bem-sol.fif .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--out</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>The output file name.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--id</span> <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>Identify the surface to be listed. The surfaces are numbered starting with
the innermost surface. Thus, for a three-layer model the surface numbers
are: 4 = scalp, 3 = outer skull, 1 = inner skull
Default value is 4.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--gdipoli</span></tt></p>
<blockquote>
<div>List the surfaces in the format required by Thom Oostendorp&#8217;s
gdipoli program. This is also the default input format for mne_surf2bem .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--meters</span></tt></p>
<blockquote>
<div>List the surface coordinates in meters instead of millimeters.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--surf</span></tt></p>
<blockquote>
<div>Write the output in the binary FreeSurfer format.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--xfit</span></tt></p>
<blockquote>
<div>Write a file compatible with xfit. This is the same effect as using
the options <tt class="docutils literal"><span class="pre">--gdipoli</span></tt> and <tt class="docutils literal"><span class="pre">--meters</span></tt> together.</div></blockquote>
</div>
<div class="section" id="mne-list-coil-def">
<span id="id63"></span><h3><a class="toc-backref" href="#id161">mne_list_coil_def</a><a class="headerlink" href="#mne-list-coil-def" title="Permalink to this headline">¶</a></h3>
<p>List available coil definitions.</p>
<p><tt class="docutils literal"><span class="pre">--in</span> <span class="pre">&lt;def&gt;</span></tt></p>
<blockquote>
<div>Validate a coil definition file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--out</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>List the coil definitions to this file (default: stdout).</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--type</span> <span class="pre">&lt;type&gt;</span></tt></p>
<blockquote>
<div>Coil type to list.</div></blockquote>
</div>
<div class="section" id="mne-list-proj">
<span id="id64"></span><h3><a class="toc-backref" href="#id162">mne_list_proj</a><a class="headerlink" href="#mne-list-proj" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--in</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Input file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--ascin</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Input file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--exclude</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Exclude these channels from the projection (set entries to zero).</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--chs</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Specify a file which contains a channel selection for the output (useful for graph)</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--asc</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Text output.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fif</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Fif output.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--lisp</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Lisp output.</div></blockquote>
</div>
<div class="section" id="mne-list-source-space">
<span id="id65"></span><h3><a class="toc-backref" href="#id163">mne_list_source_space</a><a class="headerlink" href="#mne-list-source-space" title="Permalink to this headline">¶</a></h3>
<p>The utility mne_list_source_space outputs
the source space information into text files suitable for loading
into the Neuromag MRIlab software.</p>
<p><tt class="docutils literal"><span class="pre">--src</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>The source space to be listed. This can be either the output from mne_make_source_space
(<cite>*src.fif</cite>), output from the forward calculation (<cite>*fwd.fif</cite>), or
the output from the inverse operator decomposition (<cite>*inv.fif</cite>).</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--mri</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>A file containing the transformation between the head and MRI coordinates
is specified with this option. This file can be either a Neuromag
MRI description file, the output from the forward calculation (<cite>*fwd.fif</cite>),
or the output from the inverse operator decomposition (<cite>*inv.fif</cite>).
If this file is included, the output will be in head coordinates.
Otherwise the source space will be listed in MRI coordinates.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--dip</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the &#8216;stem&#8217; for the Neuromag text format
dipole files to be output. Two files will be produced: &lt;<em>stem</em>&gt; -lh.dip
and &lt;<em>stem</em>&gt; -rh.dip. These correspond
to the left and right hemisphere part of the source space, respectively.
This source space data can be imported to MRIlab through the File/Import/Dipoles menu
item.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--pnt</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the &#8216;stem&#8217; for Neuromag text format
point files to be output. Two files will be produced: &lt;<em>stem</em>&gt; -lh.pnt
and &lt;<em>stem</em>&gt; -rh.pnt. These correspond
to the left and right hemisphere part of the source space, respectively.
This source space data can be imported to MRIlab through the File/Import/Strings menu
item.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--exclude</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Exclude the source space points defined by the given FreeSurfer &#8216;label&#8217; file
from the output. The name of the file should end with <tt class="docutils literal"><span class="pre">-lh.label</span></tt>
if it refers to the left hemisphere and with <tt class="docutils literal"><span class="pre">-rh.label</span></tt> if
it lists points in the right hemisphere, respectively.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--include</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Include only the source space points defined by the given FreeSurfer &#8216;label&#8217; file
to the output. The file naming convention is the same as described
above under the <tt class="docutils literal"><span class="pre">--exclude</span></tt> option. Are &#8216;include&#8217; labels are
processed before the &#8216;exclude&#8217; labels.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--all</span></tt></p>
<blockquote>
<div>Include all nodes in the output files instead of only those active
in the source space. Note that the output files will be huge if
this option is active.</div></blockquote>
</div>
<div class="section" id="mne-list-versions">
<span id="id66"></span><h3><a class="toc-backref" href="#id164">mne_list_versions</a><a class="headerlink" href="#mne-list-versions" title="Permalink to this headline">¶</a></h3>
<p>The utility mne_list_versions lists
version numbers and compilation dates of all software modules that
provide this information. This administration utility is located
in <tt class="docutils literal"><span class="pre">$MNE_ROOT/bin/admin</span></tt> , The output from mne_list_versions or
output of individual modules with <tt class="docutils literal"><span class="pre">--version</span></tt> option
is useful when bugs are reported to the developers of MNE software.</p>
</div>
<div class="section" id="mne-make-cor-set">
<span id="id67"></span><h3><a class="toc-backref" href="#id165">mne_make_cor_set</a><a class="headerlink" href="#mne-make-cor-set" title="Permalink to this headline">¶</a></h3>
<p>The utility mne_make_cor_set creates
a fif format MRI description
file optionally including the MRI data using FreeSurfer MRI volume
data as input. The command-line options are:</p>
<p><tt class="docutils literal"><span class="pre">--dir</span> <span class="pre">&lt;*directory*&gt;</span></tt></p>
<blockquote>
<div>Specifies a directory containing the MRI volume in COR format. Any
previous <tt class="docutils literal"><span class="pre">--mgh</span></tt> options are cancelled when this option
is encountered.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--withdata</span></tt></p>
<blockquote>
<div>Include the pixel data to the output file. This option is implied
with the <tt class="docutils literal"><span class="pre">--mgh</span></tt> option.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--mgh</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>An MRI volume volume file in mgh or mgz format.
The <tt class="docutils literal"><span class="pre">--withdata</span></tt> option is implied with this type of
input. Furthermore, the <span class="math">\(T_3\)</span> transformation,
the Talairach transformation <span class="math">\(T_4\)</span> from
the talairach.xfm file referred to in the MRI volume, and the the
fixed transforms <span class="math">\(T_-\)</span> and <span class="math">\(T_+\)</span> will
added to the output file. For definition of the coordinate transformations,
see <a class="reference internal" href="source_localization/forward.html#chdedfib"><em>MEG/EEG and MRI coordinate systems</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--talairach</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Take the Talairach transform from this file instead of the one specified
in mgh/mgz files.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--out</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the output file, which is a fif-format MRI description
file.</div></blockquote>
</div>
<div class="section" id="mne-make-derivations">
<span id="id68"></span><h3><a class="toc-backref" href="#id166">mne_make_derivations</a><a class="headerlink" href="#mne-make-derivations" title="Permalink to this headline">¶</a></h3>
<div class="section" id="id69">
<h4>Purpose<a class="headerlink" href="#id69" title="Permalink to this headline">¶</a></h4>
<p>In mne_browse_raw , channel
derivations are defined as linear combinations of real channels
existing in the data files. The utility mne_make_derivations reads
derivation data from a suitably formatted text file and produces
a fif file containing the weights of derived channels as a sparse
matrix. Two input file formats are accepted:</p>
<ul class="simple">
<li>A file containing arithmetic expressions
defining the derivations and</li>
<li>A file containing a matrix which specifies the weights of
the channels in each derivation.</li>
</ul>
<p>Both of these formats are described in</p>
</div>
<div class="section" id="id70">
<h4>Command-line options<a class="headerlink" href="#id70" title="Permalink to this headline">¶</a></h4>
<p>mne_make_derivations recognizes
the following command-line options:</p>
<p><tt class="docutils literal"><span class="pre">--in</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a measurement file which contains the EEG electrode locations.
This file is not modified.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--inmat</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the output file where the layout is stored. Suffix <tt class="docutils literal"><span class="pre">.lout</span></tt> is recommended
for layout files. mne_analyze and mne_browse_raw look
for the custom layout files from the directory <tt class="docutils literal"><span class="pre">$HOME/.mne/lout</span></tt> .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--trans</span></tt></p>
<blockquote>
<div>Indicates that the file specified with the <tt class="docutils literal"><span class="pre">--inmat</span></tt> option
contains a transpose of the derivation matrix.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--thresh</span>&nbsp; <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>Specifies the threshold between values to be considered zero and non-zero
in the input file specified with the <tt class="docutils literal"><span class="pre">--inmat</span></tt> option.
The default threshold is <span class="math">\(10^{-6}\)</span>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--out</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies output fif file to contain the derivation data. The recommended
name of the derivation file has the format  &lt;<span class="math">\(name\)</span>&gt; <tt class="docutils literal"><span class="pre">-deriv.fif</span></tt> .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--list</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>List the contents of a derivation file to standard output. If this
option is missing and <tt class="docutils literal"><span class="pre">--out</span></tt> is specified, the content
of the output file will be listed once it is complete. If neither <tt class="docutils literal"><span class="pre">--list</span></tt> nor <tt class="docutils literal"><span class="pre">--out</span></tt> is present,
and <tt class="docutils literal"><span class="pre">--in</span></tt> or <tt class="docutils literal"><span class="pre">--inmat</span></tt> is specified, the
interpreted contents of the input file is listed.</div></blockquote>
</div>
<div class="section" id="derivation-file-formats">
<h4>Derivation file formats<a class="headerlink" href="#derivation-file-formats" title="Permalink to this headline">¶</a></h4>
<p>All lines in the input files starting with the pound sign
(#) are considered to be comments. The format of a derivation in
a arithmetic input file is:</p>
<div class="math">
\[\langle name \rangle = [\langle w_1 \rangle *] \langle name_1 \rangle + [\langle w_2 \rangle *] \langle name_2 \rangle \dotso\]</div>
<p>where &lt;<span class="math">\(name\)</span>&gt; is the
name of the derived channel, <span class="math">\(name_k\)</span> are
the names of the channels comprising the derivation, and <span class="math">\(w_k\)</span> are
their weights. Note that spaces are necessary between the items.
Channel names containing spaces must be put in quotes. For example,</p>
<p><tt class="docutils literal"><span class="pre">EEG-diff</span> <span class="pre">=</span> <span class="pre">&quot;EEG</span> <span class="pre">003&quot;</span> <span class="pre">-</span> <span class="pre">&quot;EEG</span> <span class="pre">002&quot;</span></tt></p>
<p>defines a channel <tt class="docutils literal"><span class="pre">EEG-diff</span></tt> which is a difference
between <tt class="docutils literal"><span class="pre">EEG</span> <span class="pre">003</span></tt> and <tt class="docutils literal"><span class="pre">EEG</span> <span class="pre">002</span></tt> . Similarly,</p>
<p><tt class="docutils literal"><span class="pre">EEG-der</span> <span class="pre">=</span> <span class="pre">3</span> <span class="pre">*</span> <span class="pre">&quot;EEG</span> <span class="pre">010&quot;</span> <span class="pre">-</span> <span class="pre">2</span> <span class="pre">*</span> <span class="pre">&quot;EEG</span> <span class="pre">002&quot;</span></tt></p>
<p>defines a channel which is three times <tt class="docutils literal"><span class="pre">EEG</span> <span class="pre">010</span></tt> minus
two times <tt class="docutils literal"><span class="pre">EEG</span> <span class="pre">002</span></tt> .</p>
<p>The format of a matrix derivation file is:</p>
<div class="math">
\[\langle nrow \rangle \langle ncol \rangle \langle names\ of\ the\ input\ channels \rangle \langle name_1 \rangle \langle weights \rangle \dotso\]</div>
<p>The combination of the two arithmetic examples, above can
be thus represented as:</p>
<p><tt class="docutils literal"><span class="pre">2</span> <span class="pre">3</span> <span class="pre">&quot;EEG</span> <span class="pre">002&quot;</span> <span class="pre">&quot;EEG</span> <span class="pre">003&quot;</span> <span class="pre">&quot;EEG</span> <span class="pre">010&quot;</span> <span class="pre">EEG-diff</span> <span class="pre">-1</span> <span class="pre">1</span>&nbsp; <span class="pre">0</span> <span class="pre">EEG-der</span> <span class="pre">-2</span> <span class="pre">0</span>&nbsp; <span class="pre">3</span></tt></p>
<p>Before a derivation is accepted to use by mne_browse_raw ,
the following criteria have to be met:</p>
<ul class="simple">
<li>All channels to be combined into a single
derivation must have identical units of measure.</li>
<li>All channels in a single derivation have to be of the same
kind, <em>e.g.</em>, MEG channels or EEG channels.</li>
<li>All channels specified in a derivation have to be present
in the currently loaded data set.</li>
</ul>
<p>The validity check is done when a derivation file is loaded
into mne_browse_raw , see <a class="reference internal" href="gui/browse.html#cacfhafh"><em>Load derivations</em></a>.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">You might consider renaming the EEG channels    with descriptive labels related to the standard 10-20 system using    the <a class="reference internal" href="#mne-rename-channels"><em>mne_rename_channels</em></a> utility. This allows you to use standard EEG    channel names in the derivations you define as well as in the channel    selection files used in mne_browse_raw ,    see <a class="reference internal" href="gui/browse.html#caccjejd"><em>Selection</em></a>.</p>
</div>
</div>
</div>
<div class="section" id="mne-make-eeg-layout">
<span id="id71"></span><h3><a class="toc-backref" href="#id167">mne_make_eeg_layout</a><a class="headerlink" href="#mne-make-eeg-layout" title="Permalink to this headline">¶</a></h3>
<div class="section" id="id72">
<h4>Purpose<a class="headerlink" href="#id72" title="Permalink to this headline">¶</a></h4>
<p>Both MNE software (mne_analyze and mne_browse_raw)
and Neuromag software (xplotter and xfit)
employ text layout files to create topographical displays of MEG
and EEG data. While the MEG channel layout is fixed, the EEG layout
varies from experiment to experiment, depending on the number of
electrodes used and the electrode cap configuration. The utility mne_make_eeg_layout was
created to produce custom EEG layout files based on the EEG electrode
location information included in the channel description records.</p>
<p>mne_make_eeg_layout uses
azimuthal equidistant projection to map the EEG channel locations
onto a plane. The mapping consists of the following steps:</p>
<ul class="simple">
<li>A sphere is fitted to the electrode
locations and the locations are translated by the location of the
origin of the best-fitting sphere.</li>
<li>The spherical coordinates (<span class="math">\(r_k\)</span>, <span class="math">\(\theta_k\)</span>, and <span class="math">\(\phi_k\)</span>)
corresponding to each translated electrode location are computed.</li>
<li>The projected locations <span class="math">\(u_k = R \theta_k \cos{\phi_k}\)</span> and <span class="math">\(v_k = R \theta_k \sin{\phi_k}\)</span> are
computed. By default, <span class="math">\(R = 20/{^{\pi}/_2}\)</span>, <em>i.e.</em> at
the equator (<span class="math">\(\theta = ^{\pi}/_2\)</span>) the multiplier is
20. This projection radius can be adjusted with the <tt class="docutils literal"><span class="pre">--prad</span></tt> option.
Increasing or decreasing <span class="math">\(R\)</span> makes
the spacing between the channel viewports larger or smaller, respectively.</li>
<li>A viewport with width 5 and height 4 is placed centered at
the projected location. The width and height of the viewport can
be adjusted with the <tt class="docutils literal"><span class="pre">--width</span></tt> and <tt class="docutils literal"><span class="pre">--height</span></tt> options</li>
</ul>
<p>The command-line options are:</p>
<p><tt class="docutils literal"><span class="pre">--lout</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of the layout file to be output.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--nofit</span></tt></p>
<blockquote>
<div>Do not fit a sphere to the electrode locations but use a standard sphere
center (<span class="math">\(x = y = 0\)</span>, and <span class="math">\(z = 40\)</span> mm) instead.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--prad</span>&nbsp; <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>Specifies a non-standard projection radius <span class="math">\(R\)</span>,
see above.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--width</span>&nbsp; <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>Specifies the width of the viewports. Default value = 5.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--height</span>&nbsp; <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>Specifies the height of the viewports. Default value = 4.</div></blockquote>
</div>
</div>
<div class="section" id="mne-make-morph-maps">
<span id="id73"></span><h3><a class="toc-backref" href="#id168">mne_make_morph_maps</a><a class="headerlink" href="#mne-make-morph-maps" title="Permalink to this headline">¶</a></h3>
<p>Prepare the mapping data for subject-to-subject morphing.</p>
<p><tt class="docutils literal"><span class="pre">--redo</span></tt></p>
<blockquote>
<div>Recompute the morphing maps even if they already exist.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--from</span> <span class="pre">&lt;subject&gt;</span></tt></p>
<blockquote>
<div>Compute morphing maps from this subject.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--to</span> <span class="pre">&lt;subject&gt;</span></tt></p>
<blockquote>
<div>Compute morphing maps to this subject.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--all</span></tt></p>
<blockquote>
<div>Do all combinations. If this is used without either <tt class="docutils literal"><span class="pre">--from</span></tt> or <tt class="docutils literal"><span class="pre">--to</span></tt> options,
morphing maps for all possible combinations are computed. If <tt class="docutils literal"><span class="pre">--from</span></tt> or <tt class="docutils literal"><span class="pre">--to</span></tt> is
present, only maps between the specified subject and all others
are computed.</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Because all morphing map files contain maps in both directions, the choice of <tt class="docutils literal"><span class="pre">--from</span></tt> and <tt class="docutils literal"><span class="pre">--to</span></tt> options    only affect the naming of the morphing map files to be produced. mne_make_morph_maps creates directory <tt class="docutils literal"><span class="pre">$SUBJECTS_DIR/morph-maps</span></tt> if necessary.</p>
</div>
</div>
<div class="section" id="mne-make-uniform-stc">
<span id="id74"></span><h3><a class="toc-backref" href="#id169">mne_make_uniform_stc</a><a class="headerlink" href="#mne-make-uniform-stc" title="Permalink to this headline">¶</a></h3>
<p>The output will have a time range from -100 to 300 ms.
There will be one cycle of 5-Hz sine wave, with the peaks at 50 and 150 ms</p>
<p><tt class="docutils literal"><span class="pre">--src</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Source space to use.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--stc</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Stc file to produce.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--maxval</span> <span class="pre">&lt;value&gt;</span></tt></p>
<blockquote>
<div>Maximum value (at 50 ms, default 10).</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--all</span></tt></p>
<blockquote>
<div>Include all points to the output files.</div></blockquote>
</div>
<div class="section" id="mne-mark-bad-channels">
<span id="id75"></span><h3><a class="toc-backref" href="#id170">mne_mark_bad_channels</a><a class="headerlink" href="#mne-mark-bad-channels" title="Permalink to this headline">¶</a></h3>
<p>This utility adds or replaces information about unusable
(bad) channels. The command line options are:</p>
<p><tt class="docutils literal"><span class="pre">--bad</span>&nbsp; <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>Specify a text file containing the names of the bad channels, one channel
name per line. The names of the channels in this file must match
those in the data file exactly. If this option is missing, the bad channel
information is cleared.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">&lt;*data</span> <span class="pre">file</span> <span class="pre">name*&gt;</span></tt></p>
<blockquote>
<div>The remaining arguments are taken as data file names to be modified.</div></blockquote>
</div>
<div class="section" id="mne-morph-labels">
<span id="id76"></span><h3><a class="toc-backref" href="#id171">mne_morph_labels</a><a class="headerlink" href="#mne-morph-labels" title="Permalink to this headline">¶</a></h3>
<p>Morph label files from one brain to another.</p>
<p><tt class="docutils literal"><span class="pre">--from</span> <span class="pre">&lt;*subject*&gt;</span></tt></p>
<blockquote>
<div>Name of the subject for which the labels were originally defined.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--to</span> <span class="pre">&lt;*subject*&gt;</span></tt></p>
<blockquote>
<div>Name of the subject for which the morphed labels should be created.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--labeldir</span> <span class="pre">&lt;*directory*&gt;</span></tt></p>
<blockquote>
<div>A directory containing the labels to morph.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--prefix</span> <span class="pre">&lt;prefix&gt;</span></tt></p>
<blockquote>
<div>Adds &lt;<em>prefix</em>&gt; in the beginning
of the output label names. A dash will be inserted between &lt;<em>prefix</em>&gt; and
the rest of the name.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--smooth</span> <span class="pre">&lt;number&gt;</span></tt></p>
<blockquote>
<div>Apply smoothing with the indicated number of iteration steps (see <a class="reference internal" href="source_localization/morph.html#chdebahh"><em>About smoothing</em></a>) to the labels before morphing them. This is
advisable because otherwise the resulting labels may have little
holes in them since the morphing map is not a bijection. By default,
two smoothsteps are taken.</div></blockquote>
<p>As the labels are morphed, a directory with the name of the
subject specified with the <tt class="docutils literal"><span class="pre">--to</span></tt> option is created under
the directory specified with <tt class="docutils literal"><span class="pre">--labeldir</span></tt> to hold the
morphed labels.</p>
</div>
<div class="section" id="mne-organize-dicom">
<span id="id77"></span><h3><a class="toc-backref" href="#id172">mne_organize_dicom</a><a class="headerlink" href="#mne-organize-dicom" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="mne-prepare-bem-model">
<span id="id78"></span><h3><a class="toc-backref" href="#id173">mne_prepare_bem_model</a><a class="headerlink" href="#mne-prepare-bem-model" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--bem</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specify the name of the file containing the triangulations of the BEM
surfaces and the conductivities of the compartments. The standard
ending for this file is <tt class="docutils literal"><span class="pre">-bem.fif</span></tt> and it is produced
either with the utility <a class="reference internal" href="#mne-surf2bem"><em>mne_surf2bem</em></a> (<a class="reference internal" href="source_localization/forward.html#behcaccj"><em>Creating the BEM meshes</em></a>) or the
convenience script <a class="reference internal" href="#mne-setup-forward-model"><em>mne_setup_forward_model</em></a>,
see <a class="reference internal" href="cookbook.html#cihdbfeg"><em>Setting up the boundary-element model</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--sol</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specify the name of the file containing the triangulation and conductivity
information together with the BEM geometry matrix computed by mne_prepare_bem_model .
The standard ending for this file is <tt class="docutils literal"><span class="pre">-bem-sol.fif</span></tt> .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--method</span> <span class="pre">&lt;*approximation</span> <span class="pre">method*&gt;</span></tt></p>
<blockquote>
<div>Select the BEM approach. If &lt;<em>approximation method</em>&gt; is <tt class="docutils literal"><span class="pre">constant</span></tt> ,
the BEM basis functions are constant functions on each triangle
and the collocation points are the midpoints of the triangles. With <tt class="docutils literal"><span class="pre">linear</span></tt> ,
the BEM basis functions are linear functions on each triangle and
the collocation points are the vertices of the triangulation. This
is the preferred method to use. The accuracy will be the same or
better than in the constant collocation approach with about half
the number of unknowns in the BEM equations.</div></blockquote>
</div>
<div class="section" id="mne-process-stc">
<span id="id79"></span><h3><a class="toc-backref" href="#id174">mne_process_stc</a><a class="headerlink" href="#mne-process-stc" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--stc</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Specify the stc file to process.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--out</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Specify a stc  output file name.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--outasc</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Specify a text output file name.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--scaleto</span> <span class="pre">&lt;scale&gt;</span></tt></p>
<blockquote>
<div>Scale the data so that the maximum is this value.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--scaleby</span> <span class="pre">&lt;scale&gt;</span></tt></p>
<blockquote>
<div>Multiply the values by this.</div></blockquote>
</div>
<div class="section" id="mne-raw2mat">
<span id="id80"></span><h3><a class="toc-backref" href="#id175">mne_raw2mat</a><a class="headerlink" href="#mne-raw2mat" title="Permalink to this headline">¶</a></h3>
<p>The utility mne_raw2mat converts
all or selected channels from a raw data file to a Matlab mat file.
In addition, this utility can provide information about the raw
data file so that the raw data can be read directly from the original
fif file using Matlab file I/O routines.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The MNE Matlab toolbox described in <a class="reference internal" href="matlab.html#ch-matlab"><em>Matlab toolbox</em></a> provides    direct access to raw fif files without a need for conversion to    mat file format first. Therefore, it is recommended that you use    the Matlab toolbox rather than  mne_raw2mat which    creates large files occupying disk space unnecessarily.</p>
</div>
<div class="section" id="id81">
<h4>Command-line options<a class="headerlink" href="#id81" title="Permalink to this headline">¶</a></h4>
<p>mne_raw2mat accepts the
following command-line options:</p>
<p><tt class="docutils literal"><span class="pre">--raw</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of the raw data fif file to convert.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--mat</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of the destination Matlab file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--info</span></tt></p>
<blockquote>
<div>With this option present, only information about the raw data file
is included. The raw data itself is omitted.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--sel</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a text file which contains the names of the channels to include
in the output file, one channel name per line. If the <tt class="docutils literal"><span class="pre">--info</span></tt> option
is specified, <tt class="docutils literal"><span class="pre">--sel</span></tt> does not have any effect.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--tag</span> <span class="pre">&lt;*tag*&gt;</span></tt></p>
<blockquote>
<div>By default, all Matlab variables included in the output file start
with <tt class="docutils literal"><span class="pre">mne\_</span></tt>. This option changes the prefix to &lt;<em>tag</em>&gt; _.</div></blockquote>
</div>
<div class="section" id="id82">
<h4>Matlab data structures<a class="headerlink" href="#id82" title="Permalink to this headline">¶</a></h4>
<p>The Matlab files output by mne_raw2mat can
contain two data structures, &lt;<em>tag</em>&gt;_raw and &lt;<em>tag</em>&gt;_raw_info .
If <tt class="docutils literal"><span class="pre">--info</span></tt> option is specifed, the file contains the
latter structure only.</p>
<p>The &lt;<em>tag</em>&gt;_raw structure
contains only one field, data which
is a matrix containing the raw data. Each row of this matrix constitutes
the data from one channel in the original file. The data type of
this matrix is the same of the original data (2-byte signed integer,
4-byte signed integer, or single-precision float).</p>
<p>The fields of the &lt;<em>tag</em>&gt;_raw_info structure
are listed in <a class="reference internal" href="#behfdcih"><em>The fields of the raw data info structure.</em></a>. Further explanation of the bufs field
is provided in <a class="reference internal" href="#behjeihj"><em>The bufs member of the raw data info structure.</em></a>.</p>
<table border="1" class="docutils" id="behfdcih">
<caption>The fields of the raw data info structure.</caption>
<colgroup>
<col width="23%" />
<col width="17%" />
<col width="60%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td>Variable</td>
<td>Size</td>
<td>Description</td>
</tr>
<tr class="row-even"><td>orig_file</td>
<td>string</td>
<td>The name of the original fif file specified with the
<tt class="docutils literal"><span class="pre">--raw</span></tt> option.</td>
</tr>
<tr class="row-odd"><td>nchan</td>
<td>1</td>
<td>Number of channels.</td>
</tr>
<tr class="row-even"><td>nsamp</td>
<td>1</td>
<td>Total number of samples</td>
</tr>
<tr class="row-odd"><td>bufs</td>
<td>nbuf x 4</td>
<td>This field is present if <tt class="docutils literal"><span class="pre">--info</span></tt> option was specified on
the command line. For details, see <a class="reference internal" href="#behjeihj"><em>The bufs member of the raw data info structure.</em></a>.</td>
</tr>
<tr class="row-even"><td>sfreq</td>
<td>1</td>
<td>The sampling frequency in Hz.</td>
</tr>
<tr class="row-odd"><td>lowpass</td>
<td>1</td>
<td>Lowpass filter frequency (Hz)</td>
</tr>
<tr class="row-even"><td>highpass</td>
<td>1</td>
<td>Highpass filter frequency (Hz)</td>
</tr>
<tr class="row-odd"><td>ch_names</td>
<td>nchan (string)</td>
<td>String array containing the names of the channels included</td>
</tr>
<tr class="row-even"><td>ch_types</td>
<td>nchan x 2</td>
<td>The column lists the types of the channesl (1 = MEG, 2 =
EEG). The second column lists the coil types, see
<a class="reference internal" href="source_localization/forward.html#bgbbhgec"><em>Normal coil descriptions.</em></a> and <a class="reference internal" href="source_localization/forward.html#chdbdfje"><em>Accurate coil descriptions</em></a>. For EEG electrodes,
this value equals one.</td>
</tr>
<tr class="row-odd"><td>ch_lognos</td>
<td>nchan x 1</td>
<td>Logical channel numbers as listed in the fiff file</td>
</tr>
<tr class="row-even"><td>ch_units</td>
<td>nchan x 2</td>
<td>Units and unit multipliers as listed in the fif file.
The unit of the data is listed in the first column
(T = 112, T/m = 201, V = 107). At present, the second
column will be always zero, <em>i.e.</em>, no unit multiplier.</td>
</tr>
<tr class="row-odd"><td>ch_pos</td>
<td>nchan x 12</td>
<td>The location information for each channel. The first three
values specify the origin of the sensor coordinate system
or the location of the electrode. For MEG channels, the
following nine number specify the <em>x</em>, <em>y</em>, and
<em>z</em>-direction unit vectors of the sensor coordinate system.
For EEG electrodes the first vector after the electrode
location specifies the location of the reference electrode.
If the reference is not specified this value is all zeroes.
The remaining unit vectors are irrelevant for EEG
electrodes.</td>
</tr>
<tr class="row-even"><td>ch_cals</td>
<td>nchan x 2</td>
<td>The raw data output by mne_raw2mat is uncalibrated.
The first column is the range member of the fiff data
structures and while the second is the cal member. To get
calibrared data values in the units given in ch_units from
the raw data, the data must be multiplied with the product
of range and cal .</td>
</tr>
<tr class="row-odd"><td>meg_head_trans</td>
<td>4 x 4</td>
<td>The coordinate frame transformation from the MEG device
coordinates to the MEG head coordinates.</td>
</tr>
</tbody>
</table>
<table border="1" class="docutils" id="behjeihj">
<caption>The bufs member of the raw data info structure.</caption>
<colgroup>
<col width="24%" />
<col width="76%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td>Column</td>
<td>Contents</td>
</tr>
<tr class="row-even"><td>1</td>
<td>The raw data type (2 or 16 = 2-byte signed integer, 3 = 4-byte signed
integer, 4 = single-precision float). All data in the fif file are
written in the big-endian byte order. The raw data are stored sample by
sample.</td>
</tr>
<tr class="row-odd"><td>2</td>
<td>Byte location of this buffer in the original fif file.</td>
</tr>
<tr class="row-even"><td>3</td>
<td>First sample of this buffer. Since raw data storing can be switched on
and off during the acquisition, there might be gaps between the end of
one buffer and the beginning of the next.</td>
</tr>
<tr class="row-odd"><td>4</td>
<td>Number of samples in the buffer.</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section" id="mne-rename-channels">
<span id="id83"></span><h3><a class="toc-backref" href="#id176">mne_rename_channels</a><a class="headerlink" href="#mne-rename-channels" title="Permalink to this headline">¶</a></h3>
<p>Sometimes it is necessary to change the names types of channels
in MEG/EEG data files. Such situations include:</p>
<ul class="simple">
<li>Designating an EEG as an EOG channel.
For example, the EOG channels are not recognized as such in the
fif files converted from CTF data files.</li>
<li>Changing the name of the digital trigger channel of interest
to STI 014 so that mne_browse_raw and mne_process_raw will
recognize the correct channel without the need to specify the <tt class="docutils literal"><span class="pre">--digtrig</span></tt>
option or the MNE_TRIGGER_CH_NAME environment variable every time a
data file is loaded.</li>
</ul>
<p>The utility mne_rename_channels was
designed to meet the above needs. It recognizes the following command-line
options:</p>
<p><tt class="docutils literal"><span class="pre">--fif</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of the data file to modify.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--alias</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the text file which contains the modifications to be applied,
see below.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--revert</span></tt></p>
<blockquote>
<div>Reverse the roles of old and new channel names in the alias file.</div></blockquote>
<p>Each line in the alias file contains the old name and new
name for a channel, separated by a colon. The old name is a name
of one of the channels presently in the file and the new name is
the name to be assigned to it. The old name must match an existing
channel name in the file exactly. The new name may be followed by
another colon and a number which is the channel type to be assigned
to this channel. The channel type options are listed below.</p>
<table border="1" class="docutils">
<caption>Channel types.</caption>
<colgroup>
<col width="39%" />
<col width="61%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Channel type</th>
<th class="head">Corresponding number</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>MEG</td>
<td>1</td>
</tr>
<tr class="row-odd"><td>MCG</td>
<td>201</td>
</tr>
<tr class="row-even"><td>EEG</td>
<td>2</td>
</tr>
<tr class="row-odd"><td>EOG</td>
<td>202</td>
</tr>
<tr class="row-even"><td>EMG</td>
<td>302</td>
</tr>
<tr class="row-odd"><td>ECG</td>
<td>402</td>
</tr>
<tr class="row-even"><td>MISC</td>
<td>502</td>
</tr>
<tr class="row-odd"><td>STIM</td>
<td>3</td>
</tr>
</tbody>
</table>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">Do not attempt to designate MEG channels    to EEG channels or vice versa. This may result in strange errors    during source estimation.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">You might consider renaming the EEG channels    with descriptive labels related to the standard 10-20 system. This    allows you to use standard EEG channel names when defining derivations,    see <a class="reference internal" href="#mne-make-derivations"><em>mne_make_derivations</em></a> and <a class="reference internal" href="gui/browse.html#cacfhafh"><em>Load derivations</em></a>, as well as in the    channel selection files used in mne_browse_raw ,    see <a class="reference internal" href="gui/browse.html#caccjejd"><em>Selection</em></a>.</p>
</div>
</div>
<div class="section" id="mne-sensitivity-map">
<span id="id84"></span><h3><a class="toc-backref" href="#id177">mne_sensitivity_map</a><a class="headerlink" href="#mne-sensitivity-map" title="Permalink to this headline">¶</a></h3>
<div class="section" id="id85">
<h4>Purpose<a class="headerlink" href="#id85" title="Permalink to this headline">¶</a></h4>
<p>mne_sensitivity_map computes
the size of the columns of the forward operator and outputs the
result in w files.</p>
</div>
<div class="section" id="id86">
<h4>Command line options<a class="headerlink" href="#id86" title="Permalink to this headline">¶</a></h4>
<p>mne_sensitivity_map accepts
the following command-line options:</p>
<p><tt class="docutils literal"><span class="pre">--fwd</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a forward solution file to analyze. By default the MEG
forward solution is considered.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--proj</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a file containing an SSP operator to be applied. If necessary,
multiple <tt class="docutils literal"><span class="pre">--proj</span></tt> options can be specified. For map types 1 - 4 (see
below), SSP is applied to the forward model data. For map types
5 and 6, the effects of SSP are evaluated against the unmodified
forward model.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eeg</span></tt></p>
<blockquote>
<div>Use the EEG forward solution instead of the MEG one. It does not make
sense to consider a combination because of the different units of
measure. For the same reason, gradiometers and magnetometers have
to be handled separately, see <tt class="docutils literal"><span class="pre">--mag</span></tt> option below. By
default MEG gradiometers are included.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--mag</span></tt></p>
<blockquote>
<div>Include MEG magnetometers instead of gradiometers</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--w</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the stem of the output w files. To obtain the final output file
names, <tt class="docutils literal"><span class="pre">-lh.w</span></tt> and <tt class="docutils literal"><span class="pre">-rh.w</span></tt> is appended for
the left and right hemisphere, respectively.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--smooth</span>&nbsp; <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>Specifies the number of smooth steps to apply to the resulting w files.
Default: no smoothing.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--map</span>&nbsp; <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>Select the type of a sensitivity map to compute. At present, valid numbers
are 1 - 6. For details, see <a class="reference internal" href="#chdcdjij"><em>Available sensitivity maps</em></a>, below.</div></blockquote>
</div>
<div class="section" id="available-sensitivity-maps">
<span id="chdcdjij"></span><h4>Available sensitivity maps<a class="headerlink" href="#available-sensitivity-maps" title="Permalink to this headline">¶</a></h4>
<p>In the following, let</p>
<div class="math">
\[G_k = [g_{xk} g_{yk} g_{zk}]\]</div>
<p>denote the three consecutive columns of the gain matrix <span class="math">\(G\)</span> corresponding to
the fields of three orthogonal dipoles at source space location <span class="math">\(k\)</span>.
Further, lets assume that the source coordinate system has been
selected so that the <span class="math">\(z\)</span> -axis points
to the cortical normal direction and the <span class="math">\(xy\)</span> plane
is thus the tangent plane of the cortex at the source space location <span class="math">\(k\)</span>
Next, compute the SVD</p>
<div class="math">
\[G_k = U_k \Lambda_k V_k\]</div>
<p>and let <span class="math">\(g_{1k} = u_{1k} \lambda_{1k}\)</span>, where <span class="math">\(\lambda_{1k}\)</span> and <span class="math">\(u_{1k}\)</span> are
the largest singular value and the corresponding left singular vector
of <span class="math">\(G_k\)</span>, respectively. It is easy to see
that <span class="math">\(g_{1k}\)</span> is has the largest power
among the signal distributions produced by unit dipoles at source
space location <span class="math">\(k\)</span>.</p>
<p>Furthermore, assume that the colums orthogonal matrix <span class="math">\(U_P\)</span> (<span class="math">\(U_P^T U_P = I\)</span>) contain
the orthogonal basis of the noise subspace corresponding to the signal
space projection (SSP) operator <span class="math">\(P\)</span> specified
with one or more <tt class="docutils literal"><span class="pre">--proj</span></tt> options so that <span class="math">\(P = I - U_P U_P^T\)</span>.
For more information on SSP, see <a class="reference internal" href="gui/browse.html#cacchabi"><em>The Signal-Space Projection (SSP) method</em></a>.</p>
<p>With these definitions the map selections defined with the <tt class="docutils literal"><span class="pre">--map</span></tt> option correspond
to the following</p>
<p><tt class="docutils literal"><span class="pre">--map</span> <span class="pre">1</span></tt></p>
<blockquote>
<div>Compute <span class="math">\(\sqrt{g_{1k}^T g_{1k}} = \lambda_{1k}\)</span> at each source space point.
Normalize the result so that the maximum values equals one.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--map</span> <span class="pre">2</span></tt></p>
<blockquote>
<div>Compute <span class="math">\(\sqrt{g_z^T g_z}\)</span> at each source space point.
Normalize the result so that the maximum values equals one. This
is the amplitude of the signals produced by unit dipoles normal
to the cortical surface.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--map</span> <span class="pre">3</span></tt></p>
<blockquote>
<div>Compute <span class="math">\(\sqrt{g_z^T g_z / g_{1k}^T g_{1k}}\)</span> at each source space point.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--map</span> <span class="pre">4</span></tt></p>
<blockquote>
<div>Compute <span class="math">\(1 - \sqrt{g_z^T g_z / g_{1k}^T g_{1k}}\)</span> at each source space point.
This could be called the <em>radiality index</em>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--map</span> <span class="pre">5</span></tt></p>
<blockquote>
<div>Compute the subspace correlation between <span class="math">\(g_z\)</span> and <span class="math">\(U_P\)</span>: <span class="math">\(\text{subcorr}^2(g_z , U_P) = (g_z^T U_P U_P^T g_z)/(g_z^T g_z)\)</span>.
This index equals zero, if <span class="math">\(g_z\)</span> is
orthogonal to <span class="math">\(U_P\)</span> and one if <span class="math">\(g_z\)</span> lies
in the subspace defined by <span class="math">\(U_P\)</span>. This
map shows how close the field pattern of a dipole oriented perpendicular
to the cortex at each cortical location is to the subspace removed
by the SSP.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--map</span> <span class="pre">6</span></tt></p>
<blockquote>
<div>Compute <span class="math">\(\sqrt{g_z^T P g_z / g_z^T g_z}\)</span>, which is the fraction
of the field pattern of a dipole oriented perpendicular to the cortex
at each cortical location remaining after applying the SSP a dipole
remaining</div></blockquote>
</div>
</div>
<div class="section" id="mne-sensor-locations">
<span id="id87"></span><h3><a class="toc-backref" href="#id178">mne_sensor_locations</a><a class="headerlink" href="#mne-sensor-locations" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--meas</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Measurement file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--magonly</span></tt></p>
<blockquote>
<div>Magnetometers only.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--ell</span></tt></p>
<blockquote>
<div>Output sensor ellipsoids for MRIlab.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--dir</span></tt></p>
<blockquote>
<div>Output direction info as well.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--dev</span></tt></p>
<blockquote>
<div>Output in MEG device coordinates.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--out</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>Name output file</div></blockquote>
</div>
<div class="section" id="mne-show-fiff">
<span id="id88"></span><h3><a class="toc-backref" href="#id179">mne_show_fiff</a><a class="headerlink" href="#mne-show-fiff" title="Permalink to this headline">¶</a></h3>
<p>Using the utility mne_show_fiff it
is possible to display information about the contents of a fif file
to the standard output. The command line options for mne_show_fiff are:</p>
<p><tt class="docutils literal"><span class="pre">--in</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the fif file whose contents will be listed.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--verbose</span></tt></p>
<blockquote>
<div>Produce a verbose output. The data of most tags is included in the output.
This excludes matrices and vectors. Only the first 80 characters
of strings are listed unless the <tt class="docutils literal"><span class="pre">--long</span></tt> option is present.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--blocks</span></tt></p>
<blockquote>
<div>Only list the blocks (the tree structure) of the file. The tags
within each block are not listed.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--indent</span>&nbsp; <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>Number of spaces for indentation for each deeper level in the tree structure
of the fif files. The default indentation is 3 spaces in terse and
no spaces in verbose listing mode.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--long</span></tt></p>
<blockquote>
<div>List all data from string tags instead of the first 80 characters.
This options has no effect unless the <tt class="docutils literal"><span class="pre">--verbose</span></tt> option
is also present.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--tag</span>&nbsp; <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>List only tags of this kind. Multiple <tt class="docutils literal"><span class="pre">--tag</span></tt> options
can be specified to list several different kinds of data.</div></blockquote>
<p>mne_show_fiff reads the
explanations of tag kinds, block kinds, and units from <tt class="docutils literal"><span class="pre">$MNE_ROOT/share/mne/fiff_explanations.txt</span></tt> .</p>
</div>
<div class="section" id="mne-simu">
<span id="id89"></span><h3><a class="toc-backref" href="#id180">mne_simu</a><a class="headerlink" href="#mne-simu" title="Permalink to this headline">¶</a></h3>
<div class="section" id="id90">
<h4>Purpose<a class="headerlink" href="#id90" title="Permalink to this headline">¶</a></h4>
<p>The utility mne_simu creates
simulated evoked response data for investigation of the properties
of the inverse solutions. It computes MEG signals generated by dipoles
normal to the cortical mantle at one or several ROIs defined with
label files. Colored noise can be added to the signals.</p>
</div>
<div class="section" id="id91">
<h4>Command-line options<a class="headerlink" href="#id91" title="Permalink to this headline">¶</a></h4>
<p>mne_simu has the following
command-line options:</p>
<p><tt class="docutils literal"><span class="pre">--fwd</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specify a forward solution file to employ in the simulation.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--label</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specify a label</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--meg</span></tt></p>
<blockquote>
<div>Provide MEG data in the output file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--eeg</span></tt></p>
<blockquote>
<div>Provide EEG data in the output file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--out</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specify the output file. By default, this will be an evoked data
file in the fif format.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--raw</span></tt></p>
<blockquote>
<div>Output the data as a raw data fif file instead of an evoked one.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--mat</span></tt></p>
<blockquote>
<div>Produce Matlab output of the simulated fields instead of the fif evoked
file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--label</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Define an ROI. Several label files can be present. By default, the sources
in the labels will have <span class="math">\(\cos^2\)</span> -shaped non-overlapping
timecourses, see below.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--timecourse</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a text file which contains an expression for a source
time course, see <a class="reference internal" href="#chdcfibh"><em>Source waveform expressions</em></a>. If no &#8211;timecourse options
are present, the standard source time courses described in <a class="reference internal" href="#chdfiiii"><em>Simulated data</em></a> are used. Otherwise, the time course expressions
are read from the files specified. The time course expressions are
associated with the labels in the order they are specified. If the
number of expressions is smaller than the number of labels, the
last expression specified will reused for the remaining labels.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--sfreq</span>&nbsp; <span class="pre">&lt;*freq/Hz*&gt;</span></tt></p>
<blockquote>
<div>Specifies the sampling frequency of the output data (default = 1000 Hz). This
option is used only with the time course files.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--tmin</span>&nbsp; <span class="pre">&lt;*time/ms*&gt;</span></tt></p>
<blockquote>
<div>Specifies the starting time of the data, used only with time course files
(default -200 ms).</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--tmax</span>&nbsp; <span class="pre">&lt;*time/ms*&gt;</span></tt></p>
<blockquote>
<div>Specifies the ending time of the data, used only with time course files
(default 500 ms).</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--seed</span>&nbsp; <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>Specifies the seed for random numbers. This seed is used both for adding
noise, see <a class="reference internal" href="#chdfbjij"><em>Noise simulation</em></a> and for random numbers in source waveform
expressions, see <a class="reference internal" href="#chdcfibh"><em>Source waveform expressions</em></a>. If no seed is specified, the
current time in seconds since Epoch (January 1, 1970) is used.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--all</span></tt></p>
<blockquote>
<div>Activate all sources on the cortical surface uniformly. This overrides the <tt class="docutils literal"><span class="pre">--label</span></tt> options.</div></blockquote>
</div>
<div class="section" id="noise-simulation">
<span id="chdfbjij"></span><h4>Noise simulation<a class="headerlink" href="#noise-simulation" title="Permalink to this headline">¶</a></h4>
<p>Noise is added to the signals if the <tt class="docutils literal"><span class="pre">--senscov</span></tt> and <tt class="docutils literal"><span class="pre">--nave</span></tt> options
are present. If <tt class="docutils literal"><span class="pre">--nave</span></tt> is omitted the number of averages
is set to <span class="math">\(L = 100\)</span>. The noise is computed
by first generating vectors of Gaussian random numbers <span class="math">\(n(t)\)</span> with <span class="math">\(n_j(t) \sim N(0,1)\)</span>.
Thereafter, the noise-covariance matrix <span class="math">\(C\)</span> is
used to color the noise:</p>
<div class="math">
\[n_c(t) = \frac{1}{\sqrt{L}} \Lambda U^T n(t)\ ,\]</div>
<p>where we have used the eigenvalue decomposition positive-definite
covariance matrix:</p>
<div class="math">
\[C = U \Lambda^2 U^T\ .\]</div>
<p>Note that it is assumed that the noise-covariance matrix
is given for raw data, <em>i.e.</em>, for <span class="math">\(L = 1\)</span>.</p>
</div>
<div class="section" id="simulated-data">
<span id="chdfiiii"></span><h4>Simulated data<a class="headerlink" href="#simulated-data" title="Permalink to this headline">¶</a></h4>
<p>The default source waveform <span class="math">\(q_k\)</span> for
the <span class="math">\(k^{th}\)</span> label is nonzero at times <span class="math">\(t_{kp} = (100(k - 1) + p)/f_s\)</span>, <span class="math">\(p = 0 \dotso 100\)</span> with:</p>
<div class="math">
\[q_k(t_{kp}) = Q_k \cos^2{(\frac{\pi p}{100} - \frac{\pi}{2})}\ ,\]</div>
<p>i.e., the source waveforms are non-overlapping 100-samples
wide <span class="math">\(\cos^2\)</span> pulses. The sampling frequency <span class="math">\(f_s = 600\)</span> Hz.
The source amplitude <span class="math">\(Q_k\)</span> is determined
so that the strength of each of the dipoles in a label will be <span class="math">\(50 \text{nAm}/N_k\)</span>.</p>
<p>Let us denote the sums of the magnetic fields and electric
potentials produced by the dipoles normal to the cortical mantle
at label <span class="math">\(k\)</span> by <span class="math">\(x_k\)</span>. The simulated
signals are then:</p>
<div class="math">
\[x(t_j) = \sum_{k = 1}^{N_s} {q_k(t_j) x_k + n_c(t_j)}\ ,\]</div>
<p>where <span class="math">\(N_s\)</span> is the number of
sources.</p>
</div>
<div class="section" id="source-waveform-expressions">
<span id="chdcfibh"></span><h4>Source waveform expressions<a class="headerlink" href="#source-waveform-expressions" title="Permalink to this headline">¶</a></h4>
<p>The <tt class="docutils literal"><span class="pre">--timecourse</span></tt> option provides flexible possibilities
to define the source waveforms in a functional form. The source
waveform expression files consist of lines of the form:</p>
<blockquote>
<div>&lt;<em>variable</em>&gt; <tt class="docutils literal"><span class="pre">=</span></tt>  &lt;<em>arithmetic expression</em>&gt;</div></blockquote>
<p>Each file may contain multiple lines. At the end of the evaluation,
only the values in the variable <tt class="docutils literal"><span class="pre">y</span></tt> (<tt class="docutils literal"><span class="pre">q</span></tt> )
are significant, see <a class="reference internal" href="#chdjbiee"><em>Available variable names in source waveform expressions.</em></a>. They assume the role
of <span class="math">\(q_k(t_j)\)</span> to compute the simulated signals
as described in <a class="reference internal" href="#chdfiiii"><em>Simulated data</em></a>, above.</p>
<p>All expressions are case insensitive. The variables are vectors
with the length equal to the number of samples in the responses,
determined by the <tt class="docutils literal"><span class="pre">--tmin</span></tt> , <tt class="docutils literal"><span class="pre">--tmax</span></tt> , and <tt class="docutils literal"><span class="pre">--sfreq</span></tt> options.
The available variables are listed in <a class="reference internal" href="#chdjbiee"><em>Available variable names in source waveform expressions.</em></a>.</p>
<table border="1" class="docutils" id="chdjbiee">
<caption>Available variable names in source waveform expressions.</caption>
<colgroup>
<col width="29%" />
<col width="71%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Variable</th>
<th class="head">Meaning</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>x</td>
<td>time [s]</td>
</tr>
<tr class="row-odd"><td>t</td>
<td>current value of x in [ms]</td>
</tr>
<tr class="row-even"><td>y</td>
<td>the source amplitude [Am]</td>
</tr>
<tr class="row-odd"><td>q</td>
<td>synonym for y</td>
</tr>
<tr class="row-even"><td>a , b , c , d</td>
<td>help variables, initialized to zeros</td>
</tr>
</tbody>
</table>
<p>The arithmetic expressions can use usual arithmetic operations
as well as  mathematical functions listed in <a class="reference internal" href="#chdjibha"><em>Mathematical functions available for source waveform expressions</em></a>.
The arguments can be vectors or scalar numbers. In addition, standard
relational operators ( &lt;, &gt;, ==, &lt;=, &gt;=) and their textual
equivalents (lt, gt, eq, le, ge) are available. Table <a class="reference internal" href="#chddjehh"><em>Examples of source waveform expressions.</em></a> gives some
useful examples of source waveform expressions.</p>
<table border="1" class="docutils" id="chdjibha">
<caption>Mathematical functions available for source waveform expressions</caption>
<colgroup>
<col width="27%" />
<col width="73%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td>Function</td>
<td>Description</td>
</tr>
<tr class="row-even"><td>abs(x)</td>
<td>absolute value</td>
</tr>
<tr class="row-odd"><td>acos(x)</td>
<td><span class="math">\(\cos^{-1}x\)</span></td>
</tr>
<tr class="row-even"><td>asin(x)</td>
<td><span class="math">\(\sin^{-1}x\)</span></td>
</tr>
<tr class="row-odd"><td>atan(x)</td>
<td><span class="math">\(\tan^{-1}x\)</span></td>
</tr>
<tr class="row-even"><td>atan2(x,y)</td>
<td><span class="math">\(\tan^{-1}(^y/_x)\)</span></td>
</tr>
<tr class="row-odd"><td>ceil(x)</td>
<td>nearest integer larger than <span class="math">\(x\)</span></td>
</tr>
<tr class="row-even"><td>cos(x)</td>
<td><span class="math">\(\cos x\)</span></td>
</tr>
<tr class="row-odd"><td>cosw(x,a,b,c)</td>
<td><span class="math">\(\cos^2\)</span> -shaped window centered at <span class="math">\(b\)</span> with a
rising slope of length <span class="math">\(a\)</span> and a trailing slope of
length <span class="math">\(b\)</span>.</td>
</tr>
<tr class="row-even"><td>deg(x)</td>
<td>The value of <span class="math">\(x\)</span> converted to from radians to degrees</td>
</tr>
<tr class="row-odd"><td>erf(x)</td>
<td><span class="math">\(\frac{1}{2\pi} \int_0^x{\text{exp}(-t^2)dt}\)</span></td>
</tr>
<tr class="row-even"><td>erfc(x)</td>
<td><span class="math">\(1 - \text{erf}(x)\)</span></td>
</tr>
<tr class="row-odd"><td>exp(x)</td>
<td><span class="math">\(e^x\)</span></td>
</tr>
<tr class="row-even"><td>floor(x)</td>
<td>Largest integer value not larger than <span class="math">\(x\)</span></td>
</tr>
<tr class="row-odd"><td>hypot(x,y)</td>
<td><span class="math">\(\sqrt{x^2 + y^2}\)</span></td>
</tr>
<tr class="row-even"><td>ln(x)</td>
<td><span class="math">\(\ln x\)</span></td>
</tr>
<tr class="row-odd"><td>log(x)</td>
<td><span class="math">\(\log_{10} x\)</span></td>
</tr>
<tr class="row-even"><td>maxp(x,y)</td>
<td>Takes the maximum between <span class="math">\(x\)</span> and <span class="math">\(y\)</span></td>
</tr>
<tr class="row-odd"><td>minp(x,y)</td>
<td>Takes the minimum between <span class="math">\(x\)</span> and <span class="math">\(y\)</span></td>
</tr>
<tr class="row-even"><td>mod(x,y)</td>
<td>Gives the remainder of  <span class="math">\(x\)</span> divided by <span class="math">\(y\)</span></td>
</tr>
<tr class="row-odd"><td>pi</td>
<td>Ratio of the circumference of a circle and its diameter.</td>
</tr>
<tr class="row-even"><td>rand</td>
<td>Gives a vector of uniformly distributed random numbers
from 0 to 1.</td>
</tr>
<tr class="row-odd"><td>rnorm(x,y)</td>
<td>Gives a vector of Gaussian random numbers distributed as
<span class="math">\(N(x,y)\)</span>. Note that if <span class="math">\(x\)</span> and <span class="math">\(y\)</span> are
vectors, each number generated will a different mean and
variance according to the arguments.</td>
</tr>
<tr class="row-even"><td>shift(x,s)</td>
<td>Shifts the values in the input vector <span class="math">\(x\)</span> by the number
of positions given by <span class="math">\(s\)</span>. Note that <span class="math">\(s\)</span> must be
a scalar.</td>
</tr>
<tr class="row-odd"><td>sin(x)</td>
<td><span class="math">\(\sin x\)</span></td>
</tr>
<tr class="row-even"><td>sqr(x)</td>
<td><span class="math">\(x^2\)</span></td>
</tr>
<tr class="row-odd"><td>sqrt(x)</td>
<td><span class="math">\(\sqrt{x}\)</span></td>
</tr>
<tr class="row-even"><td>tan(x)</td>
<td><span class="math">\(\tan x\)</span></td>
</tr>
</tbody>
</table>
<table border="1" class="docutils" id="chddjehh">
<caption>Examples of source waveform expressions.</caption>
<colgroup>
<col width="42%" />
<col width="58%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td>Expression</td>
<td>Meaning</td>
</tr>
<tr class="row-even"><td>q = 20e-9*sin(2*pi*10*x)</td>
<td>A 10-Hz sine wave with 20 nAm amplitude</td>
</tr>
<tr class="row-odd"><td>q = 20e-9*sin(2*pi*2*x)*sin(2*pi*10*x)</td>
<td>A 10-Hz 20-nAm sine wave, amplitude modulated
sinusoidally at 2 Hz.</td>
</tr>
<tr class="row-even"><td>q = 20e-9*cosw(t,100,100,100)</td>
<td><span class="math">\(\cos^2\)</span>-shaped pulse, centered at <span class="math">\(t\)</span> = 100 ms
with 100 ms leading and trailing slopes, 20 nAm amplitude</td>
</tr>
<tr class="row-odd"><td>q = 30e-9*(t &gt; 0)*(t  &lt;* 300)*sin(2*pi*20*x)</td>
<td>20-Hz sine wave, 30 nAm amplitude, cropped in time to
0...300 ms.</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section" id="mne-smooth">
<span id="id92"></span><h3><a class="toc-backref" href="#id181">mne_smooth</a><a class="headerlink" href="#mne-smooth" title="Permalink to this headline">¶</a></h3>
<p>Produce a smoothed version of a w or an stc file</p>
<p><tt class="docutils literal"><span class="pre">--src</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>The source space file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--in</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>The w or stc file to smooth.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--smooth</span> <span class="pre">&lt;val&gt;</span></tt></p>
<blockquote>
<div>Number of smoothsteps</div></blockquote>
</div>
<div class="section" id="mne-surf2bem">
<span id="id93"></span><h3><a class="toc-backref" href="#id182">mne_surf2bem</a><a class="headerlink" href="#mne-surf2bem" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--surf</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a FreeSurfer binary format surface file. Before specifying the
next surface (<tt class="docutils literal"><span class="pre">--surf</span></tt> or <tt class="docutils literal"><span class="pre">--tri</span></tt> options)
details of the surface specification can be given with the options
listed in <a class="reference internal" href="#behcdicc"><em>Surface options</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--tri</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a text format surface file. Before specifying the next
surface (<tt class="docutils literal"><span class="pre">--surf</span></tt> or <tt class="docutils literal"><span class="pre">--tri</span></tt> options) details
of the surface specification can be given with the options listed
in <a class="reference internal" href="#behcdicc"><em>Surface options</em></a>. The format of these files is described
in <a class="reference internal" href="source_localization/forward.html#behdefcd"><em>Tessellation file format</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--check</span></tt></p>
<blockquote>
<div>Check that the surfaces are complete and that they do not intersect. This
is a recommended option. For more information, see <a class="reference internal" href="source_localization/forward.html#behcbdde"><em>Topology checks</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--checkmore</span></tt></p>
<blockquote>
<div>In addition to the checks implied by the <tt class="docutils literal"><span class="pre">--check</span></tt> option,
check skull and skull thicknesses. For more information, see <a class="reference internal" href="source_localization/forward.html#behcbdde"><em>Topology checks</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--fif</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>The output fif file containing the BEM. These files normally reside in
the bem subdirectory under the subject&#8217;s mri data. A name
ending with <tt class="docutils literal"><span class="pre">-bem.fif</span></tt> is recommended.</div></blockquote>
<div class="section" id="surface-options">
<span id="behcdicc"></span><h4>Surface options<a class="headerlink" href="#surface-options" title="Permalink to this headline">¶</a></h4>
<p>These options can be specified after each <tt class="docutils literal"><span class="pre">--surf</span></tt> or <tt class="docutils literal"><span class="pre">--tri</span></tt> option
to define details for the corresponding surface.</p>
<p><tt class="docutils literal"><span class="pre">--swap</span></tt></p>
<blockquote>
<div>Swap the ordering or the triangle vertices. The standard convention in
the MNE software is to have the vertices ordered so that the vector
cross product of the vectors from vertex 1 to 2 and 1 to 3 gives the
direction of the outward surface normal. Text format triangle files
produced by the some software packages have an opposite order. For
these files, the <tt class="docutils literal"><span class="pre">--swap</span></tt> . option is required. This option does
not have any effect on the interpretation of the FreeSurfer surface
files specified with the <tt class="docutils literal"><span class="pre">--surf</span></tt> option.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--sigma</span> <span class="pre">&lt;*value*&gt;</span></tt></p>
<blockquote>
<div>The conductivity of the compartment inside this surface in S/m.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--shift</span> <span class="pre">&lt;*value/mm*&gt;</span></tt></p>
<blockquote>
<div>Shift the vertices of this surface by this amount, given in mm,
in the outward direction, <em>i.e.</em>, in the positive
vertex normal direction.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--meters</span></tt></p>
<blockquote>
<div>The vertex coordinates of this surface are given in meters instead
of millimeters. This option applies to text format files only. This
definition does not affect the units of the shift option.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--id</span> <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>Identification number to assign to this surface. (1 = inner skull, 3
= outer skull, 4 = scalp).</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--ico</span> <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>Downsample the surface to the designated subdivision of an icosahedron.
This option is relevant (and required) only if the triangulation
is isomorphic with a recursively subdivided icosahedron. For example,
the surfaces produced by with mri_watershed are
isomorphic with the 5th subdivision of a an icosahedron thus containing 20480
triangles. However, this number of triangles is too large for present
computers. Therefore, the triangulations have to be decimated. Specifying <tt class="docutils literal"><span class="pre">--ico</span> <span class="pre">4</span></tt> yields 5120 triangles per surface while <tt class="docutils literal"><span class="pre">--ico</span> <span class="pre">3</span></tt> results
in 1280 triangles. The recommended choice is <tt class="docutils literal"><span class="pre">--ico</span> <span class="pre">4</span></tt> .</div></blockquote>
</div>
</div>
<div class="section" id="mne-toggle-skips">
<span id="id94"></span><h3><a class="toc-backref" href="#id183">mne_toggle_skips</a><a class="headerlink" href="#mne-toggle-skips" title="Permalink to this headline">¶</a></h3>
<p>Toggle skip tags on and off.</p>
<p><tt class="docutils literal"><span class="pre">--raw</span> <span class="pre">&lt;name&gt;</span></tt></p>
<blockquote>
<div>The raw data file to process.</div></blockquote>
</div>
<div class="section" id="mne-transform-points">
<span id="id95"></span><h3><a class="toc-backref" href="#id184">mne_transform_points</a><a class="headerlink" href="#mne-transform-points" title="Permalink to this headline">¶</a></h3>
<div class="section" id="id96">
<h4>Purpose<a class="headerlink" href="#id96" title="Permalink to this headline">¶</a></h4>
<p>mne_transform_points applies
the coordinate transformation relating the MEG head coordinates
and the MRI coordinates to a set of locations listed in a text file.</p>
</div>
<div class="section" id="id97">
<h4>Command line options<a class="headerlink" href="#id97" title="Permalink to this headline">¶</a></h4>
<p>mne_transform_points accepts
the following command-line options:</p>
<p><tt class="docutils literal"><span class="pre">--in</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the input file. The file must contain three numbers on
each line which are the <em>x</em>, <em>y</em>,
and <em>z</em> coordinates of point in space. By default,
the input is in millimeters.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--iso</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a name of a fif file containing Isotrak data. If this
option is present file will be used as the input instead of the
text file specified with the <tt class="docutils literal"><span class="pre">--in</span></tt> option.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--trans</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of a fif file containing the coordinate transformation
between the MEG head coordinates and MRI coordinates. If this file
is not present, the transformation will be replaced by a unit transform.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--out</span>&nbsp; <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the output file. This file has the same format as the
input file.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--hpts</span></tt></p>
<blockquote>
<div>Output the data in the head points (hpts)
format accepted by tkmedit . In
this format, the coordinates are preceded by a point category (hpi,
cardinal or fiducial, eeg, extra) and a sequence number, see <a class="reference internal" href="#cjadjebh"><em>The hpts format</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--meters</span></tt></p>
<blockquote>
<div>The coordinates are listed in meters rather than millimeters.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--tomri</span></tt></p>
<blockquote>
<div>By default, the coordinates are transformed from MRI coordinates to
MEG head coordinates. This option reverses the transformation to
be from MEG head coordinates to MRI coordinates.</div></blockquote>
</div>
</div>
<div class="section" id="mne-tufts2fiff">
<span id="id98"></span><h3><a class="toc-backref" href="#id185">mne_tufts2fiff</a><a class="headerlink" href="#mne-tufts2fiff" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--raw</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of the raw data file to process.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--cal</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>The name of the calibration data file. If calibration data are missing, the
calibration coefficients will be set to unity.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--elp</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>The name of the electrode location file. If this file is missing,
the electrode locations will be unspecified. This file is in the &#8220;probe&#8221; file
format used by the <em>Source Signal Imaging, Inc.</em> software.
For description of the format, see <a class="reference external" href="http://www.sourcesignal.com/formats_probe.html">http://www.sourcesignal.com/formats_probe.html</a>.
The fiducial marker locations, optional in the &#8220;probe&#8221; file
format specification are mandatory for mne_tufts2fiff . Note
that some other software packages may produce electrode-position
files with the elp ending not
conforming to the above specification.</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The conversion process includes a transformation from the Tufts head coordinate system convention to that used in    the Neuromag systems.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The fiducial landmark locations, optional in the probe file format, must be present for mne_tufts2fiff .</p>
</div>
</div>
<div class="section" id="mne-view-manual">
<span id="id99"></span><h3><a class="toc-backref" href="#id186">mne_view_manual</a><a class="headerlink" href="#mne-view-manual" title="Permalink to this headline">¶</a></h3>
<p>This script shows you the manual in a PDF reader.</p>
</div>
<div class="section" id="mne-volume-data2mri">
<span id="id100"></span><h3><a class="toc-backref" href="#id187">mne_volume_data2mri</a><a class="headerlink" href="#mne-volume-data2mri" title="Permalink to this headline">¶</a></h3>
<p>With help of the <a class="reference internal" href="#mne-volume-source-space"><em>mne_volume_source_space</em></a> utility
it is possible to create a source space which
is defined within a volume rather than a surface. If the <tt class="docutils literal"><span class="pre">--mri</span></tt> option
was used in <a class="reference internal" href="#mne-volume-source-space"><em>mne_volume_source_space</em></a>, the
source space file contains an interpolator matrix which performs
a trilinear interpolation into the voxel space of the MRI volume
specified.</p>
<p>The purpose of the <a class="reference internal" href="#mne-volume-data2mri"><em>mne_volume_data2mri</em></a> is
to produce MRI overlay data compatible with FreeSurfer MRI viewers
(in the mgh or mgz formats) from this type of w or stc files.</p>
<p>The command-line options are:</p>
<p><tt class="docutils literal"><span class="pre">--src</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>The name of the volumetric source space file created with mne_volume_source_space .
The source space must have been created with the <tt class="docutils literal"><span class="pre">--mri</span></tt> option,
which adds the appropriate sparse trilinear interpolator matrix
to the source space.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--w</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>The name of a w file to convert
into an MRI overlay.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--stc</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>The name of the stc file to convert
into an MRI overlay. If this file has many time frames, the output
file may be huge. Note: If both <tt class="docutils literal"><span class="pre">-w</span></tt> and <tt class="docutils literal"><span class="pre">--stc</span></tt> are
specified, <tt class="docutils literal"><span class="pre">-w</span></tt> takes precedence.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--scale</span> <span class="pre">&lt;*number*&gt;</span></tt></p>
<blockquote>
<div>Multiply the stc or w by
this scaling constant before producing the overlay.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--out</span> <span class="pre">&lt;*filename*&gt;</span></tt></p>
<blockquote>
<div>Specifies the name of the output MRI overlay file. The name must end
with either <tt class="docutils literal"><span class="pre">.mgh</span></tt> or <tt class="docutils literal"><span class="pre">.mgz</span></tt> identifying the
uncompressed and compressed FreeSurfer MRI formats, respectively.</div></blockquote>
</div>
<div class="section" id="mne-volume-source-space">
<span id="id101"></span><h3><a class="toc-backref" href="#id188">mne_volume_source_space</a><a class="headerlink" href="#mne-volume-source-space" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--surf</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a FreeSurfer surface file containing the surface which
will be used as the boundary for the source space.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--bem</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a BEM file (ending in <tt class="docutils literal"><span class="pre">-bem.fif</span></tt> ). The inner
skull surface will be used as the boundary for the source space.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--origin</span> <span class="pre">&lt;*x/mm*&gt;</span> <span class="pre">:</span> <span class="pre">&lt;*y/mm*&gt;</span> <span class="pre">:</span> <span class="pre">&lt;*z/mm*&gt;</span></tt></p>
<blockquote>
<div>If neither of the two surface options described above is present,
the source space will be spherical with the origin at this location,
given in MRI (RAS) coordinates.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--rad</span> <span class="pre">&lt;*radius/mm*&gt;</span></tt></p>
<blockquote>
<div>Specifies the radius of a spherical source space. Default value
= 90 mm</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--grid</span> <span class="pre">&lt;*spacing/mm*&gt;</span></tt></p>
<blockquote>
<div>Specifies the grid spacing in the source space.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--mindist</span> <span class="pre">&lt;*distance/mm*&gt;</span></tt></p>
<blockquote>
<div>Only points which are further than this distance from the bounding surface
are included. Default value = 5 mm.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--exclude</span> <span class="pre">&lt;*distance/mm*&gt;</span></tt></p>
<blockquote>
<div>Exclude points that are closer than this distance to the center
of mass of the bounding surface. By default, there will be no exclusion.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--mri</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a MRI volume (in mgz or mgh format).
If this argument is present the output source space file will contain
a (sparse) interpolation matrix which allows mne_volume_data2mri to
create an MRI overlay file, see <a class="reference internal" href="#mne-volume-data2mri"><em>mne_volume_data2mri</em></a>.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--pos</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies a name of a text file containing the source locations
and, optionally, orientations. Each line of the file should contain
3 or 6 values. If the number of values is 3, they indicate the source
location, in millimeters. The orientation of the sources will be
set to the z-direction. If the number of values is 6, the source
orientation will be parallel to the vector defined by the remaining
3 numbers on each line. With <tt class="docutils literal"><span class="pre">--pos</span></tt> , all of the options
defined above will be ignored. By default, the source position and
orientation data are assumed to be given in MRI coordinates.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--head</span></tt></p>
<blockquote>
<div>If this option is present, the source locations and orientations
in the file specified with the <tt class="docutils literal"><span class="pre">--pos</span></tt> option are assumed
to be given in the MEG head coordinates.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--meters</span></tt></p>
<blockquote>
<div>Indicates that the source locations in the file defined with the <tt class="docutils literal"><span class="pre">--pos</span></tt> option
are give in meters instead of millimeters.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--src</span> <span class="pre">&lt;*name*&gt;</span></tt></p>
<blockquote>
<div>Specifies the output file name. Use a name * &lt;<em>dir</em>&gt;/ &lt;<em>name</em>&gt;*-src.fif</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--all</span></tt></p>
<blockquote>
<div>Include all vertices in the output file, not just those in use.
This option is implied when the <tt class="docutils literal"><span class="pre">--mri</span></tt> option is present.
Even with the <tt class="docutils literal"><span class="pre">--all</span></tt> option, only those vertices actually
selected will be marked to be &#8220;in use&#8221; in the
output source space file.</div></blockquote>
</div>
<div class="section" id="mne-watershed-bem">
<span id="id102"></span><h3><a class="toc-backref" href="#id189">mne_watershed_bem</a><a class="headerlink" href="#mne-watershed-bem" title="Permalink to this headline">¶</a></h3>
<p><tt class="docutils literal"><span class="pre">--subject</span>&nbsp; <span class="pre">&lt;*subject*&gt;</span></tt></p>
<blockquote>
<div>Defines the name of the subject. This can be also accomplished
by setting the SUBJECT environment variable.</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--overwrite</span></tt></p>
<blockquote>
<div>Overwrite the results of previous run of mne_watershed_bem .</div></blockquote>
<p><tt class="docutils literal"><span class="pre">--atlas</span></tt></p>
<blockquote>
<div>Makes mri_watershed to employ
atlas information to correct the segmentation.</div></blockquote>
</div>
</div>
</div>


    </div>
    
  </div>
</div>
<footer class="footer">
  <div class="container">
    <p class="pull-right">
      <a href="#">Back to top</a>
      
        <br/>
        
      
    </p>
    <p>
        &copy; Copyright 2012-2016, MNE Developers. Last updated on 2016-05-11.<br/>
    </p>
  </div>
</footer>
  </body>
</html>